// zudoku/dist/lib/plugins/openapi/graphql/graphql.js
export const OperationsFragmentFragmentDoc = {
    kind: "Document",
    definitions: [
        {
            kind: "FragmentDefinition",
            name: { kind: "Name", value: "OperationsFragment" },
            typeCondition: {
                kind: "NamedType",
                name: { kind: "Name", value: "OperationItem" },
            },
            selectionSet: {
                kind: "SelectionSet",
                selections: [
                    { kind: "Field", name: { kind: "Name", value: "slug" } },
                    { kind: "Field", name: { kind: "Name", value: "summary" } },
                    { kind: "Field", name: { kind: "Name", value: "method" } },
                    { kind: "Field", name: { kind: "Name", value: "description" } },
                    { kind: "Field", name: { kind: "Name", value: "operationId" } },
                    { kind: "Field", name: { kind: "Name", value: "contentTypes" } },
                    { kind: "Field", name: { kind: "Name", value: "path" } },
                    {
                        kind: "Field",
                        name: { kind: "Name", value: "parameters" },
                        selectionSet: {
                            kind: "SelectionSet",
                            selections: [
                                { kind: "Field", name: { kind: "Name", value: "name" } },
                                { kind: "Field", name: { kind: "Name", value: "in" } },
                                { kind: "Field", name: { kind: "Name", value: "description" } },
                                { kind: "Field", name: { kind: "Name", value: "required" } },
                                { kind: "Field", name: { kind: "Name", value: "schema" } },
                                { kind: "Field", name: { kind: "Name", value: "style" } },
                                {
                                    kind: "Field",
                                    name: { kind: "Name", value: "examples" },
                                    selectionSet: {
                                        kind: "SelectionSet",
                                        selections: [
                                            { kind: "Field", name: { kind: "Name", value: "name" } },
                                            {
                                                kind: "Field",
                                                name: { kind: "Name", value: "description" },
                                            },
                                            {
                                                kind: "Field",
                                                name: { kind: "Name", value: "externalValue" },
                                            },
                                            { kind: "Field", name: { kind: "Name", value: "value" } },
                                            {
                                                kind: "Field",
                                                name: { kind: "Name", value: "summary" },
                                            },
                                        ],
                                    },
                                },
                            ],
                        },
                    },
                    {
                        kind: "Field",
                        name: { kind: "Name", value: "requestBody" },
                        selectionSet: {
                            kind: "SelectionSet",
                            selections: [
                                {
                                    kind: "Field",
                                    name: { kind: "Name", value: "content" },
                                    selectionSet: {
                                        kind: "SelectionSet",
                                        selections: [
                                            {
                                                kind: "Field",
                                                name: { kind: "Name", value: "mediaType" },
                                            },
                                            {
                                                kind: "Field",
                                                name: { kind: "Name", value: "encoding" },
                                                selectionSet: {
                                                    kind: "SelectionSet",
                                                    selections: [
                                                        {
                                                            kind: "Field",
                                                            name: { kind: "Name", value: "name" },
                                                        },
                                                    ],
                                                },
                                            },
                                            {
                                                kind: "Field",
                                                name: { kind: "Name", value: "schema" },
                                            },
                                        ],
                                    },
                                },
                                { kind: "Field", name: { kind: "Name", value: "description" } },
                                { kind: "Field", name: { kind: "Name", value: "required" } },
                            ],
                        },
                    },
                    {
                        kind: "Field",
                        name: { kind: "Name", value: "responses" },
                        selectionSet: {
                            kind: "SelectionSet",
                            selections: [
                                { kind: "Field", name: { kind: "Name", value: "statusCode" } },
                                { kind: "Field", name: { kind: "Name", value: "links" } },
                                { kind: "Field", name: { kind: "Name", value: "description" } },
                                {
                                    kind: "Field",
                                    name: { kind: "Name", value: "content" },
                                    selectionSet: {
                                        kind: "SelectionSet",
                                        selections: [
                                            {
                                                kind: "Field",
                                                name: { kind: "Name", value: "mediaType" },
                                            },
                                            {
                                                kind: "Field",
                                                name: { kind: "Name", value: "encoding" },
                                                selectionSet: {
                                                    kind: "SelectionSet",
                                                    selections: [
                                                        {
                                                            kind: "Field",
                                                            name: { kind: "Name", value: "name" },
                                                        },
                                                    ],
                                                },
                                            },
                                            {
                                                kind: "Field",
                                                name: { kind: "Name", value: "schema" },
                                            },
                                        ],
                                    },
                                },
                            ],
                        },
                    },
                ],
            },
        },
    ],
};
export const ServersQueryDocument = {
    kind: "Document",
    definitions: [
        {
            kind: "OperationDefinition",
            operation: "query",
            name: { kind: "Name", value: "ServersQuery" },
            variableDefinitions: [
                {
                    kind: "VariableDefinition",
                    variable: {
                        kind: "Variable",
                        name: { kind: "Name", value: "input" },
                    },
                    type: {
                        kind: "NonNullType",
                        type: { kind: "NamedType", name: { kind: "Name", value: "JSON" } },
                    },
                },
                {
                    kind: "VariableDefinition",
                    variable: { kind: "Variable", name: { kind: "Name", value: "type" } },
                    type: {
                        kind: "NonNullType",
                        type: {
                            kind: "NamedType",
                            name: { kind: "Name", value: "SchemaType" },
                        },
                    },
                },
            ],
            selectionSet: {
                kind: "SelectionSet",
                selections: [
                    {
                        kind: "Field",
                        name: { kind: "Name", value: "schema" },
                        arguments: [
                            {
                                kind: "Argument",
                                name: { kind: "Name", value: "input" },
                                value: {
                                    kind: "Variable",
                                    name: { kind: "Name", value: "input" },
                                },
                            },
                            {
                                kind: "Argument",
                                name: { kind: "Name", value: "type" },
                                value: {
                                    kind: "Variable",
                                    name: { kind: "Name", value: "type" },
                                },
                            },
                        ],
                        selectionSet: {
                            kind: "SelectionSet",
                            selections: [
                                { kind: "Field", name: { kind: "Name", value: "url" } },
                                {
                                    kind: "Field",
                                    name: { kind: "Name", value: "servers" },
                                    selectionSet: {
                                        kind: "SelectionSet",
                                        selections: [
                                            { kind: "Field", name: { kind: "Name", value: "url" } },
                                        ],
                                    },
                                },
                            ],
                        },
                    },
                ],
            },
        },
    ],
};
export const AllOperationsDocument = {
    kind: "Document",
    definitions: [
        {
            kind: "OperationDefinition",
            operation: "query",
            name: { kind: "Name", value: "AllOperations" },
            variableDefinitions: [
                {
                    kind: "VariableDefinition",
                    variable: {
                        kind: "Variable",
                        name: { kind: "Name", value: "input" },
                    },
                    type: {
                        kind: "NonNullType",
                        type: { kind: "NamedType", name: { kind: "Name", value: "JSON" } },
                    },
                },
                {
                    kind: "VariableDefinition",
                    variable: { kind: "Variable", name: { kind: "Name", value: "type" } },
                    type: {
                        kind: "NonNullType",
                        type: {
                            kind: "NamedType",
                            name: { kind: "Name", value: "SchemaType" },
                        },
                    },
                },
            ],
            selectionSet: {
                kind: "SelectionSet",
                selections: [
                    {
                        kind: "Field",
                        name: { kind: "Name", value: "schema" },
                        arguments: [
                            {
                                kind: "Argument",
                                name: { kind: "Name", value: "input" },
                                value: {
                                    kind: "Variable",
                                    name: { kind: "Name", value: "input" },
                                },
                            },
                            {
                                kind: "Argument",
                                name: { kind: "Name", value: "type" },
                                value: {
                                    kind: "Variable",
                                    name: { kind: "Name", value: "type" },
                                },
                            },
                        ],
                        selectionSet: {
                            kind: "SelectionSet",
                            selections: [
                                { kind: "Field", name: { kind: "Name", value: "description" } },
                                { kind: "Field", name: { kind: "Name", value: "title" } },
                                { kind: "Field", name: { kind: "Name", value: "url" } },
                                { kind: "Field", name: { kind: "Name", value: "version" } },
                                {
                                    kind: "Field",
                                    name: { kind: "Name", value: "tags" },
                                    selectionSet: {
                                        kind: "SelectionSet",
                                        selections: [
                                            { kind: "Field", name: { kind: "Name", value: "name" } },
                                            {
                                                kind: "Field",
                                                name: { kind: "Name", value: "description" },
                                            },
                                            {
                                                kind: "Field",
                                                name: { kind: "Name", value: "operations" },
                                                selectionSet: {
                                                    kind: "SelectionSet",
                                                    selections: [
                                                        {
                                                            kind: "Field",
                                                            name: { kind: "Name", value: "slug" },
                                                        },
                                                        {
                                                            kind: "FragmentSpread",
                                                            name: {
                                                                kind: "Name",
                                                                value: "OperationsFragment",
                                                            },
                                                        },
                                                    ],
                                                },
                                            },
                                        ],
                                    },
                                },
                            ],
                        },
                    },
                ],
            },
        },
        {
            kind: "FragmentDefinition",
            name: { kind: "Name", value: "OperationsFragment" },
            typeCondition: {
                kind: "NamedType",
                name: { kind: "Name", value: "OperationItem" },
            },
            selectionSet: {
                kind: "SelectionSet",
                selections: [
                    { kind: "Field", name: { kind: "Name", value: "slug" } },
                    { kind: "Field", name: { kind: "Name", value: "summary" } },
                    { kind: "Field", name: { kind: "Name", value: "method" } },
                    { kind: "Field", name: { kind: "Name", value: "description" } },
                    { kind: "Field", name: { kind: "Name", value: "operationId" } },
                    { kind: "Field", name: { kind: "Name", value: "contentTypes" } },
                    { kind: "Field", name: { kind: "Name", value: "path" } },
                    {
                        kind: "Field",
                        name: { kind: "Name", value: "parameters" },
                        selectionSet: {
                            kind: "SelectionSet",
                            selections: [
                                { kind: "Field", name: { kind: "Name", value: "name" } },
                                { kind: "Field", name: { kind: "Name", value: "in" } },
                                { kind: "Field", name: { kind: "Name", value: "description" } },
                                { kind: "Field", name: { kind: "Name", value: "required" } },
                                { kind: "Field", name: { kind: "Name", value: "schema" } },
                                { kind: "Field", name: { kind: "Name", value: "style" } },
                                {
                                    kind: "Field",
                                    name: { kind: "Name", value: "examples" },
                                    selectionSet: {
                                        kind: "SelectionSet",
                                        selections: [
                                            { kind: "Field", name: { kind: "Name", value: "name" } },
                                            {
                                                kind: "Field",
                                                name: { kind: "Name", value: "description" },
                                            },
                                            {
                                                kind: "Field",
                                                name: { kind: "Name", value: "externalValue" },
                                            },
                                            { kind: "Field", name: { kind: "Name", value: "value" } },
                                            {
                                                kind: "Field",
                                                name: { kind: "Name", value: "summary" },
                                            },
                                        ],
                                    },
                                },
                            ],
                        },
                    },
                    {
                        kind: "Field",
                        name: { kind: "Name", value: "requestBody" },
                        selectionSet: {
                            kind: "SelectionSet",
                            selections: [
                                {
                                    kind: "Field",
                                    name: { kind: "Name", value: "content" },
                                    selectionSet: {
                                        kind: "SelectionSet",
                                        selections: [
                                            {
                                                kind: "Field",
                                                name: { kind: "Name", value: "mediaType" },
                                            },
                                            {
                                                kind: "Field",
                                                name: { kind: "Name", value: "encoding" },
                                                selectionSet: {
                                                    kind: "SelectionSet",
                                                    selections: [
                                                        {
                                                            kind: "Field",
                                                            name: { kind: "Name", value: "name" },
                                                        },
                                                    ],
                                                },
                                            },
                                            {
                                                kind: "Field",
                                                name: { kind: "Name", value: "schema" },
                                            },
                                        ],
                                    },
                                },
                                { kind: "Field", name: { kind: "Name", value: "description" } },
                                { kind: "Field", name: { kind: "Name", value: "required" } },
                            ],
                        },
                    },
                    {
                        kind: "Field",
                        name: { kind: "Name", value: "responses" },
                        selectionSet: {
                            kind: "SelectionSet",
                            selections: [
                                { kind: "Field", name: { kind: "Name", value: "statusCode" } },
                                { kind: "Field", name: { kind: "Name", value: "links" } },
                                { kind: "Field", name: { kind: "Name", value: "description" } },
                                {
                                    kind: "Field",
                                    name: { kind: "Name", value: "content" },
                                    selectionSet: {
                                        kind: "SelectionSet",
                                        selections: [
                                            {
                                                kind: "Field",
                                                name: { kind: "Name", value: "mediaType" },
                                            },
                                            {
                                                kind: "Field",
                                                name: { kind: "Name", value: "encoding" },
                                                selectionSet: {
                                                    kind: "SelectionSet",
                                                    selections: [
                                                        {
                                                            kind: "Field",
                                                            name: { kind: "Name", value: "name" },
                                                        },
                                                    ],
                                                },
                                            },
                                            {
                                                kind: "Field",
                                                name: { kind: "Name", value: "schema" },
                                            },
                                        ],
                                    },
                                },
                            ],
                        },
                    },
                ],
            },
        },
    ],
};
export const GetServerQueryDocument = {
    kind: "Document",
    definitions: [
        {
            kind: "OperationDefinition",
            operation: "query",
            name: { kind: "Name", value: "getServerQuery" },
            variableDefinitions: [
                {
                    kind: "VariableDefinition",
                    variable: {
                        kind: "Variable",
                        name: { kind: "Name", value: "input" },
                    },
                    type: {
                        kind: "NonNullType",
                        type: { kind: "NamedType", name: { kind: "Name", value: "JSON" } },
                    },
                },
                {
                    kind: "VariableDefinition",
                    variable: { kind: "Variable", name: { kind: "Name", value: "type" } },
                    type: {
                        kind: "NonNullType",
                        type: {
                            kind: "NamedType",
                            name: { kind: "Name", value: "SchemaType" },
                        },
                    },
                },
            ],
            selectionSet: {
                kind: "SelectionSet",
                selections: [
                    {
                        kind: "Field",
                        name: { kind: "Name", value: "schema" },
                        arguments: [
                            {
                                kind: "Argument",
                                name: { kind: "Name", value: "input" },
                                value: {
                                    kind: "Variable",
                                    name: { kind: "Name", value: "input" },
                                },
                            },
                            {
                                kind: "Argument",
                                name: { kind: "Name", value: "type" },
                                value: {
                                    kind: "Variable",
                                    name: { kind: "Name", value: "type" },
                                },
                            },
                        ],
                        selectionSet: {
                            kind: "SelectionSet",
                            selections: [
                                { kind: "Field", name: { kind: "Name", value: "url" } },
                                {
                                    kind: "Field",
                                    name: { kind: "Name", value: "servers" },
                                    selectionSet: {
                                        kind: "SelectionSet",
                                        selections: [
                                            { kind: "Field", name: { kind: "Name", value: "url" } },
                                        ],
                                    },
                                },
                            ],
                        },
                    },
                ],
            },
        },
    ],
};
export const GetCategoriesDocument = {
    kind: "Document",
    definitions: [
        {
            kind: "OperationDefinition",
            operation: "query",
            name: { kind: "Name", value: "GetCategories" },
            variableDefinitions: [
                {
                    kind: "VariableDefinition",
                    variable: {
                        kind: "Variable",
                        name: { kind: "Name", value: "input" },
                    },
                    type: {
                        kind: "NonNullType",
                        type: { kind: "NamedType", name: { kind: "Name", value: "JSON" } },
                    },
                },
                {
                    kind: "VariableDefinition",
                    variable: { kind: "Variable", name: { kind: "Name", value: "type" } },
                    type: {
                        kind: "NonNullType",
                        type: {
                            kind: "NamedType",
                            name: { kind: "Name", value: "SchemaType" },
                        },
                    },
                },
            ],
            selectionSet: {
                kind: "SelectionSet",
                selections: [
                    {
                        kind: "Field",
                        name: { kind: "Name", value: "schema" },
                        arguments: [
                            {
                                kind: "Argument",
                                name: { kind: "Name", value: "input" },
                                value: {
                                    kind: "Variable",
                                    name: { kind: "Name", value: "input" },
                                },
                            },
                            {
                                kind: "Argument",
                                name: { kind: "Name", value: "type" },
                                value: {
                                    kind: "Variable",
                                    name: { kind: "Name", value: "type" },
                                },
                            },
                        ],
                        selectionSet: {
                            kind: "SelectionSet",
                            selections: [
                                {
                                    kind: "Field",
                                    name: { kind: "Name", value: "tags" },
                                    selectionSet: {
                                        kind: "SelectionSet",
                                        selections: [
                                            {
                                                kind: "Field",
                                                name: { kind: "Name", value: "__typename" },
                                            },
                                            { kind: "Field", name: { kind: "Name", value: "name" } },
                                            {
                                                kind: "Field",
                                                name: { kind: "Name", value: "operations" },
                                                selectionSet: {
                                                    kind: "SelectionSet",
                                                    selections: [
                                                        {
                                                            kind: "Field",
                                                            name: { kind: "Name", value: "__typename" },
                                                        },
                                                        {
                                                            kind: "Field",
                                                            name: { kind: "Name", value: "slug" },
                                                        },
                                                        {
                                                            kind: "Field",
                                                            name: { kind: "Name", value: "deprecated" },
                                                        },
                                                        {
                                                            kind: "Field",
                                                            name: { kind: "Name", value: "method" },
                                                        },
                                                        {
                                                            kind: "Field",
                                                            name: { kind: "Name", value: "summary" },
                                                        },
                                                        {
                                                            kind: "Field",
                                                            name: { kind: "Name", value: "operationId" },
                                                        },
                                                        {
                                                            kind: "Field",
                                                            name: { kind: "Name", value: "path" },
                                                        },
                                                    ],
                                                },
                                            },
                                        ],
                                    },
                                },
                            ],
                        },
                    },
                ],
            },
        },
    ],
};
