// zudoku/lib/AuthenticationPlugin-CbgJ5SAh.js
import { j as i } from "./jsx-runtime-B6kdoens.js";
import { useEffect as o } from "react";
import { b as a, a as u } from "./index-BRCiYFaL.js";
import { u as s } from "./ZudokuContext-BEmsYQoq.js";
const r = () => {
  const t = s(), [n] = a();
  return o(() => {
    var e;
    (e = t.authentication) == null || e.signIn({
      redirectTo: n.get("redirect") ?? void 0
    });
  }, [t.authentication, n]), null;
}, c = () => {
  const t = s(), n = u();
  return o(() => {
    var e;
    (e = t.authentication) == null || e.signOut().then(() => n("/"));
  }, [n, t.authentication]), null;
}, g = () => {
  const t = s();
  return o(() => {
    var n, e;
    ((n = t.authentication) == null ? void 0 : n.signUp()) ?? ((e = t.authentication) == null || e.signIn());
  }, [t.authentication]), null;
};
class x {
  getRoutes() {
    return [
      {
        path: "/signout",
        element: /* @__PURE__ */ i.jsx(c, {})
      },
      {
        path: "/signin",
        element: /* @__PURE__ */ i.jsx(r, {})
      },
      {
        path: "/signup",
        element: /* @__PURE__ */ i.jsx(g, {})
      }
    ];
  }
  getProfileMenuItems() {
    return [
      {
        label: "Logout",
        path: "/signout"
      }
    ];
  }
}
export {
  x as A
};
