// zudoku/lib/MdxPage-C3tlrV4j.js
import { j as e } from "./jsx-runtime-B6kdoens.js";
import { s as T } from "./index-D06ATMgg.js";
import { A as k, u as P, a as C, H, b as R } from "./utils-Chi3p5nE.js";
import { L as v } from "./index-BRCiYFaL.js";
import { C as S } from "./CategoryHeading-3Qtp2yZ8.js";
import { u as E, a as x, P as I, H as h } from "./Markdown-CWI6lU11.js";
import { ListTreeIcon as L } from "lucide-react";
import { useRef as y, useState as _, useEffect as $ } from "react";
const N = "data-active", b = ({
  item: r,
  children: t,
  className: s,
  isActive: i
}) => /* @__PURE__ */ e.jsxs(
  "li",
  {
    className: x(
      "truncate",
      i ? "text-primary" : "text-foreground/65 dark:text-foreground/75",
      s
    ),
    title: r.value,
    children: [
      /* @__PURE__ */ e.jsx(
        k,
        {
          to: `#${r.id}`,
          [N]: r.id,
          className: x(
            i ? "text-primary" : "text-foreground/65 dark:text-foreground/75 hover:text-foreground"
          ),
          children: r.value
        }
      ),
      t
    ]
  }
), M = ({ entries: r }) => {
  const { activeAnchor: t } = E(), s = y(null), i = y(!1), [m, o] = _({
    top: 0,
    opacity: 0
  });
  return $(() => {
    if (!s.current) return;
    const a = s.current.querySelector(
      `[${N}='${t}']`
    );
    if (!a) {
      o({ top: 0, opacity: 0 });
      return;
    }
    const n = s.current.getBoundingClientRect().top, p = a.getBoundingClientRect().top;
    o({
      opacity: 1,
      top: `${p - n}px`
    }), !i.current && requestIdleCallback(() => {
      i.current = !0;
    });
  }, [t]), /* @__PURE__ */ e.jsxs("aside", { className: "sticky scrollbar top-[--header-height] h-[calc(100vh-var(--header-height))] pt-[--padding-content-top] pb-[--padding-content-bottom] overflow-y-auto ps-1 text-sm", children: [
    /* @__PURE__ */ e.jsxs("div", { className: "flex items-center gap-2 font-medium mb-2", children: [
      /* @__PURE__ */ e.jsx(L, { size: 16 }),
      "On this page"
    ] }),
    /* @__PURE__ */ e.jsxs("div", { className: "relative ms-2 ps-4", children: [
      /* @__PURE__ */ e.jsx("div", { className: "absolute inset-0 right-auto bg-border w-[2px]" }),
      /* @__PURE__ */ e.jsx(
        "div",
        {
          className: x(
            "absolute -left-px -translate-y-1 h-6 w-[4px] rounded bg-primary",
            i.current && "ease-out [transition:top_150ms,opacity_325ms]"
          ),
          style: m
        }
      ),
      /* @__PURE__ */ e.jsx(
        "ul",
        {
          ref: s,
          className: "relative font-medium list-none space-y-2",
          children: r.map((a) => /* @__PURE__ */ e.jsx(
            b,
            {
              isActive: a.id === t,
              item: a,
              className: "pl-0",
              children: a.children && /* @__PURE__ */ e.jsx("ul", { className: "list-none pl-4 pt-2 space-y-2", children: a.children.map((n) => /* @__PURE__ */ e.jsx(
                b,
                {
                  item: n,
                  isActive: n.id === t
                },
                n.id
              )) })
            },
            a.id
          ))
        }
      )
    ] })
  ] });
}, q = {
  h2: ({ children: r, id: t }) => /* @__PURE__ */ e.jsx(h, { level: 2, id: t, registerSidebarAnchor: !0, children: r }),
  h3: ({ children: r, id: t }) => /* @__PURE__ */ e.jsx(h, { level: 3, id: t, registerSidebarAnchor: !0, children: r })
}, J = ({
  mdxComponent: r,
  frontmatter: t = {},
  defaultOptions: s,
  tableOfContents: i
}) => {
  var g, j, f;
  const m = (g = P()) == null ? void 0 : g.categoryLabel, o = t.title, a = t.category ?? m, n = t.toc === !1 || (s == null ? void 0 : s.toc) === !1, p = ((j = i.find((l) => l.depth === 1)) == null ? void 0 : j.value) ?? o, w = t.disablePager ?? (s == null ? void 0 : s.disablePager) ?? !1, u = ((f = i.find((l) => l.depth === 1)) == null ? void 0 : f.children) ?? // if `title` is provided by frontmatter it does not appear in the table of contents
  i.filter((l) => l.depth === 2), A = !n && u.length > 0, { prev: c, next: d } = C();
  return /* @__PURE__ */ e.jsxs("div", { className: "xl:grid grid-cols-[--sidecar-grid-cols] gap-8 justify-between", children: [
    /* @__PURE__ */ e.jsx(H, { children: /* @__PURE__ */ e.jsx("title", { children: p }) }),
    /* @__PURE__ */ e.jsxs(
      "div",
      {
        className: x(
          I,
          "max-w-full xl:w-full xl:max-w-prose flex-1 flex-shrink pt-[--padding-content-top] pb-[--padding-content-bottom]"
        ),
        children: [
          /* @__PURE__ */ e.jsxs("header", { children: [
            a && /* @__PURE__ */ e.jsx(S, { children: a }),
            o && /* @__PURE__ */ e.jsx(h, { level: 1, id: T(o), children: o })
          ] }),
          /* @__PURE__ */ e.jsx(
            r,
            {
              components: { ...R(), ...q }
            }
          ),
          !w && /* @__PURE__ */ e.jsxs(e.Fragment, { children: [
            /* @__PURE__ */ e.jsx("hr", {}),
            /* @__PURE__ */ e.jsxs("div", { className: "not-prose flex flex-wrap items-center justify-between gap-2 lg:gap-8", children: [
              c ? /* @__PURE__ */ e.jsxs(
                v,
                {
                  to: c.id,
                  className: "flex flex-col items-stretch gap-2 flex-1 min-w-max border rounded px-6 py-4 text-start hover:border-primary/85 transition shadow-sm hover:shadow-md",
                  title: c.label,
                  children: [
                    /* @__PURE__ */ e.jsx("div", { className: "text-sm text-muted-foreground", children: "← Previous page" }),
                    /* @__PURE__ */ e.jsx("div", { className: "text-lg text-primary truncate", children: c.label })
                  ]
                }
              ) : /* @__PURE__ */ e.jsx("div", { className: "flex-1" }),
              d ? /* @__PURE__ */ e.jsxs(
                v,
                {
                  to: d.id,
                  className: "flex flex-col items-stretch gap-2 flex-1 min-w-max border rounded px-6 py-4 text-end hover:border-primary/85 transition shadow-sm hover:shadow-md",
                  title: d.label,
                  children: [
                    /* @__PURE__ */ e.jsx("div", { className: "text-sm text-muted-foreground", children: "Next page →" }),
                    /* @__PURE__ */ e.jsx("div", { className: "text-lg text-primary truncate", children: d.label })
                  ]
                }
              ) : /* @__PURE__ */ e.jsx("div", { className: "flex-1" })
            ] })
          ] })
        ]
      }
    ),
    /* @__PURE__ */ e.jsx("div", { className: "hidden xl:block", children: A && /* @__PURE__ */ e.jsx(M, { entries: u }) })
  ] });
};
export {
  J as MdxPage
};
