// zudoku/lib/OperationList-BO-ES1C5.js
import { j as e } from "./jsx-runtime-B6kdoens.js";
import { g as N, u as P, a as T, S as z, C as R, b as p, c as E, d as F, e as B, T as D, h as H, i as M, j as G, k as Q } from "./index-DNfiZTPV.js";
import { C as I } from "./CategoryHeading-3Qtp2yZ8.js";
import { D as V } from "./DeveloperHint-BE9DzNhv.js";
import { E as _ } from "./ErrorPage-B0COs372.js";
import { I as y, M as m, H as h, a as x, P as u, S as J } from "./Markdown-CWI6lU11.js";
import { CheckIcon as K, CopyIcon as U, ChevronDownIcon as W, CircleFadingPlusIcon as Z, CircleDotIcon as X, CircleIcon as Y, SquareMinusIcon as ee, SquarePlusIcon as se, ListPlusIcon as te } from "lucide-react";
import { useTransition as re, useState as f, useCallback as ne } from "react";
import { a as ie } from "./state-CsuHT8ZO.js";
import { B as A } from "./index-Do_BBSIs.js";
import { R as v, T as C, C as O, S as ae } from "./index-Dvh1BL_e.js";
function oe(t, s) {
  return s;
}
const le = N(
  /* GraphQL */
  `
  query ServersQuery($input: JSON!, $type: SchemaType!) {
    schema(input: $input, type: $type) {
      url
      servers {
        url
      }
    }
  }
`
), w = ({ url: t }) => {
  const [s, n] = f(!1);
  return /* @__PURE__ */ e.jsx(
    A,
    {
      onClick: () => {
        navigator.clipboard.writeText(t).then(() => {
          n(!0), setTimeout(() => n(!1), 2e3);
        });
      },
      variant: "ghost",
      size: "icon",
      children: s ? /* @__PURE__ */ e.jsx(K, { className: "text-green-600", size: 14 }) : /* @__PURE__ */ e.jsx(U, { size: 14, strokeWidth: 1.3 })
    }
  );
}, ce = { suspense: !0 }, de = () => {
  const [t] = P({
    query: le,
    variables: T(),
    context: ce
  }), [, s] = re(), { selectedServer: n, setSelectedServer: i } = ie();
  if (!t.data) return null;
  const { servers: r } = t.data.schema;
  return r.length === 1 ? /* @__PURE__ */ e.jsxs("div", { className: "flex items-center gap-2", children: [
    /* @__PURE__ */ e.jsx("span", { className: "font-medium text-sm", children: "Endpoint:" }),
    /* @__PURE__ */ e.jsx(y, { className: "text-xs px-2 py-1.5", selectOnClick: !0, children: r[0].url }),
    /* @__PURE__ */ e.jsx(w, { url: r[0].url })
  ] }) : /* @__PURE__ */ e.jsxs("div", { className: "flex flex-wrap items-center gap-2", children: [
    /* @__PURE__ */ e.jsxs("span", { className: "font-medium text-sm", children: [
      r.length > 1 ? "Endpoints" : "Endpoint",
      ":"
    ] }),
    /* @__PURE__ */ e.jsx(
      z,
      {
        className: "font-mono text-xs bg-border/50 dark:bg-border/70 py-1.5 max-w-[450px] truncate",
        onChange: (a) => s(() => {
          i(a.target.value);
        }),
        value: n ?? t.data.schema.url,
        showChevrons: r.length > 1,
        options: r.map((a) => ({
          value: a.url,
          label: a.url
        }))
      }
    ),
    /* @__PURE__ */ e.jsx(w, { url: n ?? t.data.schema.url })
  ] });
}, k = (t, s) => t.reduce(
  (n, i) => {
    const r = s(i);
    return n[r] || (n[r] = []), n[r].push(i), n;
  },
  {}
), pe = (t, s) => t ? s(t) : void 0, q = (t) => t.schema != null && typeof t.schema == "object" ? t.schema : {
  type: "string"
}, me = ({
  parameter: t,
  group: s,
  id: n
}) => /* @__PURE__ */ e.jsxs("li", { className: "p-4 bg-border/20 text-sm flex flex-col gap-1", children: [
  /* @__PURE__ */ e.jsxs("div", { className: "flex items-center gap-2", children: [
    /* @__PURE__ */ e.jsx("code", { children: s === "path" ? /* @__PURE__ */ e.jsx(
      R,
      {
        name: t.name,
        backgroundOpacity: "15%",
        slug: n + "-" + t.name.toLocaleLowerCase()
      }
    ) : t.name }),
    t.required && /* @__PURE__ */ e.jsx("span", { className: "py-px px-1.5 font-medium bg-primary/75 text-muted rounded-lg", children: "required" }),
    q(t).type && /* @__PURE__ */ e.jsx("span", { className: "text-muted-foreground", children: q(t).type })
  ] }),
  t.description && /* @__PURE__ */ e.jsx(
    m,
    {
      content: t.description,
      className: "text-sm prose-p:my-1 prose-code:whitespace-pre-line"
    }
  )
] }), xe = ({
  group: t,
  parameters: s,
  id: n
}) => /* @__PURE__ */ e.jsxs(e.Fragment, { children: [
  /* @__PURE__ */ e.jsx(h, { level: 3, id: `${n}/${t}-parameters`, className: "capitalize", children: t === "header" ? "Headers" : `${t} Parameters` }),
  /* @__PURE__ */ e.jsx(p, { children: /* @__PURE__ */ e.jsx("ul", { className: "list-none m-0 px-0 divide-y ", children: s.sort((i, r) => i.required === r.required ? 0 : i.required ? -1 : 1).map((i) => /* @__PURE__ */ e.jsx(
    me,
    {
      parameter: i,
      id: n,
      group: t
    },
    `${i.name}-${i.in}`
  )) }) })
] }), ue = (t) => Object.entries(t), je = {
  AND: /* @__PURE__ */ e.jsx(Z, { size: 16, className: "fill-card" }),
  OR: /* @__PURE__ */ e.jsx(X, { size: 16, className: "fill-card" }),
  ONE: /* @__PURE__ */ e.jsx(Y, { size: 14, className: "fill-card" })
}, he = {
  AND: "text-green-500 dark:text-green-300/60",
  OR: "text-blue-400 dark:text-blue-500",
  ONE: "text-purple-500 dark:text-purple-300/60"
}, ge = ({
  type: t,
  isOpen: s,
  className: n
}) => /* @__PURE__ */ e.jsx(
  "div",
  {
    className: x(
      he[t],
      "relative text-sm flex py-2",
      "before:border-l before:absolute before:-top-2 before:-bottom-2 before:border-border before:border-dashed before:content-['']",
      n
    ),
    children: /* @__PURE__ */ e.jsxs("div", { className: "-translate-x-[7px] flex gap-1 items-center", children: [
      je[t],
      /* @__PURE__ */ e.jsx(
        "div",
        {
          className: x(
            "translate-y-px mx-px opacity-0 group-hover:opacity-100 transition",
            !s && "-rotate-90"
          ),
          children: /* @__PURE__ */ e.jsx(W, { size: 16 })
        }
      )
    ] })
  }
), fe = (t) => {
  const [s, n] = f(!0);
  return /* @__PURE__ */ e.jsxs(
    v,
    {
      open: s,
      onOpenChange: () => n((i) => !i),
      className: "group",
      children: [
        /* @__PURE__ */ e.jsx(C, { children: /* @__PURE__ */ e.jsx(ge, { type: t.type, isOpen: s }) }),
        !s && /* @__PURE__ */ e.jsx("div", { className: "wavy-line bg-border translate-y-1" }),
        /* @__PURE__ */ e.jsx(O, { children: /* @__PURE__ */ e.jsx(g, { schema: t.schema, level: t.level + 1 }) })
      ]
    }
  );
}, ye = {
  AND: "All of",
  OR: "Any of",
  ONE: "One of"
}, be = ({
  schemas: t,
  type: s,
  isOpen: n,
  level: i,
  toggleOpen: r
}) => /* @__PURE__ */ e.jsx(v, { open: n, onOpenChange: r, asChild: !0, children: /* @__PURE__ */ e.jsxs(p, { className: "px-6", children: [
  /* @__PURE__ */ e.jsxs(C, { className: "flex gap-2 items-center py-2 w-full text-sm text-muted-foreground -translate-x-1.5", children: [
    n ? /* @__PURE__ */ e.jsx(ee, { size: 14 }) : /* @__PURE__ */ e.jsx(se, { size: 14 }),
    /* @__PURE__ */ e.jsx("span", { children: ye[s] })
  ] }),
  /* @__PURE__ */ e.jsx(O, { className: "pb-4", children: t.map((a, l) => (
    // eslint-disable-next-line react/no-array-index-key
    /* @__PURE__ */ e.jsx(
      fe,
      {
        type: s,
        schema: a,
        level: i
      },
      l
    )
  )) })
] }) }), Ne = (t) => t.type === "object" || t.type === "array" && typeof t.items == "object" && (!t.items.type || t.items.type === "object"), b = (t) => !!(t.oneOf ?? t.allOf ?? t.anyOf), ve = {
  allOf: "AND",
  anyOf: "OR",
  oneOf: "ONE"
}, $ = ({
  schema: t,
  level: s
}) => {
  const [n, i] = f(!0), r = ne(() => i((a) => !a), []);
  for (const [a, l] of ue(ve))
    if (t[a])
      return /* @__PURE__ */ e.jsx(
        be,
        {
          schemas: t[a],
          type: l,
          isOpen: n,
          toggleOpen: r,
          level: s
        }
      );
}, Ce = ({
  name: t,
  schema: s,
  group: n,
  level: i,
  defaultOpen: r = !1,
  showCollapseButton: a = !0
}) => {
  const [l, j] = f(r);
  return /* @__PURE__ */ e.jsx("li", { className: "p-4 bg-border/20 hover:bg-border/30", children: /* @__PURE__ */ e.jsxs("div", { className: "flex flex-col gap-1 justify-between text-sm", children: [
    /* @__PURE__ */ e.jsxs("div", { className: "flex gap-2 items-center", children: [
      /* @__PURE__ */ e.jsx("code", { children: t }),
      /* @__PURE__ */ e.jsx("span", { className: "text-muted-foreground", children: s.type === "array" && s.items.type ? /* @__PURE__ */ e.jsxs("span", { children: [
        s.items.type,
        "[]"
      ] }) : Array.isArray(s.type) ? /* @__PURE__ */ e.jsx("span", { children: s.type.join(" | ") }) : /* @__PURE__ */ e.jsx("span", { children: s.type }) }),
      n === "optional" && /* @__PURE__ */ e.jsx("span", { className: "py-px px-1.5 font-medium border rounded-lg", children: "optional" })
    ] }),
    s.description && /* @__PURE__ */ e.jsx(
      m,
      {
        className: x(u, "text-sm leading-normal line-clamp-4"),
        content: s.description
      }
    ),
    (b(s) || Ne(s)) && /* @__PURE__ */ e.jsxs(
      v,
      {
        defaultOpen: r,
        open: l,
        onOpenChange: () => j(!l),
        children: [
          a && /* @__PURE__ */ e.jsx(C, { asChild: !0, children: /* @__PURE__ */ e.jsxs(
            A,
            {
              variant: "outline",
              size: "sm",
              className: "mt-2 flex gap-1.5",
              children: [
                /* @__PURE__ */ e.jsx(te, { size: 18 }),
                l ? "Hide nested properties" : "Show nested properties"
              ]
            }
          ) }),
          /* @__PURE__ */ e.jsx(O, { children: /* @__PURE__ */ e.jsx("div", { className: "mt-2", children: b(s) ? /* @__PURE__ */ e.jsx($, { schema: s, level: i + 1 }) : s.type === "object" ? /* @__PURE__ */ e.jsx(g, { schema: s, level: i + 1 }) : s.type === "array" && typeof s.items == "object" && /* @__PURE__ */ e.jsx(g, { schema: s.items, level: i + 1 }) }) })
        ]
      }
    )
  ] }) });
}, g = ({
  schema: t,
  level: s = 0,
  defaultOpen: n = !1
}) => {
  if (!t || Object.keys(t).length === 0)
    return /* @__PURE__ */ e.jsx(p, { className: "p-4", children: /* @__PURE__ */ e.jsx("span", { className: "text-sm text-muted-foreground italic", children: "No response specified" }) });
  const i = (r, a) => {
    if (b(r))
      return /* @__PURE__ */ e.jsx($, { schema: r, level: a });
    if (r.type === "array" && r.items) {
      const l = r.items;
      return typeof l.type == "string" && ["string", "number", "boolean", "integer"].includes(l.type) ? /* @__PURE__ */ e.jsxs(p, { className: "p-4", children: [
        /* @__PURE__ */ e.jsxs("span", { className: "text-sm text-muted-foreground", children: [
          l.type,
          "[]"
        ] }),
        r.description && /* @__PURE__ */ e.jsx(
          m,
          {
            className: x(
              u,
              "text-sm leading-normal line-clamp-4"
            ),
            content: r.description
          }
        )
      ] }) : l.type === "object" ? /* @__PURE__ */ e.jsxs(p, { className: "flex flex-col gap-2 bg-border/30 p-4", children: [
        /* @__PURE__ */ e.jsx("span", { className: "text-sm text-muted-foreground", children: "object[]" }),
        i(l, a + 1)
      ] }) : i(l, a + 1);
    }
    if (r.type === "object" && !r.properties)
      return /* @__PURE__ */ e.jsxs(p, { className: "p-4 flex gap-2 items-center", children: [
        "name" in r && /* @__PURE__ */ e.jsx(e.Fragment, { children: r.name }),
        /* @__PURE__ */ e.jsx("span", { className: "text-sm text-muted-foreground", children: "object" }),
        r.description && /* @__PURE__ */ e.jsx(
          m,
          {
            className: x(
              u,
              "text-sm leading-normal line-clamp-4"
            ),
            content: r.description
          }
        )
      ] });
    if (r.properties) {
      const l = k(
        Object.entries(r.properties),
        ([d, o]) => {
          var c;
          return o.deprecated ? "deprecated" : (c = r.required) != null && c.includes(d) ? "required" : "optional";
        }
      ), j = ["required", "optional", "deprecated"];
      return /* @__PURE__ */ e.jsx(p, { className: "divide-y overflow-hidden", children: j.map(
        (d) => l[d] && /* @__PURE__ */ e.jsx("ul", { className: "divide-y", children: l[d].map(([o, c]) => /* @__PURE__ */ e.jsx(
          Ce,
          {
            name: o,
            schema: c,
            group: d,
            level: a,
            defaultOpen: n
          },
          o
        )) }, d)
      ) });
    }
    return typeof r.type == "string" && ["string", "number", "boolean", "integer", "null"].includes(r.type) ? /* @__PURE__ */ e.jsxs(p, { className: "p-4", children: [
      /* @__PURE__ */ e.jsx("span", { className: "text-sm text-muted-foreground", children: r.type }),
      r.description && /* @__PURE__ */ e.jsx(
        m,
        {
          className: x(
            u,
            "text-sm leading-normal line-clamp-4"
          ),
          content: r.description
        }
      )
    ] }) : r.additionalProperties ? /* @__PURE__ */ e.jsxs(p, { className: "my-2", children: [
      /* @__PURE__ */ e.jsx(E, { children: /* @__PURE__ */ e.jsx(F, { children: "Additional Properties:" }) }),
      /* @__PURE__ */ e.jsx(B, { children: i(
        r.additionalProperties,
        a + 1
      ) })
    ] }) : null;
  };
  return i(t, s);
}, Oe = ["path", "query", "header", "cookie"], Se = ({
  operationFragment: t
}) => {
  var l, j, d;
  const s = oe(Ie, t), n = k(
    s.parameters ?? [],
    (o) => o.in
  ), i = s.responses.at(0), [r, a] = f(i == null ? void 0 : i.statusCode);
  return /* @__PURE__ */ e.jsxs(
    "div",
    {
      className: "grid grid-cols-1 lg:grid-cols-[4fr_3fr] gap-8 items-start border-b-2 mb-16 pb-16",
      children: [
        /* @__PURE__ */ e.jsxs("div", { className: "flex flex-col gap-4", children: [
          /* @__PURE__ */ e.jsx(h, { level: 2, id: s.slug, registerSidebarAnchor: !0, children: s.summary }),
          s.description && /* @__PURE__ */ e.jsx(
            m,
            {
              className: `${u} max-w-full prose-img:max-w-prose`,
              content: s.description
            }
          ),
          s.parameters && s.parameters.length > 0 && /* @__PURE__ */ e.jsx(e.Fragment, { children: Oe.flatMap(
            (o) => {
              var c;
              return (c = n[o]) != null && c.length ? /* @__PURE__ */ e.jsx(
                xe,
                {
                  id: s.slug,
                  parameters: n[o],
                  group: o
                },
                o
              ) : [];
            }
          ) }),
          pe((d = (j = (l = s.requestBody) == null ? void 0 : l.content) == null ? void 0 : j.at(0)) == null ? void 0 : d.schema, (o) => /* @__PURE__ */ e.jsxs("div", { className: "mt-4 flex flex-col gap-4", children: [
            /* @__PURE__ */ e.jsx(
              h,
              {
                level: 3,
                className: "capitalize",
                id: `${s.slug}/request-body`,
                registerSidebarAnchor: !0,
                children: "Request Body"
              }
            ),
            /* @__PURE__ */ e.jsx(g, { schema: o })
          ] })),
          s.responses.length > 0 && /* @__PURE__ */ e.jsxs(e.Fragment, { children: [
            /* @__PURE__ */ e.jsx(
              h,
              {
                level: 3,
                className: "capitalize mt-8 pt-8 border-t",
                id: `${s.slug}/responses`,
                registerSidebarAnchor: !0,
                children: "Responses"
              }
            ),
            /* @__PURE__ */ e.jsxs(
              D,
              {
                onValueChange: (o) => a(o),
                value: r,
                children: [
                  s.responses.length > 1 && /* @__PURE__ */ e.jsx(H, { children: s.responses.map((o) => /* @__PURE__ */ e.jsx(
                    M,
                    {
                      value: o.statusCode,
                      title: o.description ?? void 0,
                      children: o.statusCode
                    },
                    o.statusCode
                  )) }),
                  /* @__PURE__ */ e.jsx("ul", { className: "list-none m-0 px-0", children: s.responses.map((o) => {
                    var c, S;
                    return /* @__PURE__ */ e.jsx(
                      G,
                      {
                        value: o.statusCode,
                        children: /* @__PURE__ */ e.jsx(
                          g,
                          {
                            schema: (S = (c = o.content) == null ? void 0 : c.find((L) => L.schema)) == null ? void 0 : S.schema
                          }
                        )
                      },
                      o.statusCode
                    );
                  }) })
                ]
              }
            )
          ] })
        ] }),
        /* @__PURE__ */ e.jsx(
          Q,
          {
            selectedResponse: r,
            onSelectResponse: a,
            operation: s
          }
        )
      ]
    },
    s.operationId
  );
}, Ie = N(
  /* GraphQL */
  `
  fragment OperationsFragment on OperationItem {
    slug
    summary
    method
    description
    operationId
    contentTypes
    path
    parameters {
      name
      in
      description
      required
      schema
      style
      examples {
        name
        description
        externalValue
        value
        summary
      }
    }
    requestBody {
      content {
        mediaType
        encoding {
          name
        }
        schema
      }
      description
      required
    }
    responses {
      statusCode
      links
      description
      content {
        mediaType
        encoding {
          name
        }
        schema
      }
    }
  }
`
), we = N(
  /* GraphQL */
  `
  query AllOperations($input: JSON!, $type: SchemaType!) {
    schema(input: $input, type: $type) {
      description
      title
      url
      version
      tags {
        name
        description
        operations {
          slug
          ...OperationsFragment
        }
      }
    }
  }
`
), qe = { suspense: !0 }, De = () => {
  var r;
  const { type: t, input: s } = T(), [n] = P({
    query: we,
    variables: { type: t, input: s },
    context: qe
  }), i = (r = n.error) == null ? void 0 : r.graphQLErrors.at(0);
  return i ? /* @__PURE__ */ e.jsx(
    _,
    {
      category: "Error",
      title: "Schema cannot be displayed",
      message: /* @__PURE__ */ e.jsxs(e.Fragment, { children: [
        /* @__PURE__ */ e.jsxs(V, { className: "mb-4", children: [
          "Check your configuration value ",
          /* @__PURE__ */ e.jsx(y, { children: "apis.type" }),
          " ",
          "and ",
          /* @__PURE__ */ e.jsx(y, { children: "apis.input" }),
          " in the Zudoku config."
        ] }),
        "An error occurred while trying to fetch the API reference:",
        /* @__PURE__ */ e.jsx(J, { code: i.toString(), language: "plain" })
      ] })
    }
  ) : n.data ? /* @__PURE__ */ e.jsxs("div", { className: "pt-[--padding-content-top]", children: [
    /* @__PURE__ */ e.jsxs(
      "div",
      {
        className: x(u, "mb-16 max-w-full prose-img:max-w-prose"),
        children: [
          /* @__PURE__ */ e.jsx(I, { children: "Overview" }),
          /* @__PURE__ */ e.jsx(h, { level: 1, id: "description", registerSidebarAnchor: !0, children: n.data.schema.title }),
          /* @__PURE__ */ e.jsx(m, { content: n.data.schema.description ?? "" })
        ]
      }
    ),
    /* @__PURE__ */ e.jsx("hr", {}),
    /* @__PURE__ */ e.jsx("div", { className: "my-4 flex justify-end", children: /* @__PURE__ */ e.jsx(de, {}) }),
    n.data.schema.tags.filter((a) => a.operations.length > 0).map((a) => /* @__PURE__ */ e.jsxs("div", { children: [
      a.name && /* @__PURE__ */ e.jsx(I, { children: a.name }),
      a.description && /* @__PURE__ */ e.jsx(
        m,
        {
          className: `${u} max-w-full prose-img:max-w-prose w-full mt-2 mb-12`,
          content: a.description
        }
      ),
      /* @__PURE__ */ e.jsx("div", { className: "operation mb-12", children: /* @__PURE__ */ e.jsx(ae, { children: a.operations.map((l) => /* @__PURE__ */ e.jsx(
        Se,
        {
          operationFragment: l
        },
        l.slug
      )) }) })
    ] }, a.name))
  ] }) : null;
};
export {
  De as OperationList,
  Ie as OperationsFragment
};
