// zudoku/lib/SlotletProvider-RvaeLR6z.js
import { j as a } from "./jsx-runtime-B6kdoens.js";
import { UnlinkIcon as ce } from "lucide-react";
import { c as ue, L as fe, d as le } from "./index-BRCiYFaL.js";
import { C as me } from "./CategoryHeading-3Qtp2yZ8.js";
import { D as de } from "./DeveloperHint-BE9DzNhv.js";
import { P as pe, H as Ee } from "./Markdown-CWI6lU11.js";
import { b as ve } from "./router-Oe6YmY6B.js";
import j, { useContext as ye } from "react";
import { u as xe } from "./useExposedProps-Csw8oAlt.js";
const Se = () => {
  const n = ue();
  return /* @__PURE__ */ a.jsxs("div", { className: pe + " h-full pt-[--padding-content-top]", children: [
    /* @__PURE__ */ a.jsx(me, { children: "404" }),
    /* @__PURE__ */ a.jsxs(Ee, { level: 1, className: "flex gap-3.5 items-center", children: [
      "Page not found",
      /* @__PURE__ */ a.jsx(ce, { size: 24 })
    ] }),
    /* @__PURE__ */ a.jsxs(de, { children: [
      "Start by adding a file at",
      " ",
      /* @__PURE__ */ a.jsxs("code", { children: [
        "{PROJECT_ROOT}",
        "/",
        n["*"],
        ".mdx"
      ] }),
      " ",
      "and add some content to make this error go away."
    ] }),
    /* @__PURE__ */ a.jsx("p", { children: "It seems that the page you are looking for does not exist or may have been moved. Please check the URL for any typos or use the navigation menu to find the correct page." }),
    /* @__PURE__ */ a.jsx(fe, { to: "/", children: "Go back home" })
  ] });
};
function be({ error: n }) {
  const i = (n == null ? void 0 : n.message) ?? "Something went wrong", s = n == null ? void 0 : n.stack;
  return /* @__PURE__ */ a.jsx("div", { className: "flex h-screen max-h-screen min-h-full items-center justify-center bg-primary-background px-4 py-16 lg:px-8", children: /* @__PURE__ */ a.jsxs("div", { className: "mx-auto max-w-[85%] sm:max-w-[50%]", children: [
    /* @__PURE__ */ a.jsx("h1", { className: "text-4xl font-bold tracking-tight text-h1-text sm:text-5xl", children: "Something went wrong" }),
    /* @__PURE__ */ a.jsx("p", { className: "mt-5 text-h1-text", children: i }),
    s ? /* @__PURE__ */ a.jsx("pre", { className: "mt-5 max-h-[400px] w-full overflow-scroll rounded-md border border-input-border bg-input-background p-3 text-property-name-text text-red-700", children: s }) : null
  ] }) });
}
function Le() {
  const n = le();
  return ve(n) && n.status === 404 ? /* @__PURE__ */ a.jsx(Se, {}) : /* @__PURE__ */ a.jsx(be, { error: n });
}
var T = { exports: {} }, t = {};
/**
 * @license React
 * react-is.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var A;
function Re() {
  if (A) return t;
  A = 1;
  var n = Symbol.for("react.element"), i = Symbol.for("react.portal"), s = Symbol.for("react.fragment"), f = Symbol.for("react.strict_mode"), l = Symbol.for("react.profiler"), m = Symbol.for("react.provider"), d = Symbol.for("react.context"), _ = Symbol.for("react.server_context"), p = Symbol.for("react.forward_ref"), E = Symbol.for("react.suspense"), v = Symbol.for("react.suspense_list"), y = Symbol.for("react.memo"), x = Symbol.for("react.lazy"), C = Symbol.for("react.offscreen"), b;
  b = Symbol.for("react.module.reference");
  function c(e) {
    if (typeof e == "object" && e !== null) {
      var S = e.$$typeof;
      switch (S) {
        case n:
          switch (e = e.type, e) {
            case s:
            case l:
            case f:
            case E:
            case v:
              return e;
            default:
              switch (e = e && e.$$typeof, e) {
                case _:
                case d:
                case p:
                case x:
                case y:
                case m:
                  return e;
                default:
                  return S;
              }
          }
        case i:
          return S;
      }
    }
  }
  return t.ContextConsumer = d, t.ContextProvider = m, t.Element = n, t.ForwardRef = p, t.Fragment = s, t.Lazy = x, t.Memo = y, t.Portal = i, t.Profiler = l, t.StrictMode = f, t.Suspense = E, t.SuspenseList = v, t.isAsyncMode = function() {
    return !1;
  }, t.isConcurrentMode = function() {
    return !1;
  }, t.isContextConsumer = function(e) {
    return c(e) === d;
  }, t.isContextProvider = function(e) {
    return c(e) === m;
  }, t.isElement = function(e) {
    return typeof e == "object" && e !== null && e.$$typeof === n;
  }, t.isForwardRef = function(e) {
    return c(e) === p;
  }, t.isFragment = function(e) {
    return c(e) === s;
  }, t.isLazy = function(e) {
    return c(e) === x;
  }, t.isMemo = function(e) {
    return c(e) === y;
  }, t.isPortal = function(e) {
    return c(e) === i;
  }, t.isProfiler = function(e) {
    return c(e) === l;
  }, t.isStrictMode = function(e) {
    return c(e) === f;
  }, t.isSuspense = function(e) {
    return c(e) === E;
  }, t.isSuspenseList = function(e) {
    return c(e) === v;
  }, t.isValidElementType = function(e) {
    return typeof e == "string" || typeof e == "function" || e === s || e === l || e === f || e === E || e === v || e === C || typeof e == "object" && e !== null && (e.$$typeof === x || e.$$typeof === y || e.$$typeof === m || e.$$typeof === d || e.$$typeof === p || e.$$typeof === b || e.getModuleId !== void 0);
  }, t.typeOf = c, t;
}
var o = {};
/**
 * @license React
 * react-is.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var w;
function _e() {
  return w || (w = 1, process.env.NODE_ENV !== "production" && function() {
    var n = Symbol.for("react.element"), i = Symbol.for("react.portal"), s = Symbol.for("react.fragment"), f = Symbol.for("react.strict_mode"), l = Symbol.for("react.profiler"), m = Symbol.for("react.provider"), d = Symbol.for("react.context"), _ = Symbol.for("react.server_context"), p = Symbol.for("react.forward_ref"), E = Symbol.for("react.suspense"), v = Symbol.for("react.suspense_list"), y = Symbol.for("react.memo"), x = Symbol.for("react.lazy"), C = Symbol.for("react.offscreen"), b = !1, c = !1, e = !1, S = !1, I = !1, h;
    h = Symbol.for("react.module.reference");
    function F(r) {
      return !!(typeof r == "string" || typeof r == "function" || r === s || r === l || I || r === f || r === E || r === v || S || r === C || b || c || e || typeof r == "object" && r !== null && (r.$$typeof === x || r.$$typeof === y || r.$$typeof === m || r.$$typeof === d || r.$$typeof === p || // This needs to include all possible module reference object
      // types supported by any Flight configuration anywhere since
      // we don't know which Flight build this will end up being used
      // with.
      r.$$typeof === h || r.getModuleId !== void 0));
    }
    function u(r) {
      if (typeof r == "object" && r !== null) {
        var P = r.$$typeof;
        switch (P) {
          case n:
            var R = r.type;
            switch (R) {
              case s:
              case l:
              case f:
              case E:
              case v:
                return R;
              default:
                var $ = R && R.$$typeof;
                switch ($) {
                  case _:
                  case d:
                  case p:
                  case x:
                  case y:
                  case m:
                    return $;
                  default:
                    return P;
                }
            }
          case i:
            return P;
        }
      }
    }
    var N = d, O = m, Y = n, k = p, D = s, z = x, V = y, q = i, H = l, U = f, W = E, G = v, g = !1, M = !1;
    function X(r) {
      return g || (g = !0, console.warn("The ReactIs.isAsyncMode() alias has been deprecated, and will be removed in React 18+.")), !1;
    }
    function J(r) {
      return M || (M = !0, console.warn("The ReactIs.isConcurrentMode() alias has been deprecated, and will be removed in React 18+.")), !1;
    }
    function Z(r) {
      return u(r) === d;
    }
    function B(r) {
      return u(r) === m;
    }
    function K(r) {
      return typeof r == "object" && r !== null && r.$$typeof === n;
    }
    function Q(r) {
      return u(r) === p;
    }
    function ee(r) {
      return u(r) === s;
    }
    function re(r) {
      return u(r) === x;
    }
    function te(r) {
      return u(r) === y;
    }
    function oe(r) {
      return u(r) === i;
    }
    function ne(r) {
      return u(r) === l;
    }
    function se(r) {
      return u(r) === f;
    }
    function ae(r) {
      return u(r) === E;
    }
    function ie(r) {
      return u(r) === v;
    }
    o.ContextConsumer = N, o.ContextProvider = O, o.Element = Y, o.ForwardRef = k, o.Fragment = D, o.Lazy = z, o.Memo = V, o.Portal = q, o.Profiler = H, o.StrictMode = U, o.Suspense = W, o.SuspenseList = G, o.isAsyncMode = X, o.isConcurrentMode = J, o.isContextConsumer = Z, o.isContextProvider = B, o.isElement = K, o.isForwardRef = Q, o.isFragment = ee, o.isLazy = re, o.isMemo = te, o.isPortal = oe, o.isProfiler = ne, o.isStrictMode = se, o.isSuspense = ae, o.isSuspenseList = ie, o.isValidElementType = F, o.typeOf = u;
  }()), o;
}
process.env.NODE_ENV === "production" ? T.exports = Re() : T.exports = _e();
var Ce = T.exports;
const L = j.createContext({}), Ie = ({
  slotlets: n,
  children: i
}) => /* @__PURE__ */ a.jsx(L.Provider, { value: n, children: i }), Fe = ({ name: n }) => {
  const i = ye(L), s = i == null ? void 0 : i[n], f = xe();
  return Ce.isValidElementType(s) ? j.createElement(s, f) : s;
};
export {
  be as E,
  Le as R,
  Fe as S,
  Ie as a
};
