// zudoku/lib/urql-core-KJnLL26g.js
var M = {
  NAME: "Name",
  DOCUMENT: "Document",
  OPERATION_DEFINITION: "OperationDefinition",
  VARIABLE_DEFINITION: "VariableDefinition",
  SELECTION_SET: "SelectionSet",
  FIELD: "Field",
  ARGUMENT: "Argument",
  FRAGMENT_SPREAD: "FragmentSpread",
  INLINE_FRAGMENT: "InlineFragment",
  FRAGMENT_DEFINITION: "FragmentDefinition",
  VARIABLE: "Variable",
  INT: "IntValue",
  FLOAT: "FloatValue",
  STRING: "StringValue",
  BOOLEAN: "BooleanValue",
  NULL: "NullValue",
  ENUM: "EnumValue",
  LIST: "ListValue",
  OBJECT: "ObjectValue",
  OBJECT_FIELD: "ObjectField",
  DIRECTIVE: "Directive",
  NAMED_TYPE: "NamedType",
  LIST_TYPE: "ListType",
  NON_NULL_TYPE: "NonNullType"
};
class fe extends Error {
  constructor(r, t, n, i, a, o, l) {
    super(r), this.name = "GraphQLError", this.message = r, a && (this.path = a), t && (this.nodes = Array.isArray(t) ? t : [t]), n && (this.source = n), i && (this.positions = i), o && (this.originalError = o);
    var f = l;
    if (!f && o) {
      var u = o.extensions;
      u && typeof u == "object" && (f = u);
    }
    this.extensions = f || {};
  }
  toJSON() {
    return {
      ...this,
      message: this.message
    };
  }
  toString() {
    return this.message;
  }
  get [Symbol.toStringTag]() {
    return "GraphQLError";
  }
}
var g, d;
function N(e) {
  return new fe(`Syntax Error: Unexpected token at ${d} in ${e}`);
}
function _(e) {
  if (e.lastIndex = d, e.test(g))
    return g.slice(d, d = e.lastIndex);
}
var Y = / +(?=[^\s])/y;
function Qe(e) {
  for (var r = e.split(`
`), t = "", n = 0, i = 0, a = r.length - 1, o = 0; o < r.length; o++)
    Y.lastIndex = 0, Y.test(r[o]) && (o && (!n || Y.lastIndex < n) && (n = Y.lastIndex), i = i || o, a = o);
  for (var l = i; l <= a; l++)
    l !== i && (t += `
`), t += r[l].slice(n).replace(/\\"""/g, '"""');
  return t;
}
function h() {
  for (var e = 0 | g.charCodeAt(d++); e === 9 || e === 10 || e === 13 || e === 32 || e === 35 || e === 44 || e === 65279; e = 0 | g.charCodeAt(d++))
    if (e === 35)
      for (; (e = g.charCodeAt(d++)) !== 10 && e !== 13; )
        ;
  d--;
}
var O = /[_A-Za-z]\w*/y, se = new RegExp("(?:(null|true|false)|\\$(" + O.source + ')|(-?\\d+)((?:\\.\\d+)?[eE][+-]?\\d+|\\.\\d+)?|("""(?:"""|(?:[\\s\\S]*?[^\\\\])"""))|("(?:"|[^\\r\\n]*?[^\\\\]"))|(' + O.source + "))", "y"), L = function(e) {
  return e[e.Const = 1] = "Const", e[e.Var = 2] = "Var", e[e.Int = 3] = "Int", e[e.Float = 4] = "Float", e[e.BlockString = 5] = "BlockString", e[e.String = 6] = "String", e[e.Enum = 7] = "Enum", e;
}(L || {}), Ue = /\\/;
function K(e) {
  var r, t;
  if (se.lastIndex = d, g.charCodeAt(d) === 91) {
    d++, h();
    for (var n = []; g.charCodeAt(d) !== 93; )
      n.push(K(e));
    return d++, h(), {
      kind: "ListValue",
      values: n
    };
  } else if (g.charCodeAt(d) === 123) {
    d++, h();
    for (var i = []; g.charCodeAt(d) !== 125; ) {
      if ((r = _(O)) == null || (h(), g.charCodeAt(d++) !== 58))
        throw N("ObjectField");
      h(), i.push({
        kind: "ObjectField",
        name: {
          kind: "Name",
          value: r
        },
        value: K(e)
      });
    }
    return d++, h(), {
      kind: "ObjectValue",
      fields: i
    };
  } else if ((t = se.exec(g)) != null) {
    if (d = se.lastIndex, h(), (r = t[L.Const]) != null)
      return r === "null" ? {
        kind: "NullValue"
      } : {
        kind: "BooleanValue",
        value: r === "true"
      };
    if ((r = t[L.Var]) != null) {
      if (e)
        throw N("Variable");
      return {
        kind: "Variable",
        name: {
          kind: "Name",
          value: r
        }
      };
    } else if ((r = t[L.Int]) != null) {
      var a;
      return (a = t[L.Float]) != null ? {
        kind: "FloatValue",
        value: r + a
      } : {
        kind: "IntValue",
        value: r
      };
    } else {
      if ((r = t[L.BlockString]) != null)
        return {
          kind: "StringValue",
          value: Qe(r.slice(3, -3)),
          block: !0
        };
      if ((r = t[L.String]) != null)
        return {
          kind: "StringValue",
          value: Ue.test(r) ? JSON.parse(r) : r.slice(1, -1),
          block: !1
        };
      if ((r = t[L.Enum]) != null)
        return {
          kind: "EnumValue",
          value: r
        };
    }
  }
  throw N("Value");
}
function Le(e) {
  if (g.charCodeAt(d) === 40) {
    var r = [];
    d++, h();
    var t;
    do {
      if ((t = _(O)) == null || (h(), g.charCodeAt(d++) !== 58))
        throw N("Argument");
      h(), r.push({
        kind: "Argument",
        name: {
          kind: "Name",
          value: t
        },
        value: K(e)
      });
    } while (g.charCodeAt(d) !== 41);
    return d++, h(), r;
  }
}
function $(e) {
  if (g.charCodeAt(d) === 64) {
    var r = [], t;
    do {
      if (d++, (t = _(O)) == null)
        throw N("Directive");
      h(), r.push({
        kind: "Directive",
        name: {
          kind: "Name",
          value: t
        },
        arguments: Le(e)
      });
    } while (g.charCodeAt(d) === 64);
    return r;
  }
}
function ze() {
  for (var e, r = 0; g.charCodeAt(d) === 91; )
    r++, d++, h();
  if ((e = _(O)) == null)
    throw N("NamedType");
  h();
  var t = {
    kind: "NamedType",
    name: {
      kind: "Name",
      value: e
    }
  };
  do
    if (g.charCodeAt(d) === 33 && (d++, h(), t = {
      kind: "NonNullType",
      type: t
    }), r) {
      if (g.charCodeAt(d++) !== 93)
        throw N("NamedType");
      h(), t = {
        kind: "ListType",
        type: t
      };
    }
  while (r--);
  return t;
}
var le = new RegExp("(?:(\\.{3})|(" + O.source + "))", "y"), ve = function(e) {
  return e[e.Spread = 1] = "Spread", e[e.Name = 2] = "Name", e;
}(ve || {});
function X() {
  var e = [], r, t;
  do
    if (le.lastIndex = d, (t = le.exec(g)) != null) {
      if (d = le.lastIndex, t[ve.Spread] != null) {
        h();
        var n = _(O);
        if (n != null && n !== "on")
          h(), e.push({
            kind: "FragmentSpread",
            name: {
              kind: "Name",
              value: n
            },
            directives: $(!1)
          });
        else {
          if (h(), n === "on") {
            if ((n = _(O)) == null)
              throw N("NamedType");
            h();
          }
          var i = $(!1);
          if (g.charCodeAt(d++) !== 123)
            throw N("InlineFragment");
          h(), e.push({
            kind: "InlineFragment",
            typeCondition: n ? {
              kind: "NamedType",
              name: {
                kind: "Name",
                value: n
              }
            } : void 0,
            directives: i,
            selectionSet: X()
          });
        }
      } else if ((r = t[ve.Name]) != null) {
        var a = void 0;
        if (h(), g.charCodeAt(d) === 58) {
          if (d++, h(), a = r, (r = _(O)) == null)
            throw N("Field");
          h();
        }
        var o = Le(!1);
        h();
        var l = $(!1), f = void 0;
        g.charCodeAt(d) === 123 && (d++, h(), f = X()), e.push({
          kind: "Field",
          alias: a ? {
            kind: "Name",
            value: a
          } : void 0,
          name: {
            kind: "Name",
            value: r
          },
          arguments: o,
          directives: l,
          selectionSet: f
        });
      }
    } else
      throw N("SelectionSet");
  while (g.charCodeAt(d) !== 125);
  return d++, h(), {
    kind: "SelectionSet",
    selections: e
  };
}
function Ye() {
  var e, r;
  if ((e = _(O)) == null || (h(), _(O) !== "on") || (h(), (r = _(O)) == null))
    throw N("FragmentDefinition");
  h();
  var t = $(!1);
  if (g.charCodeAt(d++) !== 123)
    throw N("FragmentDefinition");
  return h(), {
    kind: "FragmentDefinition",
    name: {
      kind: "Name",
      value: e
    },
    typeCondition: {
      kind: "NamedType",
      name: {
        kind: "Name",
        value: r
      }
    },
    directives: t,
    selectionSet: X()
  };
}
var We = /(?:query|mutation|subscription|fragment)/y;
function He(e) {
  var r, t, n;
  if (e && (h(), r = _(O), t = function() {
    if (h(), g.charCodeAt(d) === 40) {
      var a = [];
      d++, h();
      var o;
      do {
        if (g.charCodeAt(d++) !== 36 || (o = _(O)) == null)
          throw N("Variable");
        if (h(), g.charCodeAt(d++) !== 58)
          throw N("VariableDefinition");
        h();
        var l = ze(), f = void 0;
        g.charCodeAt(d) === 61 && (d++, h(), f = K(!0)), h(), a.push({
          kind: "VariableDefinition",
          variable: {
            kind: "Variable",
            name: {
              kind: "Name",
              value: o
            }
          },
          type: l,
          defaultValue: f,
          directives: $(!0)
        });
      } while (g.charCodeAt(d) !== 41);
      return d++, h(), a;
    }
  }(), n = $(!1)), g.charCodeAt(d) === 123)
    return d++, h(), {
      kind: "OperationDefinition",
      operation: e || "query",
      name: r ? {
        kind: "Name",
        value: r
      } : void 0,
      variableDefinitions: t,
      directives: n,
      selectionSet: X()
    };
}
function Ze(e, r) {
  return g = typeof e.body == "string" ? e.body : e, d = 0, function() {
    var n, i;
    h();
    var a = [];
    do
      if ((n = _(We)) === "fragment")
        h(), a.push(Ye());
      else if ((i = He(n)) != null)
        a.push(i);
      else
        throw N("Document");
    while (d < g.length);
    return {
      kind: "Document",
      definitions: a
    };
  }();
}
function S(e, r, t) {
  for (var n = "", i = 0; i < e.length; i++)
    i && (n += r), n += t(e[i]);
  return n;
}
function Ke(e) {
  return JSON.stringify(e);
}
function Xe(e) {
  return `"""
` + e.replace(/"""/g, '\\"""') + `
"""`;
}
var F = `
`, b = {
  OperationDefinition(e) {
    var r = e.operation;
    return e.name && (r += " " + e.name.value), e.variableDefinitions && e.variableDefinitions.length && (e.name || (r += " "), r += "(" + S(e.variableDefinitions, ", ", b.VariableDefinition) + ")"), e.directives && e.directives.length && (r += " " + S(e.directives, " ", b.Directive)), r !== "query" ? r + " " + b.SelectionSet(e.selectionSet) : b.SelectionSet(e.selectionSet);
  },
  VariableDefinition(e) {
    var r = b.Variable(e.variable) + ": " + P(e.type);
    return e.defaultValue && (r += " = " + P(e.defaultValue)), e.directives && e.directives.length && (r += " " + S(e.directives, " ", b.Directive)), r;
  },
  Field(e) {
    var r = e.alias ? e.alias.value + ": " + e.name.value : e.name.value;
    if (e.arguments && e.arguments.length) {
      var t = S(e.arguments, ", ", b.Argument);
      r.length + t.length + 2 > 80 ? r += "(" + (F += "  ") + S(e.arguments, F, b.Argument) + (F = F.slice(0, -2)) + ")" : r += "(" + t + ")";
    }
    return e.directives && e.directives.length && (r += " " + S(e.directives, " ", b.Directive)), e.selectionSet && (r += " " + b.SelectionSet(e.selectionSet)), r;
  },
  StringValue(e) {
    return e.block ? Xe(e.value).replace(/\n/g, F) : Ke(e.value);
  },
  BooleanValue: (e) => "" + e.value,
  NullValue: (e) => "null",
  IntValue: (e) => e.value,
  FloatValue: (e) => e.value,
  EnumValue: (e) => e.value,
  Name: (e) => e.value,
  Variable: (e) => "$" + e.name.value,
  ListValue: (e) => "[" + S(e.values, ", ", P) + "]",
  ObjectValue: (e) => "{" + S(e.fields, ", ", b.ObjectField) + "}",
  ObjectField: (e) => e.name.value + ": " + P(e.value),
  Document(e) {
    return !e.definitions || !e.definitions.length ? "" : S(e.definitions, `

`, P);
  },
  SelectionSet: (e) => "{" + (F += "  ") + S(e.selections, F, P) + (F = F.slice(0, -2)) + "}",
  Argument: (e) => e.name.value + ": " + P(e.value),
  FragmentSpread(e) {
    var r = "..." + e.name.value;
    return e.directives && e.directives.length && (r += " " + S(e.directives, " ", b.Directive)), r;
  },
  InlineFragment(e) {
    var r = "...";
    return e.typeCondition && (r += " on " + e.typeCondition.name.value), e.directives && e.directives.length && (r += " " + S(e.directives, " ", b.Directive)), r += " " + b.SelectionSet(e.selectionSet);
  },
  FragmentDefinition(e) {
    var r = "fragment " + e.name.value;
    return r += " on " + e.typeCondition.name.value, e.directives && e.directives.length && (r += " " + S(e.directives, " ", b.Directive)), r + " " + b.SelectionSet(e.selectionSet);
  },
  Directive(e) {
    var r = "@" + e.name.value;
    return e.arguments && e.arguments.length && (r += "(" + S(e.arguments, ", ", b.Argument) + ")"), r;
  },
  NamedType: (e) => e.name.value,
  ListType: (e) => "[" + P(e.type) + "]",
  NonNullType: (e) => P(e.type) + "!"
}, P = (e) => b[e.kind](e);
function er(e) {
  return F = `
`, b[e.kind] ? b[e.kind](e) : "";
}
var ge = () => {
}, T = ge;
function C(e) {
  return {
    tag: 0,
    0: e
  };
}
function z(e) {
  return {
    tag: 1,
    0: e
  };
}
var Ne = () => typeof Symbol == "function" && Symbol.asyncIterator || "@@asyncIterator", rr = (e) => e;
function A(e) {
  return (r) => (t) => {
    var n = T;
    r((i) => {
      i === 0 ? t(0) : i.tag === 0 ? (n = i[0], t(i)) : e(i[0]) ? t(i) : n(0);
    });
  };
}
function W(e) {
  return (r) => (t) => r((n) => {
    n === 0 || n.tag === 0 ? t(n) : t(z(e(n[0])));
  });
}
function ee(e) {
  return (r) => (t) => {
    var n = [], i = T, a = !1, o = !1;
    r((l) => {
      o || (l === 0 ? (o = !0, n.length || t(0)) : l.tag === 0 ? i = l[0] : (a = !1, function(u) {
        var p = T;
        u((c) => {
          if (c === 0) {
            if (n.length) {
              var m = n.indexOf(p);
              m > -1 && (n = n.slice()).splice(m, 1), n.length || (o ? t(0) : a || (a = !0, i(0)));
            }
          } else c.tag === 0 ? (n.push(p = c[0]), p(0)) : n.length && (t(c), p(0));
        });
      }(e(l[0])), a || (a = !0, i(0))));
    }), t(C((l) => {
      if (l === 1) {
        o || (o = !0, i(1));
        for (var f = 0, u = n, p = n.length; f < p; f++)
          u[f](1);
        n.length = 0;
      } else {
        !o && !a ? (a = !0, i(0)) : a = !1;
        for (var c = 0, m = n, x = n.length; c < x; c++)
          m[c](0);
      }
    }));
  };
}
function tr(e) {
  return ee(rr)(e);
}
function J(e) {
  return tr(or(e));
}
function je(e) {
  return (r) => (t) => {
    var n = !1;
    r((i) => {
      if (!n) if (i === 0)
        n = !0, t(0), e();
      else if (i.tag === 0) {
        var a = i[0];
        t(C((o) => {
          o === 1 ? (n = !0, a(1), e()) : a(o);
        }));
      } else
        t(i);
    });
  };
}
function ie(e) {
  return (r) => (t) => {
    var n = !1;
    r((i) => {
      if (!n) if (i === 0)
        n = !0, t(0);
      else if (i.tag === 0) {
        var a = i[0];
        t(C((o) => {
          o === 1 && (n = !0), a(o);
        }));
      } else
        e(i[0]), t(i);
    });
  };
}
function be(e) {
  return (r) => (t) => r((n) => {
    n === 0 ? t(0) : n.tag === 0 ? (t(n), e()) : t(n);
  });
}
function Q(e) {
  var r = [], t = T, n = !1;
  return (i) => {
    r.push(i), r.length === 1 && e((a) => {
      if (a === 0) {
        for (var o = 0, l = r, f = r.length; o < f; o++)
          l[o](0);
        r.length = 0;
      } else if (a.tag === 0)
        t = a[0];
      else {
        n = !1;
        for (var u = 0, p = r, c = r.length; u < c; u++)
          p[u](a);
      }
    }), i(C((a) => {
      if (a === 1) {
        var o = r.indexOf(i);
        o > -1 && (r = r.slice()).splice(o, 1), r.length || t(1);
      } else n || (n = !0, t(0));
    }));
  };
}
function we(e) {
  return (r) => (t) => {
    var n = T, i = T, a = !1, o = !1, l = !1, f = !1;
    r((u) => {
      f || (u === 0 ? (f = !0, l || t(0)) : u.tag === 0 ? n = u[0] : (l && (i(1), i = T), a ? a = !1 : (a = !0, n(0)), function(c) {
        l = !0, c((m) => {
          l && (m === 0 ? (l = !1, f ? t(0) : a || (a = !0, n(0))) : m.tag === 0 ? (o = !1, (i = m[0])(0)) : (t(m), o ? o = !1 : i(0)));
        });
      }(e(u[0]))));
    }), t(C((u) => {
      u === 1 ? (f || (f = !0, n(1)), l && (l = !1, i(1))) : (!f && !a && (a = !0, n(0)), l && !o && (o = !0, i(0)));
    }));
  };
}
function Re(e) {
  return (r) => (t) => {
    var n = T, i = !1, a = 0;
    r((o) => {
      i || (o === 0 ? (i = !0, t(0)) : o.tag === 0 ? n = o[0] : a++ < e ? (t(o), !i && a >= e && (i = !0, t(0), n(1))) : t(o));
    }), t(C((o) => {
      o === 1 && !i ? (i = !0, n(1)) : o === 0 && !i && a < e && n(0);
    }));
  };
}
function Me(e) {
  return (r) => (t) => {
    var n = T, i = T, a = !1;
    r((o) => {
      a || (o === 0 ? (a = !0, i(1), t(0)) : o.tag === 0 ? (n = o[0], e((l) => {
        l === 0 || (l.tag === 0 ? (i = l[0])(0) : (a = !0, i(1), n(1), t(0)));
      })) : t(o));
    }), t(C((o) => {
      o === 1 && !a ? (a = !0, n(1), i(1)) : a || n(0);
    }));
  };
}
function nr(e, r) {
  return (t) => (n) => {
    var i = T, a = !1;
    t((o) => {
      a || (o === 0 ? (a = !0, n(0)) : o.tag === 0 ? (i = o[0], n(o)) : e(o[0]) ? n(o) : (a = !0, r && n(o), n(0), i(1)));
    });
  };
}
function ar(e) {
  return (r) => e()(r);
}
function $e(e) {
  return (r) => {
    var t = e[Ne()] && e[Ne()]() || e, n = !1, i = !1, a = !1, o;
    r(C(async (l) => {
      if (l === 1)
        n = !0, t.return && t.return();
      else if (i)
        a = !0;
      else {
        for (a = i = !0; a && !n; )
          if ((o = await t.next()).done)
            n = !0, t.return && await t.return(), r(0);
          else
            try {
              a = !1, r(z(o.value));
            } catch (f) {
              if (t.throw)
                (n = !!(await t.throw(f)).done) && r(0);
              else
                throw f;
            }
        i = !1;
      }
    }));
  };
}
function ir(e) {
  return e[Symbol.asyncIterator] ? $e(e) : (r) => {
    var t = e[Symbol.iterator](), n = !1, i = !1, a = !1, o;
    r(C((l) => {
      if (l === 1)
        n = !0, t.return && t.return();
      else if (i)
        a = !0;
      else {
        for (a = i = !0; a && !n; )
          if ((o = t.next()).done)
            n = !0, t.return && t.return(), r(0);
          else
            try {
              a = !1, r(z(o.value));
            } catch (f) {
              if (t.throw)
                (n = !!t.throw(f).done) && r(0);
              else
                throw f;
            }
        i = !1;
      }
    }));
  };
}
var or = ir;
function G(e) {
  return (r) => {
    var t = !1;
    r(C((n) => {
      n === 1 ? t = !0 : t || (t = !0, r(z(e)), r(0));
    }));
  };
}
function Be(e) {
  return (r) => {
    var t = !1, n = e({
      next(i) {
        t || r(z(i));
      },
      complete() {
        t || (t = !0, r(0));
      }
    });
    r(C((i) => {
      i === 1 && !t && (t = !0, n());
    }));
  };
}
function Ee() {
  var e, r;
  return {
    source: Q(Be((t) => (e = t.next, r = t.complete, ge))),
    next(t) {
      e && e(t);
    },
    complete() {
      r && r();
    }
  };
}
function Oe(e) {
  return Be((r) => (e.then((t) => {
    Promise.resolve(t).then(() => {
      r.next(t), r.complete();
    });
  }), ge));
}
function re(e) {
  return (r) => {
    var t = T, n = !1;
    return r((i) => {
      i === 0 ? n = !0 : i.tag === 0 ? (t = i[0])(0) : n || (e(i[0]), t(0));
    }), {
      unsubscribe() {
        n || (n = !0, t(1));
      }
    };
  };
}
function sr(e) {
  re((r) => {
  })(e);
}
function lr(e) {
  return new Promise((r) => {
    var t = T, n;
    e((i) => {
      i === 0 ? Promise.resolve(n).then(r) : i.tag === 0 ? (t = i[0])(0) : (n = i[0], t(0));
    });
  });
}
var ur = (e) => e && e.message && (e.extensions || e.name === "GraphQLError") ? e : typeof e == "object" && e.message ? new fe(e.message, e.nodes, e.source, e.positions, e.path, e, e.extensions || {}) : new fe(e);
class ke extends Error {
  constructor(r) {
    var t = (r.graphQLErrors || []).map(ur), n = ((i, a) => {
      var o = "";
      if (i)
        return `[Network] ${i.message}`;
      if (a)
        for (var l of a)
          o && (o += `
`), o += `[GraphQL] ${l.message}`;
      return o;
    })(r.networkError, t);
    super(n), this.name = "CombinedError", this.message = n, this.graphQLErrors = t, this.networkError = r.networkError, this.response = r.response;
  }
  toString() {
    return this.message;
  }
}
var H = (e, r) => {
  for (var t = 0 | (r || 5381), n = 0, i = 0 | e.length; n < i; n++)
    t = (t << 5) + t + e.charCodeAt(n);
  return t;
}, j = /* @__PURE__ */ new Set(), Se = /* @__PURE__ */ new WeakMap(), R = (e, r) => {
  if (e === null || j.has(e))
    return "null";
  if (typeof e != "object")
    return JSON.stringify(e) || "";
  if (e.toJSON)
    return R(e.toJSON(), r);
  if (Array.isArray(e)) {
    var t = "[";
    for (var n of e)
      t.length > 1 && (t += ","), t += R(n, r) || "null";
    return t += "]";
  } else if (!r && (ne !== B && e instanceof ne || ae !== B && e instanceof ae))
    return "null";
  var i = Object.keys(e).sort();
  if (!i.length && e.constructor && Object.getPrototypeOf(e).constructor !== Object.prototype.constructor) {
    var a = Se.get(e) || Math.random().toString(36).slice(2);
    return Se.set(e, a), R({
      __key: a
    }, r);
  }
  j.add(e);
  var o = "{";
  for (var l of i) {
    var f = R(e[l], r);
    f && (o.length > 1 && (o += ","), o += R(l, r) + ":" + f);
  }
  return j.delete(e), o += "}";
}, de = (e, r, t) => {
  if (!(t == null || typeof t != "object" || t.toJSON || j.has(t))) if (Array.isArray(t))
    for (var n = 0, i = t.length; n < i; n++)
      de(e, `${r}.${n}`, t[n]);
  else if (t instanceof ne || t instanceof ae)
    e.set(r, t);
  else {
    j.add(t);
    for (var a of Object.keys(t))
      de(e, `${r}.${a}`, t[a]);
  }
}, te = (e, r) => (j.clear(), R(e, r || !1));
class B {
}
var ne = typeof File < "u" ? File : B, ae = typeof Blob < "u" ? Blob : B, cr = /("{3}[\s\S]*"{3}|"(?:\\.|[^"])*")/g, fr = /(?:#[^\n\r]+)?(?:[\r\n]+|$)/g, vr = (e, r) => r % 2 == 0 ? e.replace(fr, `
`) : e, Ae = (e) => e.split(cr).map(vr).join("").trim(), De = /* @__PURE__ */ new Map(), Z = /* @__PURE__ */ new Map(), xe = (e) => {
  var r;
  return typeof e == "string" ? r = Ae(e) : e.loc && Z.get(e.__key) === e ? r = e.loc.source.body : (r = De.get(e) || Ae(er(e)), De.set(e, r)), typeof e != "string" && !e.loc && (e.loc = {
    start: 0,
    end: r.length,
    source: {
      body: r,
      name: "gql",
      locationOffset: {
        line: 1,
        column: 1
      }
    }
  }), r;
}, Ie = (e) => {
  var r;
  if (e.documentId)
    r = H(e.documentId);
  else if (r = H(xe(e)), e.definitions) {
    var t = Je(e);
    t && (r = H(`
# ${t}`, r));
  }
  return r;
}, Ge = (e) => {
  var r, t;
  return typeof e == "string" ? (r = Ie(e), t = Z.get(r) || Ze(e)) : (r = e.__key || Ie(e), t = Z.get(r) || e), t.loc || xe(t), t.__key = r, Z.set(r, t), t;
}, ue = (e, r, t) => {
  var n = r || {}, i = Ge(e), a = te(n, !0), o = i.__key;
  return a !== "{}" && (o = H(a, o)), {
    key: o,
    query: i,
    variables: n,
    extensions: t
  };
}, Je = (e) => {
  for (var r of e.definitions)
    if (r.kind === M.OPERATION_DEFINITION)
      return r.name ? r.name.value : void 0;
}, dr = (e) => {
  for (var r of e.definitions)
    if (r.kind === M.OPERATION_DEFINITION)
      return r.operation;
}, pe = (e, r, t) => {
  if (!("data" in r || "errors" in r && Array.isArray(r.errors)))
    throw new Error("No Content");
  var n = e.kind === "subscription";
  return {
    operation: e,
    data: r.data,
    error: Array.isArray(r.errors) ? new ke({
      graphQLErrors: r.errors,
      response: t
    }) : void 0,
    extensions: r.extensions ? {
      ...r.extensions
    } : void 0,
    hasNext: r.hasNext == null ? n : r.hasNext,
    stale: !1
  };
}, he = (e, r) => {
  if (typeof e == "object" && e != null && (!e.constructor || e.constructor === Object || Array.isArray(e))) {
    e = Array.isArray(e) ? [...e] : {
      ...e
    };
    for (var t of Object.keys(r))
      e[t] = he(e[t], r[t]);
    return e;
  }
  return r;
}, pr = (e, r, t, n) => {
  var i = e.error ? e.error.graphQLErrors : [], a = !!e.extensions || !!(r.payload || r).extensions, o = {
    ...e.extensions,
    ...(r.payload || r).extensions
  }, l = r.incremental;
  "path" in r && (l = [r]);
  var f = {
    data: e.data
  };
  if (l) {
    var u = function(c) {
      Array.isArray(c.errors) && i.push(...c.errors), c.extensions && (Object.assign(o, c.extensions), a = !0);
      var m = "data", x = f, k = [];
      if (c.path)
        k = c.path;
      else if (n) {
        var w = n.find((y) => y.id === c.id);
        c.subPath ? k = [...w.path, ...c.subPath] : k = w.path;
      }
      for (var E = 0, D = k.length; E < D; m = k[E++])
        x = x[m] = Array.isArray(x[m]) ? [...x[m]] : {
          ...x[m]
        };
      if (c.items)
        for (var q = +m >= 0 ? m : 0, I = 0, s = c.items.length; I < s; I++)
          x[q + I] = he(x[q + I], c.items[I]);
      else c.data !== void 0 && (x[m] = he(x[m], c.data));
    };
    for (var p of l)
      u(p);
  } else
    f.data = (r.payload || r).data || e.data, i = r.errors || r.payload && r.payload.errors || i;
  return {
    operation: e.operation,
    data: f.data,
    error: i.length ? new ke({
      graphQLErrors: i,
      response: t
    }) : void 0,
    extensions: a ? o : void 0,
    hasNext: r.hasNext != null ? r.hasNext : e.hasNext,
    stale: !1
  };
}, hr = (e, r, t) => ({
  operation: e,
  data: void 0,
  error: new ke({
    networkError: r,
    response: t
  }),
  extensions: void 0,
  hasNext: !1,
  stale: !1
});
function yr(e) {
  var r = {
    query: void 0,
    documentId: void 0,
    operationName: Je(e.query),
    variables: e.variables || void 0,
    extensions: e.extensions
  };
  return "documentId" in e.query && e.query.documentId && (!e.query.definitions || !e.query.definitions.length) ? r.documentId = e.query.documentId : (!e.extensions || !e.extensions.persistedQuery || e.extensions.persistedQuery.miss) && (r.query = xe(e.query)), r;
}
var mr = (e, r) => {
  var t = e.kind === "query" && e.context.preferGetMethod;
  if (!t || !r)
    return e.context.url;
  var n = gr(e.context.url);
  for (var i in r) {
    var a = r[i];
    a && n[1].set(i, typeof a == "object" ? te(a) : a);
  }
  var o = n.join("?");
  return o.length > 2047 && t !== "force" ? (e.context.preferGetMethod = !1, e.context.url) : o;
}, gr = (e) => {
  var r = e.indexOf("?");
  return r > -1 ? [e.slice(0, r), new URLSearchParams(e.slice(r + 1))] : [e, new URLSearchParams()];
}, kr = (e, r) => {
  if (r && !(e.kind === "query" && e.context.preferGetMethod)) {
    var t = te(r), n = ((l) => {
      var f = /* @__PURE__ */ new Map();
      return (ne !== B || ae !== B) && (j.clear(), de(f, "variables", l)), f;
    })(r.variables);
    if (n.size) {
      var i = new FormData();
      i.append("operations", t), i.append("map", te({
        ...[...n.keys()].map((l) => [l])
      }));
      var a = 0;
      for (var o of n.values())
        i.append("" + a++, o);
      return i;
    }
    return t;
  }
}, xr = (e, r) => {
  var t = {
    accept: e.kind === "subscription" ? "text/event-stream, multipart/mixed" : "application/graphql-response+json, application/graphql+json, application/json, text/event-stream, multipart/mixed"
  }, n = (typeof e.context.fetchOptions == "function" ? e.context.fetchOptions() : e.context.fetchOptions) || {};
  if (n.headers)
    if (((o) => "has" in o && !Object.keys(o).length)(n.headers))
      n.headers.forEach((o, l) => {
        t[l] = o;
      });
    else if (Array.isArray(n.headers))
      n.headers.forEach((o, l) => {
        Array.isArray(o) ? t[o[0]] ? t[o[0]] = `${t[o[0]]},${o[1]}` : t[o[0]] = o[1] : t[l] = o;
      });
    else
      for (var i in n.headers)
        t[i.toLowerCase()] = n.headers[i];
  var a = kr(e, r);
  return typeof a == "string" && !t["content-type"] && (t["content-type"] = "application/json"), {
    ...n,
    method: a ? "POST" : "GET",
    body: a,
    headers: t
  };
}, Nr = typeof TextDecoder < "u" ? new TextDecoder() : null, br = /boundary="?([^=";]+)"?/i, wr = /data: ?([^\n]+)/, Ve = (e) => e.constructor.name === "Buffer" ? e.toString() : Nr.decode(e);
async function* _e(e) {
  if (e.body[Symbol.asyncIterator])
    for await (var r of e.body)
      yield Ve(r);
  else {
    var t = e.body.getReader(), n;
    try {
      for (; !(n = await t.read()).done; )
        yield Ve(n.value);
    } finally {
      t.cancel();
    }
  }
}
async function* Te(e, r) {
  var t = "", n;
  for await (var i of e)
    for (t += i; (n = t.indexOf(r)) > -1; )
      yield t.slice(0, n), t = t.slice(n + r.length);
}
async function* Er(e, r, t) {
  var n = !0, i = null, a;
  try {
    yield await Promise.resolve();
    var o = (a = await (e.context.fetch || fetch)(r, t)).headers.get("Content-Type") || "", l;
    /multipart\/mixed/i.test(o) ? l = async function* (c, m) {
      var x = c.match(br), k = "--" + (x ? x[1] : "-"), w = !0, E;
      for await (var D of Te(_e(m), `\r
` + k)) {
        if (w) {
          w = !1;
          var q = D.indexOf(k);
          if (q > -1)
            D = D.slice(q + k.length);
          else
            continue;
        }
        try {
          yield E = JSON.parse(D.slice(D.indexOf(`\r
\r
`) + 4));
        } catch (I) {
          if (!E)
            throw I;
        }
        if (E && E.hasNext === !1)
          break;
      }
      E && E.hasNext !== !1 && (yield {
        hasNext: !1
      });
    }(o, a) : /text\/event-stream/i.test(o) ? l = async function* (c) {
      var m;
      for await (var x of Te(_e(c), `

`)) {
        var k = x.match(wr);
        if (k) {
          var w = k[1];
          try {
            yield m = JSON.parse(w);
          } catch (E) {
            if (!m)
              throw E;
          }
          if (m && m.hasNext === !1)
            break;
        }
      }
      m && m.hasNext !== !1 && (yield {
        hasNext: !1
      });
    }(a) : /text\//i.test(o) ? l = async function* (c) {
      var m = await c.text();
      try {
        var x = JSON.parse(m);
        process.env.NODE_ENV !== "production" && console.warn('Found response with content-type "text/plain" but it had a valid "application/json" response.'), yield x;
      } catch {
        throw new Error(m);
      }
    }(a) : l = async function* (c) {
      yield JSON.parse(await c.text());
    }(a);
    var f;
    for await (var u of l)
      u.pending && !i ? f = u.pending : u.pending && (f = [...f, ...u.pending]), i = i ? pr(i, u, a, f) : pe(e, u, a), n = !1, yield i, n = !0;
    i || (yield i = pe(e, {}, a));
  } catch (p) {
    if (!n)
      throw p;
    yield hr(e, a && (a.status < 200 || a.status >= 300) && a.statusText ? new Error(a.statusText) : p, a);
  }
}
function Or(e, r, t) {
  var n;
  return typeof AbortController < "u" && (t.signal = (n = new AbortController()).signal), je(() => {
    n && n.abort();
  })(A((i) => !!i)($e(Er(e, r, t))));
}
var ye = (e, r) => {
  if (Array.isArray(e))
    for (var t of e)
      ye(t, r);
  else if (typeof e == "object" && e !== null)
    for (var n in e)
      n === "__typename" && typeof e[n] == "string" ? r.add(e[n]) : ye(e[n], r);
  return r;
}, me = (e) => {
  if ("definitions" in e) {
    var r = [];
    for (var t of e.definitions) {
      var n = me(t);
      r.push(n);
    }
    return {
      ...e,
      definitions: r
    };
  }
  if ("directives" in e && e.directives && e.directives.length) {
    var i = [], a = {};
    for (var o of e.directives) {
      var l = o.name.value;
      l[0] !== "_" ? i.push(o) : l = l.slice(1), a[l] = o;
    }
    e = {
      ...e,
      directives: i,
      _directives: a
    };
  }
  if ("selectionSet" in e) {
    var f = [], u = e.kind === M.OPERATION_DEFINITION;
    if (e.selectionSet) {
      for (var p of e.selectionSet.selections || []) {
        u = u || p.kind === M.FIELD && p.name.value === "__typename" && !p.alias;
        var c = me(p);
        f.push(c);
      }
      return u || f.push({
        kind: M.FIELD,
        name: {
          kind: M.NAME,
          value: "__typename"
        },
        _generated: !0
      }), {
        ...e,
        selectionSet: {
          ...e.selectionSet,
          selections: f
        }
      };
    }
  }
  return e;
}, qe = /* @__PURE__ */ new Map(), Sr = (e) => {
  var r = Ge(e), t = qe.get(r.__key);
  return t || (qe.set(r.__key, t = me(r)), Object.defineProperty(t, "__key", {
    value: r.__key,
    enumerable: !1
  })), t;
};
function Ce(e) {
  var r = (t) => e(t);
  return r.toPromise = () => lr(Re(1)(A((t) => !t.stale && !t.hasNext)(r))), r.then = (t, n) => r.toPromise().then(t, n), r.subscribe = (t) => re(t)(r), r;
}
function U(e, r, t) {
  return {
    ...r,
    kind: e,
    context: r.context ? {
      ...r.context,
      ...t
    } : t || r.context
  };
}
var Fe = (e, r) => U(e.kind, e, {
  meta: {
    ...e.context.meta,
    ...r
  }
}), Ar = () => {
}, ce = ({ kind: e }) => e !== "mutation" && e !== "query", Dr = (e) => {
  var r = Sr(e.query);
  if (r !== e.query) {
    var t = U(e.kind, e);
    return t.query = r, t;
  } else
    return e;
}, _r = ({ forward: e, client: r, dispatchDebug: t }) => {
  var n = /* @__PURE__ */ new Map(), i = /* @__PURE__ */ new Map(), a = (o) => o.kind === "query" && o.context.requestPolicy !== "network-only" && (o.context.requestPolicy === "cache-only" || n.has(o.key));
  return (o) => {
    var l = W((u) => {
      var p = n.get(u.key);
      process.env.NODE_ENV !== "production" && t({
        operation: u,
        ...p ? {
          type: "cacheHit",
          message: "The result was successfully retried from the cache"
        } : {
          type: "cacheMiss",
          message: "The result could not be retrieved from the cache"
        },
        source: "cacheExchange"
      });
      var c = p || pe(u, {
        data: null
      });
      return c = {
        ...c,
        operation: process.env.NODE_ENV !== "production" ? Fe(u, {
          cacheOutcome: p ? "hit" : "miss"
        }) : u
      }, u.context.requestPolicy === "cache-and-network" && (c.stale = !0, Pe(r, u)), c;
    })(A((u) => !ce(u) && a(u))(o)), f = ie((u) => {
      var { operation: p } = u;
      if (p) {
        var c = p.context.additionalTypenames || [];
        if (u.operation.kind !== "subscription" && (c = ((y) => [...ye(y, /* @__PURE__ */ new Set())])(u.data).concat(c)), u.operation.kind === "mutation" || u.operation.kind === "subscription") {
          var m = /* @__PURE__ */ new Set();
          process.env.NODE_ENV !== "production" && t({
            type: "cacheInvalidation",
            message: `The following typenames have been invalidated: ${c}`,
            operation: p,
            data: {
              typenames: c,
              response: u
            },
            source: "cacheExchange"
          });
          for (var x = 0; x < c.length; x++) {
            var k = c[x], w = i.get(k);
            w || i.set(k, w = /* @__PURE__ */ new Set());
            for (var E of w.values())
              m.add(E);
            w.clear();
          }
          for (var D of m.values())
            n.has(D) && (p = n.get(D).operation, n.delete(D), Pe(r, p));
        } else if (p.kind === "query" && u.data) {
          n.set(p.key, u);
          for (var q = 0; q < c.length; q++) {
            var I = c[q], s = i.get(I);
            s || i.set(I, s = /* @__PURE__ */ new Set()), s.add(p.key);
          }
        }
      }
    })(e(A((u) => u.kind !== "query" || u.context.requestPolicy !== "cache-only")(W((u) => process.env.NODE_ENV !== "production" ? Fe(u, {
      cacheOutcome: "miss"
    }) : u)(J([W(Dr)(A((u) => !ce(u) && !a(u))(o)), A((u) => ce(u))(o)])))));
    return J([l, f]);
  };
}, Pe = (e, r) => e.reexecuteOperation(U(r.kind, r, {
  requestPolicy: "network-only"
})), Tr = ({ forward: e, dispatchDebug: r }) => (t) => {
  var n = ee((a) => {
    var o = yr(a), l = mr(a, o), f = xr(a, o);
    process.env.NODE_ENV !== "production" && r({
      type: "fetchRequest",
      message: "A fetch request is being executed.",
      operation: a,
      data: {
        url: l,
        fetchOptions: f
      },
      source: "fetchExchange"
    });
    var u = Me(A((p) => p.kind === "teardown" && p.key === a.key)(t))(Or(a, l, f));
    return process.env.NODE_ENV !== "production" ? ie((p) => {
      var c = p.data ? void 0 : p.error;
      process.env.NODE_ENV !== "production" && r({
        type: c ? "fetchError" : "fetchSuccess",
        message: `A ${c ? "failed" : "successful"} fetch response has been returned.`,
        operation: a,
        data: {
          url: l,
          fetchOptions: f,
          value: c || p
        },
        source: "fetchExchange"
      });
    })(u) : u;
  })(A((a) => a.kind !== "teardown" && (a.kind !== "subscription" || !!a.context.fetchSubscriptions))(t)), i = e(A((a) => a.kind === "teardown" || a.kind === "subscription" && !a.context.fetchSubscriptions)(t));
  return J([n, i]);
}, Ir = (e) => ({ client: r, forward: t, dispatchDebug: n }) => e.reduceRight((i, a) => {
  var o = !1;
  return a({
    client: r,
    forward(l) {
      if (process.env.NODE_ENV !== "production") {
        if (o)
          throw new Error("forward() must only be called once in each Exchange.");
        o = !0;
      }
      return Q(i(Q(l)));
    },
    dispatchDebug(l) {
      process.env.NODE_ENV !== "production" && n({
        timestamp: Date.now(),
        source: a.name,
        ...l
      });
    }
  });
}, t), qr = ({ onOperation: e, onResult: r, onError: t }) => ({ forward: n }) => (i) => ee((a) => {
  t && a.error && t(a.error, a.operation);
  var o = r && r(a) || a;
  return "then" in o ? Oe(o) : G(o);
})(n(ee((a) => {
  var o = e && e(a) || a;
  return "then" in o ? Oe(o) : G(o);
})(i))), Vr = ({ dispatchDebug: e }) => (r) => (process.env.NODE_ENV !== "production" && (r = ie((t) => {
  if (t.kind !== "teardown" && process.env.NODE_ENV !== "production") {
    var n = `No exchange has handled operations of kind "${t.kind}". Check whether you've added an exchange responsible for these operations.`;
    process.env.NODE_ENV !== "production" && e({
      type: "fallbackCatch",
      message: n,
      operation: t,
      source: "fallbackExchange"
    }), console.warn(n);
  }
})(r)), A((t) => !1)(r)), Cr = function e(r) {
  if (process.env.NODE_ENV !== "production" && !r.url)
    throw new Error("You are creating an urql-client without a url.");
  var t = 0, n = /* @__PURE__ */ new Map(), i = /* @__PURE__ */ new Map(), a = /* @__PURE__ */ new Set(), o = [], l = {
    url: r.url,
    fetchSubscriptions: r.fetchSubscriptions,
    fetchOptions: r.fetchOptions,
    fetch: r.fetch,
    preferGetMethod: r.preferGetMethod,
    requestPolicy: r.requestPolicy || "cache-first"
  }, f = Ee();
  function u(s) {
    (s.kind === "mutation" || s.kind === "teardown" || !a.has(s.key)) && (s.kind === "teardown" ? a.delete(s.key) : s.kind !== "mutation" && a.add(s.key), f.next(s));
  }
  var p = !1;
  function c(s) {
    if (s && u(s), !p) {
      for (p = !0; p && (s = o.shift()); )
        u(s);
      p = !1;
    }
  }
  var m = (s) => {
    var y = Me(A((v) => v.kind === "teardown" && v.key === s.key)(f.source))(A((v) => v.operation.kind === s.kind && v.operation.key === s.key && (!v.operation.context._instance || v.operation.context._instance === s.context._instance))(I));
    return s.kind !== "query" ? y = nr((v) => !!v.hasNext, !0)(y) : y = we((v) => {
      var V = G(v);
      return v.stale || v.hasNext ? V : J([V, W(() => (v.stale = !0, v))(Re(1)(A((oe) => oe.key === s.key)(f.source)))]);
    })(y), s.kind !== "mutation" ? y = je(() => {
      a.delete(s.key), n.delete(s.key), i.delete(s.key), p = !1;
      for (var v = o.length - 1; v >= 0; v--)
        o[v].key === s.key && o.splice(v, 1);
      u(U("teardown", s, s.context));
    })(ie((v) => {
      if (v.stale) {
        if (!v.hasNext)
          a.delete(s.key);
        else
          for (var V of o)
            if (V.key === v.operation.key) {
              a.delete(V.key);
              break;
            }
      } else v.hasNext || a.delete(s.key);
      n.set(s.key, v);
    })(y)) : y = be(() => {
      u(s);
    })(y), Q(y);
  }, x = this instanceof e ? this : Object.create(e.prototype), k = Object.assign(x, {
    suspense: !!r.suspense,
    operations$: f.source,
    reexecuteOperation(s) {
      if (s.kind === "teardown")
        c(s);
      else if (s.kind === "mutation")
        o.push(s), Promise.resolve().then(c);
      else if (i.has(s.key)) {
        for (var y = !1, v = 0; v < o.length; v++)
          o[v].key === s.key && (o[v] = s, y = !0);
        y || a.has(s.key) && s.context.requestPolicy !== "network-only" ? (a.delete(s.key), Promise.resolve().then(c)) : (o.push(s), Promise.resolve().then(c));
      }
    },
    createRequestOperation(s, y, v) {
      v || (v = {});
      var V;
      if (process.env.NODE_ENV !== "production" && s !== "teardown" && (V = dr(y.query)) !== s)
        throw new Error(`Expected operation of type "${s}" but found "${V}"`);
      return U(s, y, {
        _instance: s === "mutation" ? t = t + 1 | 0 : void 0,
        ...l,
        ...v,
        requestPolicy: v.requestPolicy || l.requestPolicy,
        suspense: v.suspense || v.suspense !== !1 && k.suspense
      });
    },
    executeRequestOperation(s) {
      return s.kind === "mutation" ? Ce(m(s)) : Ce(ar(() => {
        var y = i.get(s.key);
        y || i.set(s.key, y = m(s)), y = be(() => {
          c(s);
        })(y);
        var v = n.get(s.key);
        return s.kind === "query" && v && (v.stale || v.hasNext) ? we(G)(J([y, A((V) => V === n.get(s.key))(G(v))])) : y;
      }));
    },
    executeQuery(s, y) {
      var v = k.createRequestOperation("query", s, y);
      return k.executeRequestOperation(v);
    },
    executeSubscription(s, y) {
      var v = k.createRequestOperation("subscription", s, y);
      return k.executeRequestOperation(v);
    },
    executeMutation(s, y) {
      var v = k.createRequestOperation("mutation", s, y);
      return k.executeRequestOperation(v);
    },
    readQuery(s, y, v) {
      var V = null;
      return re((oe) => {
        V = oe;
      })(k.query(s, y, v)).unsubscribe(), V;
    },
    query: (s, y, v) => k.executeQuery(ue(s, y), v),
    subscription: (s, y, v) => k.executeSubscription(ue(s, y), v),
    mutation: (s, y, v) => k.executeMutation(ue(s, y), v)
  }), w = Ar;
  if (process.env.NODE_ENV !== "production") {
    var { next: E, source: D } = Ee();
    k.subscribeToDebugTarget = (s) => re(s)(D), w = E;
  }
  var q = Ir(r.exchanges), I = Q(q({
    client: k,
    dispatchDebug: w,
    forward: Vr({
      dispatchDebug: w
    })
  })(f.source));
  return sr(I), k;
};
export {
  Cr as C,
  je as a,
  _r as b,
  ue as c,
  Tr as f,
  qr as m,
  ie as o,
  re as s,
  nr as t
};
