// zudoku/lib/utils-Chi3p5nE.js
var ce = Object.defineProperty;
var le = (e, t, r) => t in e ? ce(e, t, { enumerable: !0, configurable: !0, writable: !0, value: r }) : e[t] = r;
var T = (e, t, r) => le(e, typeof t != "symbol" ? t + "" : t, r);
import h, { Component as U } from "react";
import { g as q } from "./index-DJqnphbT.js";
import { j as ue } from "./jsx-runtime-B6kdoens.js";
import { u as $, L as fe } from "./index-BRCiYFaL.js";
import { j as S } from "./joinPath-B7kNnUX4.js";
import { c as K, u as Q } from "./ZudokuContext-BEmsYQoq.js";
const _ = {}, N = h.createContext(_);
function de(e) {
  const t = h.useContext(N);
  return h.useMemo(
    function() {
      return typeof e == "function" ? e(t) : { ...t, ...e };
    },
    [t, e]
  );
}
function We(e) {
  let t;
  return e.disableParentContext ? t = typeof e.components == "function" ? e.components(_) : e.components || _ : t = de(e.components), h.createElement(
    N.Provider,
    { value: t },
    e.children
  );
}
var pe = typeof Element < "u", he = typeof Map == "function", me = typeof Set == "function", ye = typeof ArrayBuffer == "function" && !!ArrayBuffer.isView;
function P(e, t) {
  if (e === t) return !0;
  if (e && t && typeof e == "object" && typeof t == "object") {
    if (e.constructor !== t.constructor) return !1;
    var r, n, s;
    if (Array.isArray(e)) {
      if (r = e.length, r != t.length) return !1;
      for (n = r; n-- !== 0; )
        if (!P(e[n], t[n])) return !1;
      return !0;
    }
    var o;
    if (he && e instanceof Map && t instanceof Map) {
      if (e.size !== t.size) return !1;
      for (o = e.entries(); !(n = o.next()).done; )
        if (!t.has(n.value[0])) return !1;
      for (o = e.entries(); !(n = o.next()).done; )
        if (!P(n.value[1], t.get(n.value[0]))) return !1;
      return !0;
    }
    if (me && e instanceof Set && t instanceof Set) {
      if (e.size !== t.size) return !1;
      for (o = e.entries(); !(n = o.next()).done; )
        if (!t.has(n.value[0])) return !1;
      return !0;
    }
    if (ye && ArrayBuffer.isView(e) && ArrayBuffer.isView(t)) {
      if (r = e.length, r != t.length) return !1;
      for (n = r; n-- !== 0; )
        if (e[n] !== t[n]) return !1;
      return !0;
    }
    if (e.constructor === RegExp) return e.source === t.source && e.flags === t.flags;
    if (e.valueOf !== Object.prototype.valueOf && typeof e.valueOf == "function" && typeof t.valueOf == "function") return e.valueOf() === t.valueOf();
    if (e.toString !== Object.prototype.toString && typeof e.toString == "function" && typeof t.toString == "function") return e.toString() === t.toString();
    if (s = Object.keys(e), r = s.length, r !== Object.keys(t).length) return !1;
    for (n = r; n-- !== 0; )
      if (!Object.prototype.hasOwnProperty.call(t, s[n])) return !1;
    if (pe && e instanceof Element) return !1;
    for (n = r; n-- !== 0; )
      if (!((s[n] === "_owner" || s[n] === "__v" || s[n] === "__o") && e.$$typeof) && !P(e[s[n]], t[s[n]]))
        return !1;
    return !0;
  }
  return e !== e && t !== t;
}
var Te = function(t, r) {
  try {
    return P(t, r);
  } catch (n) {
    if ((n.message || "").match(/stack|recursion/i))
      return console.warn("react-fast-compare cannot handle circular refs"), !1;
    throw n;
  }
};
const ge = /* @__PURE__ */ q(Te);
var ve = function(e, t, r, n, s, o, l, i) {
  if (process.env.NODE_ENV !== "production" && t === void 0)
    throw new Error("invariant requires an error message argument");
  if (!e) {
    var a;
    if (t === void 0)
      a = new Error(
        "Minified exception occurred; use the non-minified dev environment for the full error message and additional helpful warnings."
      );
    else {
      var c = [r, n, s, o, l, i], u = 0;
      a = new Error(
        t.replace(/%s/g, function() {
          return c[u++];
        })
      ), a.name = "Invariant Violation";
    }
    throw a.framesToPop = 1, a;
  }
}, Ae = ve;
const X = /* @__PURE__ */ q(Ae);
var be = function(t, r, n, s) {
  var o = n ? n.call(s, t, r) : void 0;
  if (o !== void 0)
    return !!o;
  if (t === r)
    return !0;
  if (typeof t != "object" || !t || typeof r != "object" || !r)
    return !1;
  var l = Object.keys(t), i = Object.keys(r);
  if (l.length !== i.length)
    return !1;
  for (var a = Object.prototype.hasOwnProperty.bind(r), c = 0; c < l.length; c++) {
    var u = l[c];
    if (!a(u))
      return !1;
    var f = t[u], m = r[u];
    if (o = n ? n.call(s, f, m, u) : void 0, o === !1 || o === void 0 && f !== m)
      return !1;
  }
  return !0;
};
const Ce = /* @__PURE__ */ q(be);
var ee = /* @__PURE__ */ ((e) => (e.BASE = "base", e.BODY = "body", e.HEAD = "head", e.HTML = "html", e.LINK = "link", e.META = "meta", e.NOSCRIPT = "noscript", e.SCRIPT = "script", e.STYLE = "style", e.TITLE = "title", e.FRAGMENT = "Symbol(react.fragment)", e))(ee || {}), D = {
  link: { rel: ["amphtml", "canonical", "alternate"] },
  script: { type: ["application/ld+json"] },
  meta: {
    charset: "",
    name: ["generator", "robots", "description"],
    property: [
      "og:type",
      "og:title",
      "og:url",
      "og:image",
      "og:image:alt",
      "og:description",
      "twitter:url",
      "twitter:title",
      "twitter:description",
      "twitter:image",
      "twitter:image:alt",
      "twitter:card",
      "twitter:site"
    ]
  }
}, Z = Object.values(ee), z = {
  accesskey: "accessKey",
  charset: "charSet",
  class: "className",
  contenteditable: "contentEditable",
  contextmenu: "contextMenu",
  "http-equiv": "httpEquiv",
  itemprop: "itemProp",
  tabindex: "tabIndex"
}, Ee = Object.entries(z).reduce(
  (e, [t, r]) => (e[r] = t, e),
  {}
), y = "data-rh", b = {
  DEFAULT_TITLE: "defaultTitle",
  DEFER: "defer",
  ENCODE_SPECIAL_CHARACTERS: "encodeSpecialCharacters",
  ON_CHANGE_CLIENT_STATE: "onChangeClientState",
  TITLE_TEMPLATE: "titleTemplate",
  PRIORITIZE_SEO_TAGS: "prioritizeSeoTags"
}, C = (e, t) => {
  for (let r = e.length - 1; r >= 0; r -= 1) {
    const n = e[r];
    if (Object.prototype.hasOwnProperty.call(n, t))
      return n[t];
  }
  return null;
}, Oe = (e) => {
  let t = C(
    e,
    "title"
    /* TITLE */
  );
  const r = C(e, b.TITLE_TEMPLATE);
  if (Array.isArray(t) && (t = t.join("")), r && t)
    return r.replace(/%s/g, () => t);
  const n = C(e, b.DEFAULT_TITLE);
  return t || n || void 0;
}, Se = (e) => C(e, b.ON_CHANGE_CLIENT_STATE) || (() => {
}), M = (e, t) => t.filter((r) => typeof r[e] < "u").map((r) => r[e]).reduce((r, n) => ({ ...r, ...n }), {}), we = (e, t) => t.filter((r) => typeof r.base < "u").map((r) => r.base).reverse().reduce((r, n) => {
  if (!r.length) {
    const s = Object.keys(n);
    for (let o = 0; o < s.length; o += 1) {
      const i = s[o].toLowerCase();
      if (e.indexOf(i) !== -1 && n[i])
        return r.concat(n);
    }
  }
  return r;
}, []), xe = (e) => console && typeof console.warn == "function" && console.warn(e), E = (e, t, r) => {
  const n = {};
  return r.filter((s) => Array.isArray(s[e]) ? !0 : (typeof s[e] < "u" && xe(
    `Helmet: ${e} should be of type "Array". Instead found type "${typeof s[e]}"`
  ), !1)).map((s) => s[e]).reverse().reduce((s, o) => {
    const l = {};
    o.filter((a) => {
      let c;
      const u = Object.keys(a);
      for (let m = 0; m < u.length; m += 1) {
        const d = u[m], g = d.toLowerCase();
        t.indexOf(g) !== -1 && !(c === "rel" && a[c].toLowerCase() === "canonical") && !(g === "rel" && a[g].toLowerCase() === "stylesheet") && (c = g), t.indexOf(d) !== -1 && (d === "innerHTML" || d === "cssText" || d === "itemprop") && (c = d);
      }
      if (!c || !a[c])
        return !1;
      const f = a[c].toLowerCase();
      return n[c] || (n[c] = {}), l[c] || (l[c] = {}), n[c][f] ? !1 : (l[c][f] = !0, !0);
    }).reverse().forEach((a) => s.push(a));
    const i = Object.keys(l);
    for (let a = 0; a < i.length; a += 1) {
      const c = i[a], u = {
        ...n[c],
        ...l[c]
      };
      n[c] = u;
    }
    return s;
  }, []).reverse();
}, Pe = (e, t) => {
  if (Array.isArray(e) && e.length) {
    for (let r = 0; r < e.length; r += 1)
      if (e[r][t])
        return !0;
  }
  return !1;
}, ke = (e) => ({
  baseTag: we([
    "href"
    /* HREF */
  ], e),
  bodyAttributes: M("bodyAttributes", e),
  defer: C(e, b.DEFER),
  encode: C(e, b.ENCODE_SPECIAL_CHARACTERS),
  htmlAttributes: M("htmlAttributes", e),
  linkTags: E(
    "link",
    [
      "rel",
      "href"
      /* HREF */
    ],
    e
  ),
  metaTags: E(
    "meta",
    [
      "name",
      "charset",
      "http-equiv",
      "property",
      "itemprop"
      /* ITEM_PROP */
    ],
    e
  ),
  noscriptTags: E("noscript", [
    "innerHTML"
    /* INNER_HTML */
  ], e),
  onChangeClientState: Se(e),
  scriptTags: E(
    "script",
    [
      "src",
      "innerHTML"
      /* INNER_HTML */
    ],
    e
  ),
  styleTags: E("style", [
    "cssText"
    /* CSS_TEXT */
  ], e),
  title: Oe(e),
  titleAttributes: M("titleAttributes", e),
  prioritizeSeoTags: Pe(e, b.PRIORITIZE_SEO_TAGS)
}), te = (e) => Array.isArray(e) ? e.join("") : e, $e = (e, t) => {
  const r = Object.keys(e);
  for (let n = 0; n < r.length; n += 1)
    if (t[r[n]] && t[r[n]].includes(e[r[n]]))
      return !0;
  return !1;
}, I = (e, t) => Array.isArray(e) ? e.reduce(
  (r, n) => ($e(n, t) ? r.priority.push(n) : r.default.push(n), r),
  { priority: [], default: [] }
) : { default: e, priority: [] }, W = (e, t) => ({
  ...e,
  [t]: void 0
}), De = [
  "noscript",
  "script",
  "style"
  /* STYLE */
], R = (e, t = !0) => t === !1 ? String(e) : String(e).replace(/&/g, "&amp;").replace(/</g, "&lt;").replace(/>/g, "&gt;").replace(/"/g, "&quot;").replace(/'/g, "&#x27;"), re = (e) => Object.keys(e).reduce((t, r) => {
  const n = typeof e[r] < "u" ? `${r}="${e[r]}"` : `${r}`;
  return t ? `${t} ${n}` : n;
}, ""), Me = (e, t, r, n) => {
  const s = re(r), o = te(t);
  return s ? `<${e} ${y}="true" ${s}>${R(
    o,
    n
  )}</${e}>` : `<${e} ${y}="true">${R(
    o,
    n
  )}</${e}>`;
}, Ie = (e, t, r = !0) => t.reduce((n, s) => {
  const o = s, l = Object.keys(o).filter(
    (c) => !(c === "innerHTML" || c === "cssText")
  ).reduce((c, u) => {
    const f = typeof o[u] > "u" ? u : `${u}="${R(o[u], r)}"`;
    return c ? `${c} ${f}` : f;
  }, ""), i = o.innerHTML || o.cssText || "", a = De.indexOf(e) === -1;
  return `${n}<${e} ${y}="true" ${l}${a ? "/>" : `>${i}</${e}>`}`;
}, ""), ne = (e, t = {}) => Object.keys(e).reduce((r, n) => {
  const s = z[n];
  return r[s || n] = e[n], r;
}, t), He = (e, t, r) => {
  const n = {
    key: t,
    [y]: !0
  }, s = ne(r, n);
  return [h.createElement("title", s, t)];
}, k = (e, t) => t.map((r, n) => {
  const s = {
    key: n,
    [y]: !0
  };
  return Object.keys(r).forEach((o) => {
    const i = z[o] || o;
    if (i === "innerHTML" || i === "cssText") {
      const a = r.innerHTML || r.cssText;
      s.dangerouslySetInnerHTML = { __html: a };
    } else
      s[i] = r[o];
  }), h.createElement(e, s);
}), p = (e, t, r = !0) => {
  switch (e) {
    case "title":
      return {
        toComponent: () => He(e, t.title, t.titleAttributes),
        toString: () => Me(e, t.title, t.titleAttributes, r)
      };
    case "bodyAttributes":
    case "htmlAttributes":
      return {
        toComponent: () => ne(t),
        toString: () => re(t)
      };
    default:
      return {
        toComponent: () => k(e, t),
        toString: () => Ie(e, t, r)
      };
  }
}, _e = ({ metaTags: e, linkTags: t, scriptTags: r, encode: n }) => {
  const s = I(e, D.meta), o = I(t, D.link), l = I(r, D.script);
  return {
    priorityMethods: {
      toComponent: () => [
        ...k("meta", s.priority),
        ...k("link", o.priority),
        ...k("script", l.priority)
      ],
      toString: () => (
        // generate all the tags as strings and concatenate them
        `${p("meta", s.priority, n)} ${p(
          "link",
          o.priority,
          n
        )} ${p("script", l.priority, n)}`
      )
    },
    metaTags: s.default,
    linkTags: o.default,
    scriptTags: l.default
  };
}, Re = (e) => {
  const {
    baseTag: t,
    bodyAttributes: r,
    encode: n = !0,
    htmlAttributes: s,
    noscriptTags: o,
    styleTags: l,
    title: i = "",
    titleAttributes: a,
    prioritizeSeoTags: c
  } = e;
  let { linkTags: u, metaTags: f, scriptTags: m } = e, d = {
    toComponent: () => {
    },
    toString: () => ""
  };
  return c && ({ priorityMethods: d, linkTags: u, metaTags: f, scriptTags: m } = _e(e)), {
    priority: d,
    base: p("base", t, n),
    bodyAttributes: p("bodyAttributes", r, n),
    htmlAttributes: p("htmlAttributes", s, n),
    link: p("link", u, n),
    meta: p("meta", f, n),
    noscript: p("noscript", o, n),
    script: p("script", m, n),
    style: p("style", l, n),
    title: p("title", { title: i, titleAttributes: a }, n)
  };
}, j = Re, x = [], se = !!(typeof window < "u" && window.document && window.document.createElement), L = class {
  constructor(e, t) {
    T(this, "instances", []);
    T(this, "canUseDOM", se);
    T(this, "context");
    T(this, "value", {
      setHelmet: (e) => {
        this.context.helmet = e;
      },
      helmetInstances: {
        get: () => this.canUseDOM ? x : this.instances,
        add: (e) => {
          (this.canUseDOM ? x : this.instances).push(e);
        },
        remove: (e) => {
          const t = (this.canUseDOM ? x : this.instances).indexOf(e);
          (this.canUseDOM ? x : this.instances).splice(t, 1);
        }
      }
    });
    this.context = e, this.canUseDOM = t || !1, t || (e.helmet = j({
      baseTag: [],
      bodyAttributes: {},
      encodeSpecialCharacters: !0,
      htmlAttributes: {},
      linkTags: [],
      metaTags: [],
      noscriptTags: [],
      scriptTags: [],
      styleTags: [],
      title: "",
      titleAttributes: {}
    }));
  }
}, je = {}, oe = h.createContext(je), v, Le = (v = class extends U {
  constructor(r) {
    super(r);
    T(this, "helmetData");
    this.helmetData = new L(this.props.context || {}, v.canUseDOM);
  }
  render() {
    return /* @__PURE__ */ h.createElement(oe.Provider, { value: this.helmetData.value }, this.props.children);
  }
}, T(v, "canUseDOM", se), v), A = (e, t) => {
  const r = document.head || document.querySelector(
    "head"
    /* HEAD */
  ), n = r.querySelectorAll(`${e}[${y}]`), s = [].slice.call(n), o = [];
  let l;
  return t && t.length && t.forEach((i) => {
    const a = document.createElement(e);
    for (const c in i)
      if (Object.prototype.hasOwnProperty.call(i, c))
        if (c === "innerHTML")
          a.innerHTML = i.innerHTML;
        else if (c === "cssText")
          a.styleSheet ? a.styleSheet.cssText = i.cssText : a.appendChild(document.createTextNode(i.cssText));
        else {
          const u = c, f = typeof i[u] > "u" ? "" : i[u];
          a.setAttribute(c, f);
        }
    a.setAttribute(y, "true"), s.some((c, u) => (l = u, a.isEqualNode(c))) ? s.splice(l, 1) : o.push(a);
  }), s.forEach((i) => {
    var a;
    return (a = i.parentNode) == null ? void 0 : a.removeChild(i);
  }), o.forEach((i) => r.appendChild(i)), {
    oldTags: s,
    newTags: o
  };
}, F = (e, t) => {
  const r = document.getElementsByTagName(e)[0];
  if (!r)
    return;
  const n = r.getAttribute(y), s = n ? n.split(",") : [], o = [...s], l = Object.keys(t);
  for (const i of l) {
    const a = t[i] || "";
    r.getAttribute(i) !== a && r.setAttribute(i, a), s.indexOf(i) === -1 && s.push(i);
    const c = o.indexOf(i);
    c !== -1 && o.splice(c, 1);
  }
  for (let i = o.length - 1; i >= 0; i -= 1)
    r.removeAttribute(o[i]);
  s.length === o.length ? r.removeAttribute(y) : r.getAttribute(y) !== l.join(",") && r.setAttribute(y, l.join(","));
}, Fe = (e, t) => {
  typeof e < "u" && document.title !== e && (document.title = te(e)), F("title", t);
}, G = (e, t) => {
  const {
    baseTag: r,
    bodyAttributes: n,
    htmlAttributes: s,
    linkTags: o,
    metaTags: l,
    noscriptTags: i,
    onChangeClientState: a,
    scriptTags: c,
    styleTags: u,
    title: f,
    titleAttributes: m
  } = e;
  F("body", n), F("html", s), Fe(f, m);
  const d = {
    baseTag: A("base", r),
    linkTags: A("link", o),
    metaTags: A("meta", l),
    noscriptTags: A("noscript", i),
    scriptTags: A("script", c),
    styleTags: A("style", u)
  }, g = {}, Y = {};
  Object.keys(d).forEach((w) => {
    const { newTags: B, oldTags: ae } = d[w];
    B.length && (g[w] = B), ae.length && (Y[w] = d[w].oldTags);
  }), t && t(), a(e, g, Y);
}, O = null, Ue = (e) => {
  O && cancelAnimationFrame(O), e.defer ? O = requestAnimationFrame(() => {
    G(e, () => {
      O = null;
    });
  }) : (G(e), O = null);
}, qe = Ue, J = class extends U {
  constructor() {
    super(...arguments);
    T(this, "rendered", !1);
  }
  shouldComponentUpdate(t) {
    return !Ce(t, this.props);
  }
  componentDidUpdate() {
    this.emitChange();
  }
  componentWillUnmount() {
    const { helmetInstances: t } = this.props.context;
    t.remove(this), this.emitChange();
  }
  emitChange() {
    const { helmetInstances: t, setHelmet: r } = this.props.context;
    let n = null;
    const s = ke(
      t.get().map((o) => {
        const l = { ...o.props };
        return delete l.context, l;
      })
    );
    Le.canUseDOM ? qe(s) : j && (n = j(s)), r(n);
  }
  // componentWillMount will be deprecated
  // for SSR, initialize on first render
  // constructor is also unsafe in StrictMode
  init() {
    if (this.rendered)
      return;
    this.rendered = !0;
    const { helmetInstances: t } = this.props.context;
    t.add(this), this.emitChange();
  }
  render() {
    return this.init(), null;
  }
}, H, Ge = (H = class extends U {
  shouldComponentUpdate(e) {
    return !ge(W(this.props, "helmetData"), W(e, "helmetData"));
  }
  mapNestedChildrenToProps(e, t) {
    if (!t)
      return null;
    switch (e.type) {
      case "script":
      case "noscript":
        return {
          innerHTML: t
        };
      case "style":
        return {
          cssText: t
        };
      default:
        throw new Error(
          `<${e.type} /> elements are self-closing and can not contain children. Refer to our API for more information.`
        );
    }
  }
  flattenArrayTypeChildren(e, t, r, n) {
    return {
      ...t,
      [e.type]: [
        ...t[e.type] || [],
        {
          ...r,
          ...this.mapNestedChildrenToProps(e, n)
        }
      ]
    };
  }
  mapObjectTypeChildren(e, t, r, n) {
    switch (e.type) {
      case "title":
        return {
          ...t,
          [e.type]: n,
          titleAttributes: { ...r }
        };
      case "body":
        return {
          ...t,
          bodyAttributes: { ...r }
        };
      case "html":
        return {
          ...t,
          htmlAttributes: { ...r }
        };
      default:
        return {
          ...t,
          [e.type]: { ...r }
        };
    }
  }
  mapArrayTypeChildrenToProps(e, t) {
    let r = { ...t };
    return Object.keys(e).forEach((n) => {
      r = {
        ...r,
        [n]: e[n]
      };
    }), r;
  }
  warnOnInvalidChildren(e, t) {
    return X(
      Z.some((r) => e.type === r),
      typeof e.type == "function" ? "You may be attempting to nest <Helmet> components within each other, which is not allowed. Refer to our API for more information." : `Only elements types ${Z.join(
        ", "
      )} are allowed. Helmet does not support rendering <${e.type}> elements. Refer to our API for more information.`
    ), X(
      !t || typeof t == "string" || Array.isArray(t) && !t.some((r) => typeof r != "string"),
      `Helmet expects a string as a child of <${e.type}>. Did you forget to wrap your children in braces? ( <${e.type}>{\`\`}</${e.type}> ) Refer to our API for more information.`
    ), !0;
  }
  mapChildrenToProps(e, t) {
    let r = {};
    return h.Children.forEach(e, (n) => {
      if (!n || !n.props)
        return;
      const { children: s, ...o } = n.props, l = Object.keys(o).reduce((a, c) => (a[Ee[c] || c] = o[c], a), {});
      let { type: i } = n;
      switch (typeof i == "symbol" ? i = i.toString() : this.warnOnInvalidChildren(n, s), i) {
        case "Symbol(react.fragment)":
          t = this.mapChildrenToProps(s, t);
          break;
        case "link":
        case "meta":
        case "noscript":
        case "script":
        case "style":
          r = this.flattenArrayTypeChildren(
            n,
            r,
            l,
            s
          );
          break;
        default:
          t = this.mapObjectTypeChildren(n, t, l, s);
          break;
      }
    }), this.mapArrayTypeChildrenToProps(r, t);
  }
  render() {
    const { children: e, ...t } = this.props;
    let r = { ...t }, { helmetData: n } = t;
    if (e && (r = this.mapChildrenToProps(e, r)), n && !(n instanceof L)) {
      const s = n;
      n = new L(s.context, !0), delete r.helmetData;
    }
    return n ? /* @__PURE__ */ h.createElement(J, { ...r, context: n.value }) : /* @__PURE__ */ h.createElement(oe.Consumer, null, (s) => /* @__PURE__ */ h.createElement(J, { ...r, context: s }));
  }
}, T(H, "defaultProps", {
  defer: !0,
  encodeSpecialCharacters: !0,
  prioritizeSeoTags: !1
}), H);
const Je = (e) => {
  const t = $(), r = typeof e.to == "string" ? e.to : e.to.hash, n = (s) => {
    var o;
    !(r != null && r.startsWith("#")) || r !== t.hash || (s.preventDefault(), (o = document.getElementById(r.slice(1))) == null || o.scrollIntoView());
  };
  return /* @__PURE__ */ ue.jsx(fe, { onClick: n, ...e });
}, ie = (e, t) => {
  for (const r of e) {
    const n = V(r, t);
    if (n !== void 0) return n;
  }
}, V = (e, t, r = []) => {
  const n = t(e, r);
  if (n !== void 0) return n;
  if (e.type === "category")
    for (const s of e.items) {
      const o = V(s, t, [
        ...r,
        e
      ]);
      if (o !== void 0) return o;
    }
}, Qe = () => {
  const e = $(), t = K(), { sidebars: r } = Q(), n = t != null && t.id ? r[t.id] : [];
  return ie(n, (s) => {
    if (s.type === "doc" && S(t == null ? void 0 : t.id, s.id) === e.pathname)
      return s;
  });
}, Ne = (e) => {
  const t = $(), r = K();
  return V(e, (n) => {
    if (n.type === "category" && n.link && S(r == null ? void 0 : r.id, n.link.id) === t.pathname || n.type === "doc" && S(r == null ? void 0 : r.id, n.id) === t.pathname)
      return !0;
  });
}, et = () => {
  const e = $().pathname, { sidebars: t } = Q(), r = K(), n = r != null && r.id ? t[r.id] : [];
  let s, o, l = !1;
  return ie(n, (i) => {
    const a = i.type === "doc" ? S(r == null ? void 0 : r.id, i.id) : i.type === "category" && i.link ? S(r == null ? void 0 : r.id, i.link.id) : void 0;
    if (a) {
      if (l)
        return o = { label: i.label, id: a }, !0;
      e === a ? l = !0 : s = { label: i.label, id: a };
    }
  }), { prev: s, next: o };
};
export {
  Je as A,
  Ge as H,
  We as M,
  et as a,
  de as b,
  Le as c,
  Ne as d,
  Qe as u
};
