// zudoku/lib/zudoku.components.js
var Ba = Object.defineProperty;
var fn = (e) => {
  throw TypeError(e);
};
var Ha = (e, t, n) => t in e ? Ba(e, t, { enumerable: !0, configurable: !0, writable: !0, value: n }) : e[t] = n;
var ae = (e, t, n) => Ha(e, typeof t != "symbol" ? t + "" : t, n), hn = (e, t, n) => t.has(e) || fn("Cannot " + n);
var w = (e, t, n) => (hn(e, t, "read from private field"), n ? n.call(e) : t.get(e)), ce = (e, t, n) => t.has(e) ? fn("Cannot add the same private member more than once") : t instanceof WeakSet ? t.add(e) : t.set(e, n), X = (e, t, n, r) => (hn(e, t, "write to private field"), r ? r.call(e, n) : t.set(e, n), n);
var ut = (e, t, n, r) => ({
  set _(o) {
    X(e, t, o, n);
  },
  get _() {
    return w(e, t, r);
  }
});
import { c as On, H as It, M as Ua, d as za, A as Ka, b as qa } from "./utils-Chi3p5nE.js";
import { e as Qa, f as Va, R as Ga, N as Ze, L as At, g as Wa, O as In, b as Ya, u as kt } from "./index-BRCiYFaL.js";
import { E as An, S as xe, a as Xa, R as Ja } from "./SlotletProvider-RvaeLR6z.js";
import { j as a } from "./jsx-runtime-B6kdoens.js";
import { M as Za, f as eo, c as kn, p as $n, A as to, d as rt, j as no, b as A, k as ro, F as ao, D as oo, C as so, P as Ke, i as $t, q as io, s as Fn, R as lo, S as co, t as uo, h as fo, r as Ln, m as ho, a as gt, u as Bn, V as Hn, n as pn, B as Pt } from "./index-Do_BBSIs.js";
import { a as O, c as Un, b as po, T as mo, d as go, V as vo, u as Ft, e as wo, C as xo } from "./Markdown-CWI6lU11.js";
import * as h from "react";
import p, { StrictMode as zn, createContext as Kn, Component as yo, createElement as mn, isValidElement as bo, useState as Ie, useMemo as mt, useLayoutEffect as So, useEffect as Ce, useCallback as qn, Suspense as Qn, memo as Vn, useContext as Co, Fragment as Mo, useRef as Lt, forwardRef as Ro } from "react";
import { a as Bt } from "./index-DJqnphbT.js";
import { B as Eo } from "./router-Oe6YmY6B.js";
import { a as jt, R as Do, T as Po, C as jo } from "./index-Dvh1BL_e.js";
import { S as Gn, f as Wn, Q as No, n as se, m as gn, g as vn, i as be, j as To, k as _o, l as Oo, o as Io, p as wn, r as xn, q as Ao, h as yn, t as bn, v as ko, u as qe, b as $o, Z as Fo, w as Lo, c as Yn, x as Bo } from "./ZudokuContext-BEmsYQoq.js";
import { j as Ht } from "./joinPath-B7kNnUX4.js";
import { ChevronRightIcon as Xn, CheckIcon as Ho, DotIcon as Uo, CircleXIcon as zo, SearchIcon as Ko, MenuIcon as qo, MoonStarIcon as Qo, SunIcon as Vo, ExternalLinkIcon as Go, PanelLeftIcon as Wo } from "lucide-react";
import { c as Jn, P as xt, R as Yo, I as Xo, O as Jo, C as Zo, T as es, a as ts, b as ns, D as rs, d as as, e as os, S as St } from "./SidebarBadge-DmI5hT04.js";
import { S as ss } from "./Spinner-3cQDBVGr.js";
function is({ error: e }) {
  return /* @__PURE__ */ a.jsx(An, { error: e });
}
var at = {};
const ls = /* @__PURE__ */ Bt(Eo), cs = /* @__PURE__ */ Bt(Qa), us = /* @__PURE__ */ Bt(Va);
Object.defineProperty(at, "__esModule", { value: !0 });
var ds = p, we = ls, Ut = cs, Se = us;
function fs(e) {
  if (e && e.__esModule) return e;
  var t = /* @__PURE__ */ Object.create(null);
  return e && Object.keys(e).forEach(function(n) {
    if (n !== "default") {
      var r = Object.getOwnPropertyDescriptor(e, n);
      Object.defineProperty(t, n, r.get ? r : {
        enumerable: !0,
        get: function() {
          return e[n];
        }
      });
    }
  }), t.default = e, Object.freeze(t);
}
var ge = /* @__PURE__ */ fs(ds);
function hs({
  basename: e,
  children: t,
  location: n = "/",
  future: r
}) {
  typeof n == "string" && (n = Se.parsePath(n));
  let o = we.Action.Pop, s = {
    pathname: n.pathname || "/",
    search: n.search || "",
    hash: n.hash || "",
    state: n.state != null ? n.state : null,
    key: n.key || "default"
  }, i = Zn();
  return /* @__PURE__ */ ge.createElement(Se.Router, {
    basename: e,
    children: t,
    location: s,
    navigationType: o,
    navigator: i,
    future: r,
    static: !0
  });
}
function ps({
  context: e,
  router: t,
  hydrate: n = !0,
  nonce: r
}) {
  t && e || (process.env.NODE_ENV !== "production" ? we.UNSAFE_invariant(!1, "You must provide `router` and `context` to <StaticRouterProvider>") : we.UNSAFE_invariant(!1));
  let o = {
    router: t,
    navigator: Zn(),
    static: !0,
    staticContext: e,
    basename: e.basename || "/"
  }, s = /* @__PURE__ */ new Map(), i = "";
  if (n !== !1) {
    let u = {
      loaderData: e.loaderData,
      actionData: e.actionData,
      errors: gs(e.errors)
    };
    i = `window.__staticRouterHydrationData = JSON.parse(${Ss(JSON.stringify(JSON.stringify(u)))});`;
  }
  let {
    state: l
  } = o.router;
  return /* @__PURE__ */ ge.createElement(ge.Fragment, null, /* @__PURE__ */ ge.createElement(Se.UNSAFE_DataRouterContext.Provider, {
    value: o
  }, /* @__PURE__ */ ge.createElement(Se.UNSAFE_DataRouterStateContext.Provider, {
    value: l
  }, /* @__PURE__ */ ge.createElement(Se.UNSAFE_FetchersContext.Provider, {
    value: s
  }, /* @__PURE__ */ ge.createElement(Se.UNSAFE_ViewTransitionContext.Provider, {
    value: {
      isTransitioning: !1
    }
  }, /* @__PURE__ */ ge.createElement(Se.Router, {
    basename: o.basename,
    location: l.location,
    navigationType: l.historyAction,
    navigator: o.navigator,
    static: o.static,
    future: {
      v7_relativeSplatPath: t.future.v7_relativeSplatPath
    }
  }, /* @__PURE__ */ ge.createElement(ms, {
    routes: t.routes,
    future: t.future,
    state: l
  })))))), i ? /* @__PURE__ */ ge.createElement("script", {
    suppressHydrationWarning: !0,
    nonce: r,
    dangerouslySetInnerHTML: {
      __html: i
    }
  }) : null);
}
function ms({
  routes: e,
  future: t,
  state: n
}) {
  return Ut.UNSAFE_useRoutesImpl(e, void 0, n, t);
}
function gs(e) {
  if (!e) return null;
  let t = Object.entries(e), n = {};
  for (let [r, o] of t)
    we.isRouteErrorResponse(o) ? n[r] = {
      ...o,
      __type: "RouteErrorResponse"
    } : o instanceof Error ? n[r] = {
      message: o.message,
      __type: "Error",
      // If this is a subclass (i.e., ReferenceError), send up the type so we
      // can re-create the same type during hydration.
      ...o.name !== "Error" ? {
        __subType: o.name
      } : {}
    } : n[r] = o;
  return n;
}
function Zn() {
  return {
    createHref: er,
    encodeLocation: tr,
    push(e) {
      throw new Error(`You cannot use navigator.push() on the server because it is a stateless environment. This error was probably triggered when you did a \`navigate(${JSON.stringify(e)})\` somewhere in your app.`);
    },
    replace(e) {
      throw new Error(`You cannot use navigator.replace() on the server because it is a stateless environment. This error was probably triggered when you did a \`navigate(${JSON.stringify(e)}, { replace: true })\` somewhere in your app.`);
    },
    go(e) {
      throw new Error(`You cannot use navigator.go() on the server because it is a stateless environment. This error was probably triggered when you did a \`navigate(${e})\` somewhere in your app.`);
    },
    back() {
      throw new Error("You cannot use navigator.back() on the server because it is a stateless environment.");
    },
    forward() {
      throw new Error("You cannot use navigator.forward() on the server because it is a stateless environment.");
    }
  };
}
function vs(e, t) {
  return we.createStaticHandler(e, {
    ...t,
    mapRouteProperties: Ut.UNSAFE_mapRouteProperties
  });
}
function ws(e, t, n = {}) {
  let r = {}, o = we.UNSAFE_convertRoutesToDataRoutes(e, Ut.UNSAFE_mapRouteProperties, void 0, r), s = t.matches.map((l) => {
    let u = r[l.route.id] || l.route;
    return {
      ...l,
      route: u
    };
  }), i = (l) => `You cannot use router.${l}() on the server because it is a stateless environment`;
  return {
    get basename() {
      return t.basename;
    },
    get future() {
      var l, u;
      return {
        v7_fetcherPersist: !1,
        v7_normalizeFormMethod: !1,
        v7_partialHydration: ((l = n.future) == null ? void 0 : l.v7_partialHydration) === !0,
        v7_prependBasename: !1,
        v7_relativeSplatPath: ((u = n.future) == null ? void 0 : u.v7_relativeSplatPath) === !0,
        v7_skipActionErrorRevalidation: !1
      };
    },
    get state() {
      return {
        historyAction: we.Action.Pop,
        location: t.location,
        matches: s,
        loaderData: t.loaderData,
        actionData: t.actionData,
        errors: t.errors,
        initialized: !0,
        navigation: we.IDLE_NAVIGATION,
        restoreScrollPosition: null,
        preventScrollReset: !1,
        revalidation: "idle",
        fetchers: /* @__PURE__ */ new Map(),
        blockers: /* @__PURE__ */ new Map()
      };
    },
    get routes() {
      return o;
    },
    get window() {
    },
    initialize() {
      throw i("initialize");
    },
    subscribe() {
      throw i("subscribe");
    },
    enableScrollRestoration() {
      throw i("enableScrollRestoration");
    },
    navigate() {
      throw i("navigate");
    },
    fetch() {
      throw i("fetch");
    },
    revalidate() {
      throw i("revalidate");
    },
    createHref: er,
    encodeLocation: tr,
    getFetcher() {
      return we.IDLE_FETCHER;
    },
    deleteFetcher() {
      throw i("deleteFetcher");
    },
    dispose() {
      throw i("dispose");
    },
    getBlocker() {
      return we.IDLE_BLOCKER;
    },
    deleteBlocker() {
      throw i("deleteBlocker");
    },
    patchRoutes() {
      throw i("patchRoutes");
    },
    _internalFetchControllers: /* @__PURE__ */ new Map(),
    _internalActiveDeferreds: /* @__PURE__ */ new Map(),
    _internalSetRoutes() {
      throw i("_internalSetRoutes");
    }
  };
}
function er(e) {
  return typeof e == "string" ? e : Se.createPath(e);
}
function tr(e) {
  let t = typeof e == "string" ? e : Se.createPath(e);
  t = t.replace(/ $/, "%20");
  let n = xs.test(t) ? new URL(t) : new URL(t, "http://localhost");
  return {
    pathname: n.pathname,
    search: n.search,
    hash: n.hash
  };
}
const xs = /^(?:[a-z][a-z0-9+.-]*:|\/\/)/i, ys = {
  "&": "\\u0026",
  ">": "\\u003e",
  "<": "\\u003c",
  "\u2028": "\\u2028",
  "\u2029": "\\u2029"
}, bs = /[&><\u2028\u2029]/g;
function Ss(e) {
  return e.replace(bs, (t) => ys[t]);
}
at.StaticRouter = hs;
var Cs = at.StaticRouterProvider = ps;
at.createStaticHandler = vs;
at.createStaticRouter = ws;
const Ms = ({
  router: e,
  hydrate: t = !1
}) => /* @__PURE__ */ a.jsx(zn, { children: /* @__PURE__ */ a.jsx(On, { children: /* @__PURE__ */ a.jsx(jt.Provider, { value: { stagger: !t }, children: /* @__PURE__ */ a.jsx(Ga, { router: e }) }) }) }), Rs = ({
  router: e,
  context: t,
  helmetContext: n
}) => /* @__PURE__ */ a.jsx(zn, { children: /* @__PURE__ */ a.jsx(On, { context: n, children: /* @__PURE__ */ a.jsx(Cs, { router: e, context: t }) }) });
var ve, Nn, Es = (Nn = class extends Gn {
  constructor(t = {}) {
    super();
    ce(this, ve);
    this.config = t, X(this, ve, /* @__PURE__ */ new Map());
  }
  build(t, n, r) {
    const o = n.queryKey, s = n.queryHash ?? Wn(o, n);
    let i = this.get(s);
    return i || (i = new No({
      cache: this,
      queryKey: o,
      queryHash: s,
      options: t.defaultQueryOptions(n),
      state: r,
      defaultOptions: t.getQueryDefaults(o)
    }), this.add(i)), i;
  }
  add(t) {
    w(this, ve).has(t.queryHash) || (w(this, ve).set(t.queryHash, t), this.notify({
      type: "added",
      query: t
    }));
  }
  remove(t) {
    const n = w(this, ve).get(t.queryHash);
    n && (t.destroy(), n === t && w(this, ve).delete(t.queryHash), this.notify({ type: "removed", query: t }));
  }
  clear() {
    se.batch(() => {
      this.getAll().forEach((t) => {
        this.remove(t);
      });
    });
  }
  get(t) {
    return w(this, ve).get(t);
  }
  getAll() {
    return [...w(this, ve).values()];
  }
  find(t) {
    const n = { exact: !0, ...t };
    return this.getAll().find(
      (r) => gn(n, r)
    );
  }
  findAll(t = {}) {
    const n = this.getAll();
    return Object.keys(t).length > 0 ? n.filter((r) => gn(t, r)) : n;
  }
  notify(t) {
    se.batch(() => {
      this.listeners.forEach((n) => {
        n(t);
      });
    });
  }
  onFocus() {
    se.batch(() => {
      this.getAll().forEach((t) => {
        t.onFocus();
      });
    });
  }
  onOnline() {
    se.batch(() => {
      this.getAll().forEach((t) => {
        t.onOnline();
      });
    });
  }
}, ve = new WeakMap(), Nn), oe, nt, Tn, Ds = (Tn = class extends Gn {
  constructor(t = {}) {
    super();
    ce(this, oe);
    ce(this, nt);
    this.config = t, X(this, oe, /* @__PURE__ */ new Map()), X(this, nt, Date.now());
  }
  build(t, n, r) {
    const o = new Za({
      mutationCache: this,
      mutationId: ++ut(this, nt)._,
      options: t.defaultMutationOptions(n),
      state: r
    });
    return this.add(o), o;
  }
  add(t) {
    const n = dt(t), r = w(this, oe).get(n) ?? [];
    r.push(t), w(this, oe).set(n, r), this.notify({ type: "added", mutation: t });
  }
  remove(t) {
    var r;
    const n = dt(t);
    if (w(this, oe).has(n)) {
      const o = (r = w(this, oe).get(n)) == null ? void 0 : r.filter((s) => s !== t);
      o && (o.length === 0 ? w(this, oe).delete(n) : w(this, oe).set(n, o));
    }
    this.notify({ type: "removed", mutation: t });
  }
  canRun(t) {
    var r;
    const n = (r = w(this, oe).get(dt(t))) == null ? void 0 : r.find((o) => o.state.status === "pending");
    return !n || n === t;
  }
  runNext(t) {
    var r;
    const n = (r = w(this, oe).get(dt(t))) == null ? void 0 : r.find((o) => o !== t && o.state.isPaused);
    return (n == null ? void 0 : n.continue()) ?? Promise.resolve();
  }
  clear() {
    se.batch(() => {
      this.getAll().forEach((t) => {
        this.remove(t);
      });
    });
  }
  getAll() {
    return [...w(this, oe).values()].flat();
  }
  find(t) {
    const n = { exact: !0, ...t };
    return this.getAll().find(
      (r) => vn(n, r)
    );
  }
  findAll(t = {}) {
    return this.getAll().filter((n) => vn(t, n));
  }
  notify(t) {
    se.batch(() => {
      this.listeners.forEach((n) => {
        n(t);
      });
    });
  }
  resumePausedMutations() {
    const t = this.getAll().filter((n) => n.state.isPaused);
    return se.batch(
      () => Promise.all(
        t.map((n) => n.continue().catch(be))
      )
    );
  }
}, oe = new WeakMap(), nt = new WeakMap(), Tn);
function dt(e) {
  var t;
  return ((t = e.options.scope) == null ? void 0 : t.id) ?? String(e.mutationId);
}
function Sn(e) {
  return {
    onFetch: (t, n) => {
      var f, c, d, v, M;
      const r = t.options, o = (d = (c = (f = t.fetchOptions) == null ? void 0 : f.meta) == null ? void 0 : c.fetchMore) == null ? void 0 : d.direction, s = ((v = t.state.data) == null ? void 0 : v.pages) || [], i = ((M = t.state.data) == null ? void 0 : M.pageParams) || [];
      let l = { pages: [], pageParams: [] }, u = 0;
      const m = async () => {
        let C = !1;
        const _ = (P) => {
          Object.defineProperty(P, "signal", {
            enumerable: !0,
            get: () => (t.signal.aborted ? C = !0 : t.signal.addEventListener("abort", () => {
              C = !0;
            }), t.signal)
          });
        }, R = To(t.options, t.fetchOptions), k = async (P, g, z) => {
          if (C)
            return Promise.reject();
          if (g == null && P.pages.length)
            return Promise.resolve(P);
          const Z = {
            queryKey: t.queryKey,
            pageParam: g,
            direction: z ? "backward" : "forward",
            meta: t.options.meta
          };
          _(Z);
          const b = await R(
            Z
          ), { maxPages: j } = t.options, D = z ? _o : Oo;
          return {
            pages: D(P.pages, b, j),
            pageParams: D(P.pageParams, g, j)
          };
        };
        if (o && s.length) {
          const P = o === "backward", g = P ? Ps : Cn, z = {
            pages: s,
            pageParams: i
          }, Z = g(r, z);
          l = await k(z, Z, P);
        } else {
          const P = e ?? s.length;
          do {
            const g = u === 0 ? i[0] ?? r.initialPageParam : Cn(r, l);
            if (u > 0 && g == null)
              break;
            l = await k(l, g), u++;
          } while (u < P);
        }
        return l;
      };
      t.options.persister ? t.fetchFn = () => {
        var C, _;
        return (_ = (C = t.options).persister) == null ? void 0 : _.call(
          C,
          m,
          {
            queryKey: t.queryKey,
            meta: t.options.meta,
            signal: t.signal
          },
          n
        );
      } : t.fetchFn = m;
    }
  };
}
function Cn(e, { pages: t, pageParams: n }) {
  const r = t.length - 1;
  return t.length > 0 ? e.getNextPageParam(
    t[r],
    t,
    n[r],
    n
  ) : void 0;
}
function Ps(e, { pages: t, pageParams: n }) {
  var r;
  return t.length > 0 ? (r = e.getPreviousPageParam) == null ? void 0 : r.call(e, t[0], t, n[0], n) : void 0;
}
var F, Pe, je, Be, He, Ne, Ue, ze, _n, js = (_n = class {
  constructor(e = {}) {
    ce(this, F);
    ce(this, Pe);
    ce(this, je);
    ce(this, Be);
    ce(this, He);
    ce(this, Ne);
    ce(this, Ue);
    ce(this, ze);
    X(this, F, e.queryCache || new Es()), X(this, Pe, e.mutationCache || new Ds()), X(this, je, e.defaultOptions || {}), X(this, Be, /* @__PURE__ */ new Map()), X(this, He, /* @__PURE__ */ new Map()), X(this, Ne, 0);
  }
  mount() {
    ut(this, Ne)._++, w(this, Ne) === 1 && (X(this, Ue, Io.subscribe(async (e) => {
      e && (await this.resumePausedMutations(), w(this, F).onFocus());
    })), X(this, ze, wn.subscribe(async (e) => {
      e && (await this.resumePausedMutations(), w(this, F).onOnline());
    })));
  }
  unmount() {
    var e, t;
    ut(this, Ne)._--, w(this, Ne) === 0 && ((e = w(this, Ue)) == null || e.call(this), X(this, Ue, void 0), (t = w(this, ze)) == null || t.call(this), X(this, ze, void 0));
  }
  isFetching(e) {
    return w(this, F).findAll({ ...e, fetchStatus: "fetching" }).length;
  }
  isMutating(e) {
    return w(this, Pe).findAll({ ...e, status: "pending" }).length;
  }
  getQueryData(e) {
    var n;
    const t = this.defaultQueryOptions({ queryKey: e });
    return (n = w(this, F).get(t.queryHash)) == null ? void 0 : n.state.data;
  }
  ensureQueryData(e) {
    const t = this.getQueryData(e.queryKey);
    if (t === void 0)
      return this.fetchQuery(e);
    {
      const n = this.defaultQueryOptions(e), r = w(this, F).build(this, n);
      return e.revalidateIfStale && r.isStaleByTime(xn(n.staleTime, r)) && this.prefetchQuery(n), Promise.resolve(t);
    }
  }
  getQueriesData(e) {
    return w(this, F).findAll(e).map(({ queryKey: t, state: n }) => {
      const r = n.data;
      return [t, r];
    });
  }
  setQueryData(e, t, n) {
    const r = this.defaultQueryOptions({ queryKey: e }), o = w(this, F).get(
      r.queryHash
    ), s = o == null ? void 0 : o.state.data, i = Ao(t, s);
    if (i !== void 0)
      return w(this, F).build(this, r).setData(i, { ...n, manual: !0 });
  }
  setQueriesData(e, t, n) {
    return se.batch(
      () => w(this, F).findAll(e).map(({ queryKey: r }) => [
        r,
        this.setQueryData(r, t, n)
      ])
    );
  }
  getQueryState(e) {
    var n;
    const t = this.defaultQueryOptions({ queryKey: e });
    return (n = w(this, F).get(t.queryHash)) == null ? void 0 : n.state;
  }
  removeQueries(e) {
    const t = w(this, F);
    se.batch(() => {
      t.findAll(e).forEach((n) => {
        t.remove(n);
      });
    });
  }
  resetQueries(e, t) {
    const n = w(this, F), r = {
      type: "active",
      ...e
    };
    return se.batch(() => (n.findAll(e).forEach((o) => {
      o.reset();
    }), this.refetchQueries(r, t)));
  }
  cancelQueries(e = {}, t = {}) {
    const n = { revert: !0, ...t }, r = se.batch(
      () => w(this, F).findAll(e).map((o) => o.cancel(n))
    );
    return Promise.all(r).then(be).catch(be);
  }
  invalidateQueries(e = {}, t = {}) {
    return se.batch(() => {
      if (w(this, F).findAll(e).forEach((r) => {
        r.invalidate();
      }), e.refetchType === "none")
        return Promise.resolve();
      const n = {
        ...e,
        type: e.refetchType ?? e.type ?? "active"
      };
      return this.refetchQueries(n, t);
    });
  }
  refetchQueries(e = {}, t) {
    const n = {
      ...t,
      cancelRefetch: (t == null ? void 0 : t.cancelRefetch) ?? !0
    }, r = se.batch(
      () => w(this, F).findAll(e).filter((o) => !o.isDisabled()).map((o) => {
        let s = o.fetch(void 0, n);
        return n.throwOnError || (s = s.catch(be)), o.state.fetchStatus === "paused" ? Promise.resolve() : s;
      })
    );
    return Promise.all(r).then(be);
  }
  fetchQuery(e) {
    const t = this.defaultQueryOptions(e);
    t.retry === void 0 && (t.retry = !1);
    const n = w(this, F).build(this, t);
    return n.isStaleByTime(
      xn(t.staleTime, n)
    ) ? n.fetch(t) : Promise.resolve(n.state.data);
  }
  prefetchQuery(e) {
    return this.fetchQuery(e).then(be).catch(be);
  }
  fetchInfiniteQuery(e) {
    return e.behavior = Sn(e.pages), this.fetchQuery(e);
  }
  prefetchInfiniteQuery(e) {
    return this.fetchInfiniteQuery(e).then(be).catch(be);
  }
  ensureInfiniteQueryData(e) {
    return e.behavior = Sn(e.pages), this.ensureQueryData(e);
  }
  resumePausedMutations() {
    return wn.isOnline() ? w(this, Pe).resumePausedMutations() : Promise.resolve();
  }
  getQueryCache() {
    return w(this, F);
  }
  getMutationCache() {
    return w(this, Pe);
  }
  getDefaultOptions() {
    return w(this, je);
  }
  setDefaultOptions(e) {
    X(this, je, e);
  }
  setQueryDefaults(e, t) {
    w(this, Be).set(yn(e), {
      queryKey: e,
      defaultOptions: t
    });
  }
  getQueryDefaults(e) {
    const t = [...w(this, Be).values()];
    let n = {};
    return t.forEach((r) => {
      bn(e, r.queryKey) && (n = { ...n, ...r.defaultOptions });
    }), n;
  }
  setMutationDefaults(e, t) {
    w(this, He).set(yn(e), {
      mutationKey: e,
      defaultOptions: t
    });
  }
  getMutationDefaults(e) {
    const t = [...w(this, He).values()];
    let n = {};
    return t.forEach((r) => {
      bn(e, r.mutationKey) && (n = { ...n, ...r.defaultOptions });
    }), n;
  }
  defaultQueryOptions(e) {
    if (e._defaulted)
      return e;
    const t = {
      ...w(this, je).queries,
      ...this.getQueryDefaults(e.queryKey),
      ...e,
      _defaulted: !0
    };
    return t.queryHash || (t.queryHash = Wn(
      t.queryKey,
      t
    )), t.refetchOnReconnect === void 0 && (t.refetchOnReconnect = t.networkMode !== "always"), t.throwOnError === void 0 && (t.throwOnError = !!t.suspense), !t.networkMode && t.persister && (t.networkMode = "offlineFirst"), t.enabled !== !0 && t.queryFn === ko && (t.enabled = !1), t;
  }
  defaultMutationOptions(e) {
    return e != null && e._defaulted ? e : {
      ...w(this, je).mutations,
      ...(e == null ? void 0 : e.mutationKey) && this.getMutationDefaults(e.mutationKey),
      ...e,
      _defaulted: !0
    };
  }
  clear() {
    w(this, F).clear(), w(this, Pe).clear();
  }
}, F = new WeakMap(), Pe = new WeakMap(), je = new WeakMap(), Be = new WeakMap(), He = new WeakMap(), Ne = new WeakMap(), Ue = new WeakMap(), ze = new WeakMap(), _n);
const Ns = Kn(null), Ct = {
  didCatch: !1,
  error: null
};
class Ts extends yo {
  constructor(t) {
    super(t), this.resetErrorBoundary = this.resetErrorBoundary.bind(this), this.state = Ct;
  }
  static getDerivedStateFromError(t) {
    return {
      didCatch: !0,
      error: t
    };
  }
  resetErrorBoundary() {
    const {
      error: t
    } = this.state;
    if (t !== null) {
      for (var n, r, o = arguments.length, s = new Array(o), i = 0; i < o; i++)
        s[i] = arguments[i];
      (n = (r = this.props).onReset) === null || n === void 0 || n.call(r, {
        args: s,
        reason: "imperative-api"
      }), this.setState(Ct);
    }
  }
  componentDidCatch(t, n) {
    var r, o;
    (r = (o = this.props).onError) === null || r === void 0 || r.call(o, t, n);
  }
  componentDidUpdate(t, n) {
    const {
      didCatch: r
    } = this.state, {
      resetKeys: o
    } = this.props;
    if (r && n.error !== null && _s(t.resetKeys, o)) {
      var s, i;
      (s = (i = this.props).onReset) === null || s === void 0 || s.call(i, {
        next: o,
        prev: t.resetKeys,
        reason: "keys"
      }), this.setState(Ct);
    }
  }
  render() {
    const {
      children: t,
      fallbackRender: n,
      FallbackComponent: r,
      fallback: o
    } = this.props, {
      didCatch: s,
      error: i
    } = this.state;
    let l = t;
    if (s) {
      const u = {
        error: i,
        resetErrorBoundary: this.resetErrorBoundary
      };
      if (typeof n == "function")
        l = n(u);
      else if (r)
        l = mn(r, u);
      else if (o === null || bo(o))
        l = o;
      else
        throw i;
    }
    return mn(Ns.Provider, {
      value: {
        didCatch: s,
        error: i,
        resetErrorBoundary: this.resetErrorBoundary
      }
    }, l);
  }
}
function _s() {
  let e = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : [], t = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : [];
  return e.length !== t.length || e.some((n, r) => !Object.is(n, t[r]));
}
const Os = (e) => "getProfileMenuItems" in e && typeof e.getProfileMenuItems == "function", Is = (e) => "getRoutes" in e && typeof e.getRoutes == "function", As = (e) => "renderSearch" in e && typeof e.renderSearch == "function", ks = (e) => "initialize" in e && typeof e.initialize == "function", $s = (e) => "getHead" in e && typeof e.getHead == "function", Fs = (e) => "getMdxComponents" in e && typeof e.getMdxComponents == "function", Ls = (e) => "getIdentities" in e && typeof e.getIdentities == "function", nr = new js();
class Bs {
  constructor(t) {
    ae(this, "plugins");
    ae(this, "sidebars");
    ae(this, "topNavigation");
    ae(this, "meta");
    ae(this, "page");
    ae(this, "authentication");
    ae(this, "navigationPlugins");
    ae(this, "initialize", async () => {
      await Promise.all(
        this.plugins.filter(ks).map((t) => {
          var n;
          return (n = t.initialize) == null ? void 0 : n.call(t, this);
        })
      );
    });
    ae(this, "invalidateCache", async (t) => {
      await nr.invalidateQueries({ queryKey: t });
    });
    ae(this, "getApiIdentities", async () => (await Promise.all(
      this.plugins.filter(Ls).map((n) => n.getIdentities(this))
    )).flat());
    ae(this, "getPluginSidebar", async (t) => (await Promise.all(
      this.navigationPlugins.map(
        (r) => {
          var o;
          return (o = r.getSidebar) == null ? void 0 : o.call(r, Ht(t));
        }
      )
    )).flatMap((r) => r ?? []));
    ae(this, "signRequest", async (t) => {
      if (!this.authentication)
        throw new Error("No authentication provider configured");
      const n = await this.authentication.getAccessToken();
      return t.headers.set("Authorization", `Bearer ${n}`), t;
    });
    this.plugins = t.plugins ?? [], this.topNavigation = t.topNavigation ?? [], this.sidebars = t.sidebars ?? {}, this.navigationPlugins = this.plugins.filter(Is), this.authentication = t.authentication, this.meta = t.metadata, this.page = t.page;
  }
}
function Hs({ error: e, resetErrorBoundary: t }) {
  return /* @__PURE__ */ a.jsx(An, { error: e });
}
const ft = globalThis;
(!ft.requestIdleCallback || !ft.cancelIdleCallback) && (ft.requestIdleCallback = (e) => setTimeout(e, 1), ft.cancelIdleCallback = clearTimeout);
var Nt = ["Enter", " "], Us = ["ArrowDown", "PageUp", "Home"], rr = ["ArrowUp", "PageDown", "End"], zs = [...Us, ...rr], Ks = {
  ltr: [...Nt, "ArrowRight"],
  rtl: [...Nt, "ArrowLeft"]
}, qs = {
  ltr: ["ArrowLeft"],
  rtl: ["ArrowRight"]
}, ot = "Menu", [et, Qs, Vs] = eo(ot), [Ae, ar] = kn(ot, [
  Vs,
  $n,
  Jn
]), st = $n(), or = Jn(), [sr, Te] = Ae(ot), [Gs, it] = Ae(ot), ir = (e) => {
  const { __scopeMenu: t, open: n = !1, children: r, dir: o, onOpenChange: s, modal: i = !0 } = e, l = st(t), [u, m] = h.useState(null), f = h.useRef(!1), c = $t(s), d = fo(o);
  return h.useEffect(() => {
    const v = () => {
      f.current = !0, document.addEventListener("pointerdown", M, { capture: !0, once: !0 }), document.addEventListener("pointermove", M, { capture: !0, once: !0 });
    }, M = () => f.current = !1;
    return document.addEventListener("keydown", v, { capture: !0 }), () => {
      document.removeEventListener("keydown", v, { capture: !0 }), document.removeEventListener("pointerdown", M, { capture: !0 }), document.removeEventListener("pointermove", M, { capture: !0 });
    };
  }, []), /* @__PURE__ */ a.jsx(Ln, { ...l, children: /* @__PURE__ */ a.jsx(
    sr,
    {
      scope: t,
      open: n,
      onOpenChange: c,
      content: u,
      onContentChange: m,
      children: /* @__PURE__ */ a.jsx(
        Gs,
        {
          scope: t,
          onClose: h.useCallback(() => c(!1), [c]),
          isUsingKeyboardRef: f,
          dir: d,
          modal: i,
          children: r
        }
      )
    }
  ) });
};
ir.displayName = ot;
var Ws = "MenuAnchor", zt = h.forwardRef(
  (e, t) => {
    const { __scopeMenu: n, ...r } = e, o = st(n);
    return /* @__PURE__ */ a.jsx(to, { ...o, ...r, ref: t });
  }
);
zt.displayName = Ws;
var Kt = "MenuPortal", [Ys, lr] = Ae(Kt, {
  forceMount: void 0
}), cr = (e) => {
  const { __scopeMenu: t, forceMount: n, children: r, container: o } = e, s = Te(Kt, t);
  return /* @__PURE__ */ a.jsx(Ys, { scope: t, forceMount: n, children: /* @__PURE__ */ a.jsx(xt, { present: n || s.open, children: /* @__PURE__ */ a.jsx(ho, { asChild: !0, container: o, children: r }) }) });
};
cr.displayName = Kt;
var ue = "MenuContent", [Xs, qt] = Ae(ue), ur = h.forwardRef(
  (e, t) => {
    const n = lr(ue, e.__scopeMenu), { forceMount: r = n.forceMount, ...o } = e, s = Te(ue, e.__scopeMenu), i = it(ue, e.__scopeMenu);
    return /* @__PURE__ */ a.jsx(et.Provider, { scope: e.__scopeMenu, children: /* @__PURE__ */ a.jsx(xt, { present: r || s.open, children: /* @__PURE__ */ a.jsx(et.Slot, { scope: e.__scopeMenu, children: i.modal ? /* @__PURE__ */ a.jsx(Js, { ...o, ref: t }) : /* @__PURE__ */ a.jsx(Zs, { ...o, ref: t }) }) }) });
  }
), Js = h.forwardRef(
  (e, t) => {
    const n = Te(ue, e.__scopeMenu), r = h.useRef(null), o = rt(t, r);
    return h.useEffect(() => {
      const s = r.current;
      if (s) return no(s);
    }, []), /* @__PURE__ */ a.jsx(
      Qt,
      {
        ...e,
        ref: o,
        trapFocus: n.open,
        disableOutsidePointerEvents: n.open,
        disableOutsideScroll: !0,
        onFocusOutside: A(
          e.onFocusOutside,
          (s) => s.preventDefault(),
          { checkForDefaultPrevented: !1 }
        ),
        onDismiss: () => n.onOpenChange(!1)
      }
    );
  }
), Zs = h.forwardRef((e, t) => {
  const n = Te(ue, e.__scopeMenu);
  return /* @__PURE__ */ a.jsx(
    Qt,
    {
      ...e,
      ref: t,
      trapFocus: !1,
      disableOutsidePointerEvents: !1,
      disableOutsideScroll: !1,
      onDismiss: () => n.onOpenChange(!1)
    }
  );
}), Qt = h.forwardRef(
  (e, t) => {
    const {
      __scopeMenu: n,
      loop: r = !1,
      trapFocus: o,
      onOpenAutoFocus: s,
      onCloseAutoFocus: i,
      disableOutsidePointerEvents: l,
      onEntryFocus: u,
      onEscapeKeyDown: m,
      onPointerDownOutside: f,
      onFocusOutside: c,
      onInteractOutside: d,
      onDismiss: v,
      disableOutsideScroll: M,
      ...C
    } = e, _ = Te(ue, n), R = it(ue, n), k = st(n), P = or(n), g = Qs(n), [z, Z] = h.useState(null), b = h.useRef(null), j = rt(t, b, _.onContentChange), D = h.useRef(0), B = h.useRef(""), K = h.useRef(0), W = h.useRef(null), x = h.useRef("right"), V = h.useRef(0), G = M ? lo : h.Fragment, te = M ? { as: co, allowPinchZoom: !0 } : void 0, q = (y) => {
      var le, Re;
      const Q = B.current + y, ie = g().filter((re) => !re.disabled), de = document.activeElement, pe = (le = ie.find((re) => re.ref.current === de)) == null ? void 0 : le.textValue, Me = ie.map((re) => re.textValue), ke = di(Me, Q, pe), ye = (Re = ie.find((re) => re.textValue === ke)) == null ? void 0 : Re.ref.current;
      (function re($e) {
        B.current = $e, window.clearTimeout(D.current), $e !== "" && (D.current = window.setTimeout(() => re(""), 1e3));
      })(Q), ye && setTimeout(() => ye.focus());
    };
    h.useEffect(() => () => window.clearTimeout(D.current), []), ro();
    const Y = h.useCallback((y) => {
      var ie, de;
      return x.current === ((ie = W.current) == null ? void 0 : ie.side) && hi(y, (de = W.current) == null ? void 0 : de.area);
    }, []);
    return /* @__PURE__ */ a.jsx(
      Xs,
      {
        scope: n,
        searchRef: B,
        onItemEnter: h.useCallback(
          (y) => {
            Y(y) && y.preventDefault();
          },
          [Y]
        ),
        onItemLeave: h.useCallback(
          (y) => {
            var Q;
            Y(y) || ((Q = b.current) == null || Q.focus(), Z(null));
          },
          [Y]
        ),
        onTriggerLeave: h.useCallback(
          (y) => {
            Y(y) && y.preventDefault();
          },
          [Y]
        ),
        pointerGraceTimerRef: K,
        onPointerGraceIntentChange: h.useCallback((y) => {
          W.current = y;
        }, []),
        children: /* @__PURE__ */ a.jsx(G, { ...te, children: /* @__PURE__ */ a.jsx(
          ao,
          {
            asChild: !0,
            trapped: o,
            onMountAutoFocus: A(s, (y) => {
              var Q;
              y.preventDefault(), (Q = b.current) == null || Q.focus({ preventScroll: !0 });
            }),
            onUnmountAutoFocus: i,
            children: /* @__PURE__ */ a.jsx(
              oo,
              {
                asChild: !0,
                disableOutsidePointerEvents: l,
                onEscapeKeyDown: m,
                onPointerDownOutside: f,
                onFocusOutside: c,
                onInteractOutside: d,
                onDismiss: v,
                children: /* @__PURE__ */ a.jsx(
                  Yo,
                  {
                    asChild: !0,
                    ...P,
                    dir: R.dir,
                    orientation: "vertical",
                    loop: r,
                    currentTabStopId: z,
                    onCurrentTabStopIdChange: Z,
                    onEntryFocus: A(u, (y) => {
                      R.isUsingKeyboardRef.current || y.preventDefault();
                    }),
                    preventScrollOnEntryFocus: !0,
                    children: /* @__PURE__ */ a.jsx(
                      so,
                      {
                        role: "menu",
                        "aria-orientation": "vertical",
                        "data-state": Dr(_.open),
                        "data-radix-menu-content": "",
                        dir: R.dir,
                        ...k,
                        ...C,
                        ref: j,
                        style: { outline: "none", ...C.style },
                        onKeyDown: A(C.onKeyDown, (y) => {
                          const ie = y.target.closest("[data-radix-menu-content]") === y.currentTarget, de = y.ctrlKey || y.altKey || y.metaKey, pe = y.key.length === 1;
                          ie && (y.key === "Tab" && y.preventDefault(), !de && pe && q(y.key));
                          const Me = b.current;
                          if (y.target !== Me || !zs.includes(y.key)) return;
                          y.preventDefault();
                          const ye = g().filter((le) => !le.disabled).map((le) => le.ref.current);
                          rr.includes(y.key) && ye.reverse(), ci(ye);
                        }),
                        onBlur: A(e.onBlur, (y) => {
                          y.currentTarget.contains(y.target) || (window.clearTimeout(D.current), B.current = "");
                        }),
                        onPointerMove: A(
                          e.onPointerMove,
                          tt((y) => {
                            const Q = y.target, ie = V.current !== y.clientX;
                            if (y.currentTarget.contains(Q) && ie) {
                              const de = y.clientX > V.current ? "right" : "left";
                              x.current = de, V.current = y.clientX;
                            }
                          })
                        )
                      }
                    )
                  }
                )
              }
            )
          }
        ) })
      }
    );
  }
);
ur.displayName = ue;
var ei = "MenuGroup", Vt = h.forwardRef(
  (e, t) => {
    const { __scopeMenu: n, ...r } = e;
    return /* @__PURE__ */ a.jsx(Ke.div, { role: "group", ...r, ref: t });
  }
);
Vt.displayName = ei;
var ti = "MenuLabel", dr = h.forwardRef(
  (e, t) => {
    const { __scopeMenu: n, ...r } = e;
    return /* @__PURE__ */ a.jsx(Ke.div, { ...r, ref: t });
  }
);
dr.displayName = ti;
var vt = "MenuItem", Mn = "menu.itemSelect", yt = h.forwardRef(
  (e, t) => {
    const { disabled: n = !1, onSelect: r, ...o } = e, s = h.useRef(null), i = it(vt, e.__scopeMenu), l = qt(vt, e.__scopeMenu), u = rt(t, s), m = h.useRef(!1), f = () => {
      const c = s.current;
      if (!n && c) {
        const d = new CustomEvent(Mn, { bubbles: !0, cancelable: !0 });
        c.addEventListener(Mn, (v) => r == null ? void 0 : r(v), { once: !0 }), uo(c, d), d.defaultPrevented ? m.current = !1 : i.onClose();
      }
    };
    return /* @__PURE__ */ a.jsx(
      fr,
      {
        ...o,
        ref: u,
        disabled: n,
        onClick: A(e.onClick, f),
        onPointerDown: (c) => {
          var d;
          (d = e.onPointerDown) == null || d.call(e, c), m.current = !0;
        },
        onPointerUp: A(e.onPointerUp, (c) => {
          var d;
          m.current || (d = c.currentTarget) == null || d.click();
        }),
        onKeyDown: A(e.onKeyDown, (c) => {
          const d = l.searchRef.current !== "";
          n || d && c.key === " " || Nt.includes(c.key) && (c.currentTarget.click(), c.preventDefault());
        })
      }
    );
  }
);
yt.displayName = vt;
var fr = h.forwardRef(
  (e, t) => {
    const { __scopeMenu: n, disabled: r = !1, textValue: o, ...s } = e, i = qt(vt, n), l = or(n), u = h.useRef(null), m = rt(t, u), [f, c] = h.useState(!1), [d, v] = h.useState("");
    return h.useEffect(() => {
      const M = u.current;
      M && v((M.textContent ?? "").trim());
    }, [s.children]), /* @__PURE__ */ a.jsx(
      et.ItemSlot,
      {
        scope: n,
        disabled: r,
        textValue: o ?? d,
        children: /* @__PURE__ */ a.jsx(Xo, { asChild: !0, ...l, focusable: !r, children: /* @__PURE__ */ a.jsx(
          Ke.div,
          {
            role: "menuitem",
            "data-highlighted": f ? "" : void 0,
            "aria-disabled": r || void 0,
            "data-disabled": r ? "" : void 0,
            ...s,
            ref: m,
            onPointerMove: A(
              e.onPointerMove,
              tt((M) => {
                r ? i.onItemLeave(M) : (i.onItemEnter(M), M.defaultPrevented || M.currentTarget.focus({ preventScroll: !0 }));
              })
            ),
            onPointerLeave: A(
              e.onPointerLeave,
              tt((M) => i.onItemLeave(M))
            ),
            onFocus: A(e.onFocus, () => c(!0)),
            onBlur: A(e.onBlur, () => c(!1))
          }
        ) })
      }
    );
  }
), ni = "MenuCheckboxItem", hr = h.forwardRef(
  (e, t) => {
    const { checked: n = !1, onCheckedChange: r, ...o } = e;
    return /* @__PURE__ */ a.jsx(wr, { scope: e.__scopeMenu, checked: n, children: /* @__PURE__ */ a.jsx(
      yt,
      {
        role: "menuitemcheckbox",
        "aria-checked": wt(n) ? "mixed" : n,
        ...o,
        ref: t,
        "data-state": Yt(n),
        onSelect: A(
          o.onSelect,
          () => r == null ? void 0 : r(wt(n) ? !0 : !n),
          { checkForDefaultPrevented: !1 }
        )
      }
    ) });
  }
);
hr.displayName = ni;
var pr = "MenuRadioGroup", [ri, ai] = Ae(
  pr,
  { value: void 0, onValueChange: () => {
  } }
), mr = h.forwardRef(
  (e, t) => {
    const { value: n, onValueChange: r, ...o } = e, s = $t(r);
    return /* @__PURE__ */ a.jsx(ri, { scope: e.__scopeMenu, value: n, onValueChange: s, children: /* @__PURE__ */ a.jsx(Vt, { ...o, ref: t }) });
  }
);
mr.displayName = pr;
var gr = "MenuRadioItem", vr = h.forwardRef(
  (e, t) => {
    const { value: n, ...r } = e, o = ai(gr, e.__scopeMenu), s = n === o.value;
    return /* @__PURE__ */ a.jsx(wr, { scope: e.__scopeMenu, checked: s, children: /* @__PURE__ */ a.jsx(
      yt,
      {
        role: "menuitemradio",
        "aria-checked": s,
        ...r,
        ref: t,
        "data-state": Yt(s),
        onSelect: A(
          r.onSelect,
          () => {
            var i;
            return (i = o.onValueChange) == null ? void 0 : i.call(o, n);
          },
          { checkForDefaultPrevented: !1 }
        )
      }
    ) });
  }
);
vr.displayName = gr;
var Gt = "MenuItemIndicator", [wr, oi] = Ae(
  Gt,
  { checked: !1 }
), xr = h.forwardRef(
  (e, t) => {
    const { __scopeMenu: n, forceMount: r, ...o } = e, s = oi(Gt, n);
    return /* @__PURE__ */ a.jsx(
      xt,
      {
        present: r || wt(s.checked) || s.checked === !0,
        children: /* @__PURE__ */ a.jsx(
          Ke.span,
          {
            ...o,
            ref: t,
            "data-state": Yt(s.checked)
          }
        )
      }
    );
  }
);
xr.displayName = Gt;
var si = "MenuSeparator", yr = h.forwardRef(
  (e, t) => {
    const { __scopeMenu: n, ...r } = e;
    return /* @__PURE__ */ a.jsx(
      Ke.div,
      {
        role: "separator",
        "aria-orientation": "horizontal",
        ...r,
        ref: t
      }
    );
  }
);
yr.displayName = si;
var ii = "MenuArrow", br = h.forwardRef(
  (e, t) => {
    const { __scopeMenu: n, ...r } = e, o = st(n);
    return /* @__PURE__ */ a.jsx(io, { ...o, ...r, ref: t });
  }
);
br.displayName = ii;
var Wt = "MenuSub", [li, Sr] = Ae(Wt), Cr = (e) => {
  const { __scopeMenu: t, children: n, open: r = !1, onOpenChange: o } = e, s = Te(Wt, t), i = st(t), [l, u] = h.useState(null), [m, f] = h.useState(null), c = $t(o);
  return h.useEffect(() => (s.open === !1 && c(!1), () => c(!1)), [s.open, c]), /* @__PURE__ */ a.jsx(Ln, { ...i, children: /* @__PURE__ */ a.jsx(
    sr,
    {
      scope: t,
      open: r,
      onOpenChange: c,
      content: m,
      onContentChange: f,
      children: /* @__PURE__ */ a.jsx(
        li,
        {
          scope: t,
          contentId: gt(),
          triggerId: gt(),
          trigger: l,
          onTriggerChange: u,
          children: n
        }
      )
    }
  ) });
};
Cr.displayName = Wt;
var Xe = "MenuSubTrigger", Mr = h.forwardRef(
  (e, t) => {
    const n = Te(Xe, e.__scopeMenu), r = it(Xe, e.__scopeMenu), o = Sr(Xe, e.__scopeMenu), s = qt(Xe, e.__scopeMenu), i = h.useRef(null), { pointerGraceTimerRef: l, onPointerGraceIntentChange: u } = s, m = { __scopeMenu: e.__scopeMenu }, f = h.useCallback(() => {
      i.current && window.clearTimeout(i.current), i.current = null;
    }, []);
    return h.useEffect(() => f, [f]), h.useEffect(() => {
      const c = l.current;
      return () => {
        window.clearTimeout(c), u(null);
      };
    }, [l, u]), /* @__PURE__ */ a.jsx(zt, { asChild: !0, ...m, children: /* @__PURE__ */ a.jsx(
      fr,
      {
        id: o.triggerId,
        "aria-haspopup": "menu",
        "aria-expanded": n.open,
        "aria-controls": o.contentId,
        "data-state": Dr(n.open),
        ...e,
        ref: Fn(t, o.onTriggerChange),
        onClick: (c) => {
          var d;
          (d = e.onClick) == null || d.call(e, c), !(e.disabled || c.defaultPrevented) && (c.currentTarget.focus(), n.open || n.onOpenChange(!0));
        },
        onPointerMove: A(
          e.onPointerMove,
          tt((c) => {
            s.onItemEnter(c), !c.defaultPrevented && !e.disabled && !n.open && !i.current && (s.onPointerGraceIntentChange(null), i.current = window.setTimeout(() => {
              n.onOpenChange(!0), f();
            }, 100));
          })
        ),
        onPointerLeave: A(
          e.onPointerLeave,
          tt((c) => {
            var v, M;
            f();
            const d = (v = n.content) == null ? void 0 : v.getBoundingClientRect();
            if (d) {
              const C = (M = n.content) == null ? void 0 : M.dataset.side, _ = C === "right", R = _ ? -5 : 5, k = d[_ ? "left" : "right"], P = d[_ ? "right" : "left"];
              s.onPointerGraceIntentChange({
                area: [
                  // Apply a bleed on clientX to ensure that our exit point is
                  // consistently within polygon bounds
                  { x: c.clientX + R, y: c.clientY },
                  { x: k, y: d.top },
                  { x: P, y: d.top },
                  { x: P, y: d.bottom },
                  { x: k, y: d.bottom }
                ],
                side: C
              }), window.clearTimeout(l.current), l.current = window.setTimeout(
                () => s.onPointerGraceIntentChange(null),
                300
              );
            } else {
              if (s.onTriggerLeave(c), c.defaultPrevented) return;
              s.onPointerGraceIntentChange(null);
            }
          })
        ),
        onKeyDown: A(e.onKeyDown, (c) => {
          var v;
          const d = s.searchRef.current !== "";
          e.disabled || d && c.key === " " || Ks[r.dir].includes(c.key) && (n.onOpenChange(!0), (v = n.content) == null || v.focus(), c.preventDefault());
        })
      }
    ) });
  }
);
Mr.displayName = Xe;
var Rr = "MenuSubContent", Er = h.forwardRef(
  (e, t) => {
    const n = lr(ue, e.__scopeMenu), { forceMount: r = n.forceMount, ...o } = e, s = Te(ue, e.__scopeMenu), i = it(ue, e.__scopeMenu), l = Sr(Rr, e.__scopeMenu), u = h.useRef(null), m = rt(t, u);
    return /* @__PURE__ */ a.jsx(et.Provider, { scope: e.__scopeMenu, children: /* @__PURE__ */ a.jsx(xt, { present: r || s.open, children: /* @__PURE__ */ a.jsx(et.Slot, { scope: e.__scopeMenu, children: /* @__PURE__ */ a.jsx(
      Qt,
      {
        id: l.contentId,
        "aria-labelledby": l.triggerId,
        ...o,
        ref: m,
        align: "start",
        side: i.dir === "rtl" ? "left" : "right",
        disableOutsidePointerEvents: !1,
        disableOutsideScroll: !1,
        trapFocus: !1,
        onOpenAutoFocus: (f) => {
          var c;
          i.isUsingKeyboardRef.current && ((c = u.current) == null || c.focus()), f.preventDefault();
        },
        onCloseAutoFocus: (f) => f.preventDefault(),
        onFocusOutside: A(e.onFocusOutside, (f) => {
          f.target !== l.trigger && s.onOpenChange(!1);
        }),
        onEscapeKeyDown: A(e.onEscapeKeyDown, (f) => {
          i.onClose(), f.preventDefault();
        }),
        onKeyDown: A(e.onKeyDown, (f) => {
          var v;
          const c = f.currentTarget.contains(f.target), d = qs[i.dir].includes(f.key);
          c && d && (s.onOpenChange(!1), (v = l.trigger) == null || v.focus(), f.preventDefault());
        })
      }
    ) }) }) });
  }
);
Er.displayName = Rr;
function Dr(e) {
  return e ? "open" : "closed";
}
function wt(e) {
  return e === "indeterminate";
}
function Yt(e) {
  return wt(e) ? "indeterminate" : e ? "checked" : "unchecked";
}
function ci(e) {
  const t = document.activeElement;
  for (const n of e)
    if (n === t || (n.focus(), document.activeElement !== t)) return;
}
function ui(e, t) {
  return e.map((n, r) => e[(t + r) % e.length]);
}
function di(e, t, n) {
  const o = t.length > 1 && Array.from(t).every((m) => m === t[0]) ? t[0] : t, s = n ? e.indexOf(n) : -1;
  let i = ui(e, Math.max(s, 0));
  o.length === 1 && (i = i.filter((m) => m !== n));
  const u = i.find(
    (m) => m.toLowerCase().startsWith(o.toLowerCase())
  );
  return u !== n ? u : void 0;
}
function fi(e, t) {
  const { x: n, y: r } = e;
  let o = !1;
  for (let s = 0, i = t.length - 1; s < t.length; i = s++) {
    const l = t[s].x, u = t[s].y, m = t[i].x, f = t[i].y;
    u > r != f > r && n < (m - l) * (r - u) / (f - u) + l && (o = !o);
  }
  return o;
}
function hi(e, t) {
  if (!t) return !1;
  const n = { x: e.clientX, y: e.clientY };
  return fi(n, t);
}
function tt(e) {
  return (t) => t.pointerType === "mouse" ? e(t) : void 0;
}
var pi = ir, mi = zt, gi = cr, vi = ur, wi = Vt, xi = dr, yi = yt, bi = hr, Si = mr, Ci = vr, Mi = xr, Ri = yr, Ei = br, Di = Cr, Pi = Mr, ji = Er, Xt = "DropdownMenu", [Ni, uc] = kn(
  Xt,
  [ar]
), J = ar(), [Ti, Pr] = Ni(Xt), jr = (e) => {
  const {
    __scopeDropdownMenu: t,
    children: n,
    dir: r,
    open: o,
    defaultOpen: s,
    onOpenChange: i,
    modal: l = !0
  } = e, u = J(t), m = h.useRef(null), [f = !1, c] = Bn({
    prop: o,
    defaultProp: s,
    onChange: i
  });
  return /* @__PURE__ */ a.jsx(
    Ti,
    {
      scope: t,
      triggerId: gt(),
      triggerRef: m,
      contentId: gt(),
      open: f,
      onOpenChange: c,
      onOpenToggle: h.useCallback(() => c((d) => !d), [c]),
      modal: l,
      children: /* @__PURE__ */ a.jsx(pi, { ...u, open: f, onOpenChange: c, dir: r, modal: l, children: n })
    }
  );
};
jr.displayName = Xt;
var Nr = "DropdownMenuTrigger", Tr = h.forwardRef(
  (e, t) => {
    const { __scopeDropdownMenu: n, disabled: r = !1, ...o } = e, s = Pr(Nr, n), i = J(n);
    return /* @__PURE__ */ a.jsx(mi, { asChild: !0, ...i, children: /* @__PURE__ */ a.jsx(
      Ke.button,
      {
        type: "button",
        id: s.triggerId,
        "aria-haspopup": "menu",
        "aria-expanded": s.open,
        "aria-controls": s.open ? s.contentId : void 0,
        "data-state": s.open ? "open" : "closed",
        "data-disabled": r ? "" : void 0,
        disabled: r,
        ...o,
        ref: Fn(t, s.triggerRef),
        onPointerDown: A(e.onPointerDown, (l) => {
          !r && l.button === 0 && l.ctrlKey === !1 && (s.onOpenToggle(), s.open || l.preventDefault());
        }),
        onKeyDown: A(e.onKeyDown, (l) => {
          r || (["Enter", " "].includes(l.key) && s.onOpenToggle(), l.key === "ArrowDown" && s.onOpenChange(!0), ["Enter", " ", "ArrowDown"].includes(l.key) && l.preventDefault());
        })
      }
    ) });
  }
);
Tr.displayName = Nr;
var _i = "DropdownMenuPortal", _r = (e) => {
  const { __scopeDropdownMenu: t, ...n } = e, r = J(t);
  return /* @__PURE__ */ a.jsx(gi, { ...r, ...n });
};
_r.displayName = _i;
var Or = "DropdownMenuContent", Ir = h.forwardRef(
  (e, t) => {
    const { __scopeDropdownMenu: n, ...r } = e, o = Pr(Or, n), s = J(n), i = h.useRef(!1);
    return /* @__PURE__ */ a.jsx(
      vi,
      {
        id: o.contentId,
        "aria-labelledby": o.triggerId,
        ...s,
        ...r,
        ref: t,
        onCloseAutoFocus: A(e.onCloseAutoFocus, (l) => {
          var u;
          i.current || (u = o.triggerRef.current) == null || u.focus(), i.current = !1, l.preventDefault();
        }),
        onInteractOutside: A(e.onInteractOutside, (l) => {
          const u = l.detail.originalEvent, m = u.button === 0 && u.ctrlKey === !0, f = u.button === 2 || m;
          (!o.modal || f) && (i.current = !0);
        }),
        style: {
          ...e.style,
          "--radix-dropdown-menu-content-transform-origin": "var(--radix-popper-transform-origin)",
          "--radix-dropdown-menu-content-available-width": "var(--radix-popper-available-width)",
          "--radix-dropdown-menu-content-available-height": "var(--radix-popper-available-height)",
          "--radix-dropdown-menu-trigger-width": "var(--radix-popper-anchor-width)",
          "--radix-dropdown-menu-trigger-height": "var(--radix-popper-anchor-height)"
        }
      }
    );
  }
);
Ir.displayName = Or;
var Oi = "DropdownMenuGroup", Ii = h.forwardRef(
  (e, t) => {
    const { __scopeDropdownMenu: n, ...r } = e, o = J(n);
    return /* @__PURE__ */ a.jsx(wi, { ...o, ...r, ref: t });
  }
);
Ii.displayName = Oi;
var Ai = "DropdownMenuLabel", Ar = h.forwardRef(
  (e, t) => {
    const { __scopeDropdownMenu: n, ...r } = e, o = J(n);
    return /* @__PURE__ */ a.jsx(xi, { ...o, ...r, ref: t });
  }
);
Ar.displayName = Ai;
var ki = "DropdownMenuItem", kr = h.forwardRef(
  (e, t) => {
    const { __scopeDropdownMenu: n, ...r } = e, o = J(n);
    return /* @__PURE__ */ a.jsx(yi, { ...o, ...r, ref: t });
  }
);
kr.displayName = ki;
var $i = "DropdownMenuCheckboxItem", $r = h.forwardRef((e, t) => {
  const { __scopeDropdownMenu: n, ...r } = e, o = J(n);
  return /* @__PURE__ */ a.jsx(bi, { ...o, ...r, ref: t });
});
$r.displayName = $i;
var Fi = "DropdownMenuRadioGroup", Li = h.forwardRef((e, t) => {
  const { __scopeDropdownMenu: n, ...r } = e, o = J(n);
  return /* @__PURE__ */ a.jsx(Si, { ...o, ...r, ref: t });
});
Li.displayName = Fi;
var Bi = "DropdownMenuRadioItem", Fr = h.forwardRef((e, t) => {
  const { __scopeDropdownMenu: n, ...r } = e, o = J(n);
  return /* @__PURE__ */ a.jsx(Ci, { ...o, ...r, ref: t });
});
Fr.displayName = Bi;
var Hi = "DropdownMenuItemIndicator", Lr = h.forwardRef((e, t) => {
  const { __scopeDropdownMenu: n, ...r } = e, o = J(n);
  return /* @__PURE__ */ a.jsx(Mi, { ...o, ...r, ref: t });
});
Lr.displayName = Hi;
var Ui = "DropdownMenuSeparator", Br = h.forwardRef((e, t) => {
  const { __scopeDropdownMenu: n, ...r } = e, o = J(n);
  return /* @__PURE__ */ a.jsx(Ri, { ...o, ...r, ref: t });
});
Br.displayName = Ui;
var zi = "DropdownMenuArrow", Ki = h.forwardRef(
  (e, t) => {
    const { __scopeDropdownMenu: n, ...r } = e, o = J(n);
    return /* @__PURE__ */ a.jsx(Ei, { ...o, ...r, ref: t });
  }
);
Ki.displayName = zi;
var qi = (e) => {
  const { __scopeDropdownMenu: t, children: n, open: r, onOpenChange: o, defaultOpen: s } = e, i = J(t), [l = !1, u] = Bn({
    prop: r,
    defaultProp: s,
    onChange: o
  });
  return /* @__PURE__ */ a.jsx(Di, { ...i, open: l, onOpenChange: u, children: n });
}, Qi = "DropdownMenuSubTrigger", Hr = h.forwardRef((e, t) => {
  const { __scopeDropdownMenu: n, ...r } = e, o = J(n);
  return /* @__PURE__ */ a.jsx(Pi, { ...o, ...r, ref: t });
});
Hr.displayName = Qi;
var Vi = "DropdownMenuSubContent", Ur = h.forwardRef((e, t) => {
  const { __scopeDropdownMenu: n, ...r } = e, o = J(n);
  return /* @__PURE__ */ a.jsx(
    ji,
    {
      ...o,
      ...r,
      ref: t,
      style: {
        ...e.style,
        "--radix-dropdown-menu-content-transform-origin": "var(--radix-popper-transform-origin)",
        "--radix-dropdown-menu-content-available-width": "var(--radix-popper-available-width)",
        "--radix-dropdown-menu-content-available-height": "var(--radix-popper-available-height)",
        "--radix-dropdown-menu-trigger-width": "var(--radix-popper-anchor-width)",
        "--radix-dropdown-menu-trigger-height": "var(--radix-popper-anchor-height)"
      }
    }
  );
});
Ur.displayName = Vi;
var Gi = jr, Wi = Tr, zr = _r, Kr = Ir, qr = Ar, Qr = kr, Vr = $r, Gr = Fr, Wr = Lr, Yr = Br, Yi = qi, Xr = Hr, Jr = Ur;
const Xi = Gi, Ji = Wi, Zi = zr, el = Yi, Zr = h.forwardRef(({ className: e, inset: t, children: n, ...r }, o) => /* @__PURE__ */ a.jsxs(
  Xr,
  {
    ref: o,
    className: O(
      "flex cursor-default select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none focus:bg-accent data-[state=open]:bg-accent",
      t && "pl-8",
      e
    ),
    ...r,
    children: [
      n,
      /* @__PURE__ */ a.jsx(Xn, { className: "ml-auto h-4 w-4" })
    ]
  }
));
Zr.displayName = Xr.displayName;
const ea = h.forwardRef(({ className: e, ...t }, n) => /* @__PURE__ */ a.jsx(
  Jr,
  {
    ref: n,
    className: O(
      "z-50 min-w-[8rem] overflow-hidden rounded-md border bg-popover p-1 text-popover-foreground shadow-lg data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2",
      e
    ),
    ...t
  }
));
ea.displayName = Jr.displayName;
const ta = h.forwardRef(({ className: e, sideOffset: t = 4, ...n }, r) => /* @__PURE__ */ a.jsx(zr, { children: /* @__PURE__ */ a.jsx(
  Kr,
  {
    ref: r,
    sideOffset: t,
    className: O(
      "z-50 min-w-[8rem] overflow-hidden rounded-md border bg-popover p-1 text-popover-foreground shadow-md",
      "data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2",
      e
    ),
    ...n
  }
) }));
ta.displayName = Kr.displayName;
const na = h.forwardRef(({ className: e, inset: t, ...n }, r) => /* @__PURE__ */ a.jsx(
  Qr,
  {
    ref: r,
    className: O(
      "relative flex cursor-default select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none transition-colors focus:bg-accent focus:text-accent-foreground data-[disabled]:pointer-events-none data-[disabled]:opacity-50",
      t && "pl-8",
      e
    ),
    ...n
  }
));
na.displayName = Qr.displayName;
const tl = h.forwardRef(({ className: e, children: t, checked: n, ...r }, o) => /* @__PURE__ */ a.jsxs(
  Vr,
  {
    ref: o,
    className: O(
      "relative flex cursor-default select-none items-center rounded-sm py-1.5 pl-8 pr-2 text-sm outline-none transition-colors focus:bg-accent focus:text-accent-foreground data-[disabled]:pointer-events-none data-[disabled]:opacity-50",
      e
    ),
    checked: n,
    ...r,
    children: [
      /* @__PURE__ */ a.jsx("span", { className: "absolute left-2 flex h-3.5 w-3.5 items-center justify-center", children: /* @__PURE__ */ a.jsx(Wr, { children: /* @__PURE__ */ a.jsx(Ho, { className: "h-4 w-4" }) }) }),
      t
    ]
  }
));
tl.displayName = Vr.displayName;
const nl = h.forwardRef(({ className: e, children: t, ...n }, r) => /* @__PURE__ */ a.jsxs(
  Gr,
  {
    ref: r,
    className: O(
      "relative flex cursor-default select-none items-center rounded-sm py-1.5 pl-8 pr-2 text-sm outline-none transition-colors focus:bg-accent focus:text-accent-foreground data-[disabled]:pointer-events-none data-[disabled]:opacity-50",
      e
    ),
    ...n,
    children: [
      /* @__PURE__ */ a.jsx("span", { className: "absolute left-2 flex h-3.5 w-3.5 items-center justify-center", children: /* @__PURE__ */ a.jsx(Wr, { children: /* @__PURE__ */ a.jsx(Uo, { className: "h-4 w-4 fill-current" }) }) }),
      t
    ]
  }
));
nl.displayName = Gr.displayName;
const ra = h.forwardRef(({ className: e, inset: t, ...n }, r) => /* @__PURE__ */ a.jsx(
  qr,
  {
    ref: r,
    className: O(
      "px-2 py-1.5 text-sm font-semibold",
      t && "pl-8",
      e
    ),
    ...n
  }
));
ra.displayName = qr.displayName;
const aa = h.forwardRef(({ className: e, ...t }, n) => /* @__PURE__ */ a.jsx(
  Yr,
  {
    ref: n,
    className: O("-mx-1 my-1 h-px bg-muted", e),
    ...t
  }
));
aa.displayName = Yr.displayName;
const Rn = {
  info: "bg-blue-500",
  note: "bg-gray-500",
  tip: "bg-green-600",
  caution: "bg-orange-500",
  danger: "bg-rose-500"
}, rl = () => {
  const { page: e } = qe(), [t, n] = Ie(!0);
  if (!(e != null && e.banner) || !t)
    return /* @__PURE__ */ a.jsx("style", { children: ":root { --banner-height: 0px; }" });
  const r = e.banner.color && e.banner.color in Rn ? Rn[e.banner.color] : e.banner.color ? void 0 : "bg-primary", o = r ? {} : { backgroundColor: e.banner.color };
  return /* @__PURE__ */ a.jsxs(
    "div",
    {
      className: O(
        "relative text-primary-foreground text-sm font-medium px-4 py-2 flex gap-2 items-center",
        r
      ),
      style: o,
      children: [
        /* @__PURE__ */ a.jsx("div", { className: "w-full", children: e.banner.message }),
        e.banner.dismissible && /* @__PURE__ */ a.jsx(
          "button",
          {
            type: "button",
            className: "md:absolute md:right-4 -m-1.5 p-1.5 hover:bg-accent-foreground/10 rounded-md",
            onClick: () => n(!1),
            children: /* @__PURE__ */ a.jsx(zo, { size: 16 })
          }
        )
      ]
    }
  );
};
function al(e) {
  if (typeof document > "u") return;
  let t = document.head || document.getElementsByTagName("head")[0], n = document.createElement("style");
  n.type = "text/css", t.appendChild(n), n.styleSheet ? n.styleSheet.cssText = e : n.appendChild(document.createTextNode(e));
}
const oa = p.createContext({
  drawerRef: {
    current: null
  },
  overlayRef: {
    current: null
  },
  onPress: () => {
  },
  onRelease: () => {
  },
  onDrag: () => {
  },
  onNestedDrag: () => {
  },
  onNestedOpenChange: () => {
  },
  onNestedRelease: () => {
  },
  openProp: void 0,
  dismissible: !1,
  isOpen: !1,
  isDragging: !1,
  keyboardIsOpen: {
    current: !1
  },
  snapPointsOffset: null,
  snapPoints: null,
  handleOnly: !1,
  modal: !1,
  shouldFade: !1,
  activeSnapPoint: null,
  onOpenChange: () => {
  },
  setActiveSnapPoint: () => {
  },
  closeDrawer: () => {
  },
  direction: "bottom",
  shouldAnimate: {
    current: !0
  },
  shouldScaleBackground: !1,
  setBackgroundColorOnScale: !0,
  noBodyStyles: !1,
  container: null,
  autoFocus: !1
}), Qe = () => {
  const e = p.useContext(oa);
  if (!e)
    throw new Error("useDrawerContext must be used within a Drawer.Root");
  return e;
};
al(`[data-vaul-drawer]{touch-action:none;will-change:transform;transition:transform .5s cubic-bezier(.32, .72, 0, 1);animation-duration:.5s;animation-timing-function:cubic-bezier(0.32,0.72,0,1)}[data-vaul-drawer][data-vaul-snap-points=false][data-vaul-drawer-direction=bottom][data-state=open]{animation-name:slideFromBottom}[data-vaul-drawer][data-vaul-snap-points=false][data-vaul-drawer-direction=bottom][data-state=closed]{animation-name:slideToBottom}[data-vaul-drawer][data-vaul-snap-points=false][data-vaul-drawer-direction=top][data-state=open]{animation-name:slideFromTop}[data-vaul-drawer][data-vaul-snap-points=false][data-vaul-drawer-direction=top][data-state=closed]{animation-name:slideToTop}[data-vaul-drawer][data-vaul-snap-points=false][data-vaul-drawer-direction=left][data-state=open]{animation-name:slideFromLeft}[data-vaul-drawer][data-vaul-snap-points=false][data-vaul-drawer-direction=left][data-state=closed]{animation-name:slideToLeft}[data-vaul-drawer][data-vaul-snap-points=false][data-vaul-drawer-direction=right][data-state=open]{animation-name:slideFromRight}[data-vaul-drawer][data-vaul-snap-points=false][data-vaul-drawer-direction=right][data-state=closed]{animation-name:slideToRight}[data-vaul-drawer][data-vaul-snap-points=true][data-vaul-drawer-direction=bottom]{transform:translate3d(0,var(--initial-transform,100%),0)}[data-vaul-drawer][data-vaul-snap-points=true][data-vaul-drawer-direction=top]{transform:translate3d(0,calc(var(--initial-transform,100%) * -1),0)}[data-vaul-drawer][data-vaul-snap-points=true][data-vaul-drawer-direction=left]{transform:translate3d(calc(var(--initial-transform,100%) * -1),0,0)}[data-vaul-drawer][data-vaul-snap-points=true][data-vaul-drawer-direction=right]{transform:translate3d(var(--initial-transform,100%),0,0)}[data-vaul-drawer][data-vaul-delayed-snap-points=true][data-vaul-drawer-direction=top]{transform:translate3d(0,var(--snap-point-height,0),0)}[data-vaul-drawer][data-vaul-delayed-snap-points=true][data-vaul-drawer-direction=bottom]{transform:translate3d(0,var(--snap-point-height,0),0)}[data-vaul-drawer][data-vaul-delayed-snap-points=true][data-vaul-drawer-direction=left]{transform:translate3d(var(--snap-point-height,0),0,0)}[data-vaul-drawer][data-vaul-delayed-snap-points=true][data-vaul-drawer-direction=right]{transform:translate3d(var(--snap-point-height,0),0,0)}[data-vaul-overlay][data-vaul-snap-points=false]{animation-duration:.5s;animation-timing-function:cubic-bezier(0.32,0.72,0,1)}[data-vaul-overlay][data-vaul-snap-points=false][data-state=open]{animation-name:fadeIn}[data-vaul-overlay][data-state=closed]{animation-name:fadeOut}[data-vaul-animate=false]{animation:none!important}[data-vaul-overlay][data-vaul-snap-points=true]{opacity:0;transition:opacity .5s cubic-bezier(.32, .72, 0, 1)}[data-vaul-overlay][data-vaul-snap-points=true]{opacity:1}[data-vaul-drawer]:not([data-vaul-custom-container=true])::after{content:'';position:absolute;background:inherit;background-color:inherit}[data-vaul-drawer][data-vaul-drawer-direction=top]::after{top:initial;bottom:100%;left:0;right:0;height:200%}[data-vaul-drawer][data-vaul-drawer-direction=bottom]::after{top:100%;bottom:initial;left:0;right:0;height:200%}[data-vaul-drawer][data-vaul-drawer-direction=left]::after{left:initial;right:100%;top:0;bottom:0;width:200%}[data-vaul-drawer][data-vaul-drawer-direction=right]::after{left:100%;right:initial;top:0;bottom:0;width:200%}[data-vaul-overlay][data-vaul-snap-points=true]:not([data-vaul-snap-points-overlay=true]):not(
[data-state=closed]
){opacity:0}[data-vaul-overlay][data-vaul-snap-points-overlay=true]{opacity:1}[data-vaul-handle]{display:block;position:relative;opacity:.7;background:#e2e2e4;margin-left:auto;margin-right:auto;height:5px;width:32px;border-radius:1rem;touch-action:pan-y}[data-vaul-handle]:active,[data-vaul-handle]:hover{opacity:1}[data-vaul-handle-hitarea]{position:absolute;left:50%;top:50%;transform:translate(-50%,-50%);width:max(100%,2.75rem);height:max(100%,2.75rem);touch-action:inherit}@media (hover:hover) and (pointer:fine){[data-vaul-drawer]{user-select:none}}@media (pointer:fine){[data-vaul-handle-hitarea]:{width:100%;height:100%}}@keyframes fadeIn{from{opacity:0}to{opacity:1}}@keyframes fadeOut{to{opacity:0}}@keyframes slideFromBottom{from{transform:translate3d(0,var(--initial-transform,100%),0)}to{transform:translate3d(0,0,0)}}@keyframes slideToBottom{to{transform:translate3d(0,var(--initial-transform,100%),0)}}@keyframes slideFromTop{from{transform:translate3d(0,calc(var(--initial-transform,100%) * -1),0)}to{transform:translate3d(0,0,0)}}@keyframes slideToTop{to{transform:translate3d(0,calc(var(--initial-transform,100%) * -1),0)}}@keyframes slideFromLeft{from{transform:translate3d(calc(var(--initial-transform,100%) * -1),0,0)}to{transform:translate3d(0,0,0)}}@keyframes slideToLeft{to{transform:translate3d(calc(var(--initial-transform,100%) * -1),0,0)}}@keyframes slideFromRight{from{transform:translate3d(var(--initial-transform,100%),0,0)}to{transform:translate3d(0,0,0)}}@keyframes slideToRight{to{transform:translate3d(var(--initial-transform,100%),0,0)}}`);
function ol() {
  const e = navigator.userAgent;
  return typeof window < "u" && (/Firefox/.test(e) && /Mobile/.test(e) || // Android Firefox
  /FxiOS/.test(e));
}
function sl() {
  return Jt(/^Mac/);
}
function il() {
  return Jt(/^iPhone/);
}
function En() {
  return /^((?!chrome|android).)*safari/i.test(navigator.userAgent);
}
function ll() {
  return Jt(/^iPad/) || // iPadOS 13 lies and says it's a Mac, but we can distinguish by detecting touch support.
  sl() && navigator.maxTouchPoints > 1;
}
function sa() {
  return il() || ll();
}
function Jt(e) {
  return typeof window < "u" && window.navigator != null ? e.test(window.navigator.platform) : void 0;
}
const cl = 24, ul = typeof window < "u" ? So : Ce;
function Dn(...e) {
  return (...t) => {
    for (let n of e)
      typeof n == "function" && n(...t);
  };
}
const Mt = typeof document < "u" && window.visualViewport;
function Pn(e) {
  let t = window.getComputedStyle(e);
  return /(auto|scroll)/.test(t.overflow + t.overflowX + t.overflowY);
}
function ia(e) {
  for (Pn(e) && (e = e.parentElement); e && !Pn(e); )
    e = e.parentElement;
  return e || document.scrollingElement || document.documentElement;
}
const dl = /* @__PURE__ */ new Set([
  "checkbox",
  "radio",
  "range",
  "color",
  "file",
  "image",
  "button",
  "submit",
  "reset"
]);
let ht = 0, Rt;
function fl(e = {}) {
  let { isDisabled: t } = e;
  ul(() => {
    if (!t)
      return ht++, ht === 1 && sa() && (Rt = hl()), () => {
        ht--, ht === 0 && (Rt == null || Rt());
      };
  }, [
    t
  ]);
}
function hl() {
  let e, t = 0, n = (c) => {
    e = ia(c.target), !(e === document.documentElement && e === document.body) && (t = c.changedTouches[0].pageY);
  }, r = (c) => {
    if (!e || e === document.documentElement || e === document.body) {
      c.preventDefault();
      return;
    }
    let d = c.changedTouches[0].pageY, v = e.scrollTop, M = e.scrollHeight - e.clientHeight;
    M !== 0 && ((v <= 0 && d > t || v >= M && d < t) && c.preventDefault(), t = d);
  }, o = (c) => {
    let d = c.target;
    Tt(d) && d !== document.activeElement && (c.preventDefault(), d.style.transform = "translateY(-2000px)", d.focus(), requestAnimationFrame(() => {
      d.style.transform = "";
    }));
  }, s = (c) => {
    let d = c.target;
    Tt(d) && (d.style.transform = "translateY(-2000px)", requestAnimationFrame(() => {
      d.style.transform = "", Mt && (Mt.height < window.innerHeight ? requestAnimationFrame(() => {
        jn(d);
      }) : Mt.addEventListener("resize", () => jn(d), {
        once: !0
      }));
    }));
  }, i = () => {
    window.scrollTo(0, 0);
  }, l = window.pageXOffset, u = window.pageYOffset, m = Dn(pl(document.documentElement, "paddingRight", `${window.innerWidth - document.documentElement.clientWidth}px`));
  window.scrollTo(0, 0);
  let f = Dn(We(document, "touchstart", n, {
    passive: !1,
    capture: !0
  }), We(document, "touchmove", r, {
    passive: !1,
    capture: !0
  }), We(document, "touchend", o, {
    passive: !1,
    capture: !0
  }), We(document, "focus", s, !0), We(window, "scroll", i));
  return () => {
    m(), f(), window.scrollTo(l, u);
  };
}
function pl(e, t, n) {
  let r = e.style[t];
  return e.style[t] = n, () => {
    e.style[t] = r;
  };
}
function We(e, t, n, r) {
  return e.addEventListener(t, n, r), () => {
    e.removeEventListener(t, n, r);
  };
}
function jn(e) {
  let t = document.scrollingElement || document.documentElement;
  for (; e && e !== t; ) {
    let n = ia(e);
    if (n !== document.documentElement && n !== document.body && n !== e) {
      let r = n.getBoundingClientRect().top, o = e.getBoundingClientRect().top, s = e.getBoundingClientRect().bottom;
      const i = n.getBoundingClientRect().bottom + cl;
      s > i && (n.scrollTop += o - r);
    }
    e = n.parentElement;
  }
}
function Tt(e) {
  return e instanceof HTMLInputElement && !dl.has(e.type) || e instanceof HTMLTextAreaElement || e instanceof HTMLElement && e.isContentEditable;
}
function ml(e, t) {
  typeof e == "function" ? e(t) : e != null && (e.current = t);
}
function gl(...e) {
  return (t) => e.forEach((n) => ml(n, t));
}
function la(...e) {
  return h.useCallback(gl(...e), e);
}
const ca = /* @__PURE__ */ new WeakMap();
function H(e, t, n = !1) {
  if (!e || !(e instanceof HTMLElement)) return;
  let r = {};
  Object.entries(t).forEach(([o, s]) => {
    if (o.startsWith("--")) {
      e.style.setProperty(o, s);
      return;
    }
    r[o] = e.style[o], e.style[o] = s;
  }), !n && ca.set(e, r);
}
function vl(e, t) {
  if (!e || !(e instanceof HTMLElement)) return;
  let n = ca.get(e);
  n && (e.style[t] = n[t]);
}
const U = (e) => {
  switch (e) {
    case "top":
    case "bottom":
      return !0;
    case "left":
    case "right":
      return !1;
    default:
      return e;
  }
};
function pt(e, t) {
  if (!e)
    return null;
  const n = window.getComputedStyle(e), r = (
    // @ts-ignore
    n.transform || n.webkitTransform || n.mozTransform
  );
  let o = r.match(/^matrix3d\((.+)\)$/);
  return o ? parseFloat(o[1].split(", ")[U(t) ? 13 : 12]) : (o = r.match(/^matrix\((.+)\)$/), o ? parseFloat(o[1].split(", ")[U(t) ? 5 : 4]) : null);
}
function wl(e) {
  return 8 * (Math.log(e + 1) - 2);
}
function Et(e, t) {
  if (!e) return () => {
  };
  const n = e.style.cssText;
  return Object.assign(e.style, t), () => {
    e.style.cssText = n;
  };
}
function xl(...e) {
  return (...t) => {
    for (const n of e)
      typeof n == "function" && n(...t);
  };
}
const $ = {
  DURATION: 0.5,
  EASE: [
    0.32,
    0.72,
    0,
    1
  ]
}, ua = 0.4, yl = 0.25, bl = 100, da = 8, Oe = 16, _t = 26, Dt = "vaul-dragging";
function fa(e) {
  const t = p.useRef(e);
  return p.useEffect(() => {
    t.current = e;
  }), p.useMemo(() => (...n) => t.current == null ? void 0 : t.current.call(t, ...n), []);
}
function Sl({ defaultProp: e, onChange: t }) {
  const n = p.useState(e), [r] = n, o = p.useRef(r), s = fa(t);
  return p.useEffect(() => {
    o.current !== r && (s(r), o.current = r);
  }, [
    r,
    o,
    s
  ]), n;
}
function ha({ prop: e, defaultProp: t, onChange: n = () => {
} }) {
  const [r, o] = Sl({
    defaultProp: t,
    onChange: n
  }), s = e !== void 0, i = s ? e : r, l = fa(n), u = p.useCallback((m) => {
    if (s) {
      const c = typeof m == "function" ? m(e) : m;
      c !== e && l(c);
    } else
      o(m);
  }, [
    s,
    e,
    o,
    l
  ]);
  return [
    i,
    u
  ];
}
function Cl({ activeSnapPointProp: e, setActiveSnapPointProp: t, snapPoints: n, drawerRef: r, overlayRef: o, fadeFromIndex: s, onSnapPointChange: i, direction: l = "bottom", container: u, snapToSequentialPoint: m }) {
  const [f, c] = ha({
    prop: e,
    defaultProp: n == null ? void 0 : n[0],
    onChange: t
  }), [d, v] = p.useState(typeof window < "u" ? {
    innerWidth: window.innerWidth,
    innerHeight: window.innerHeight
  } : void 0);
  p.useEffect(() => {
    function b() {
      v({
        innerWidth: window.innerWidth,
        innerHeight: window.innerHeight
      });
    }
    return window.addEventListener("resize", b), () => window.removeEventListener("resize", b);
  }, []);
  const M = p.useMemo(() => f === (n == null ? void 0 : n[n.length - 1]) || null, [
    n,
    f
  ]), C = p.useMemo(() => {
    var b;
    return (b = n == null ? void 0 : n.findIndex((j) => j === f)) != null ? b : null;
  }, [
    n,
    f
  ]), _ = n && n.length > 0 && (s || s === 0) && !Number.isNaN(s) && n[s] === f || !n, R = p.useMemo(() => {
    const b = u ? {
      width: u.getBoundingClientRect().width,
      height: u.getBoundingClientRect().height
    } : typeof window < "u" ? {
      width: window.innerWidth,
      height: window.innerHeight
    } : {
      width: 0,
      height: 0
    };
    var j;
    return (j = n == null ? void 0 : n.map((D) => {
      const B = typeof D == "string";
      let K = 0;
      if (B && (K = parseInt(D, 10)), U(l)) {
        const x = B ? K : d ? D * b.height : 0;
        return d ? l === "bottom" ? b.height - x : -b.height + x : x;
      }
      const W = B ? K : d ? D * b.width : 0;
      return d ? l === "right" ? b.width - W : -b.width + W : W;
    })) != null ? j : [];
  }, [
    n,
    d,
    u
  ]), k = p.useMemo(() => C !== null ? R == null ? void 0 : R[C] : null, [
    R,
    C
  ]), P = p.useCallback((b) => {
    var j;
    const D = (j = R == null ? void 0 : R.findIndex((B) => B === b)) != null ? j : null;
    i(D), H(r.current, {
      transition: `transform ${$.DURATION}s cubic-bezier(${$.EASE.join(",")})`,
      transform: U(l) ? `translate3d(0, ${b}px, 0)` : `translate3d(${b}px, 0, 0)`
    }), R && D !== R.length - 1 && s !== void 0 && D !== s && D < s ? H(o.current, {
      transition: `opacity ${$.DURATION}s cubic-bezier(${$.EASE.join(",")})`,
      opacity: "0"
    }) : H(o.current, {
      transition: `opacity ${$.DURATION}s cubic-bezier(${$.EASE.join(",")})`,
      opacity: "1"
    }), c(n == null ? void 0 : n[Math.max(D, 0)]);
  }, [
    r.current,
    n,
    R,
    s,
    o,
    c
  ]);
  p.useEffect(() => {
    if (f || e) {
      var b;
      const j = (b = n == null ? void 0 : n.findIndex((D) => D === e || D === f)) != null ? b : -1;
      R && j !== -1 && typeof R[j] == "number" && P(R[j]);
    }
  }, [
    f,
    e,
    n,
    R,
    P
  ]);
  function g({ draggedDistance: b, closeDrawer: j, velocity: D, dismissible: B }) {
    if (s === void 0) return;
    const K = l === "bottom" || l === "right" ? (k ?? 0) - b : (k ?? 0) + b, W = C === s - 1, x = C === 0, V = b > 0;
    if (W && H(o.current, {
      transition: `opacity ${$.DURATION}s cubic-bezier(${$.EASE.join(",")})`
    }), !m && D > 2 && !V) {
      B ? j() : P(R[0]);
      return;
    }
    if (!m && D > 2 && V && R && n) {
      P(R[n.length - 1]);
      return;
    }
    const G = R == null ? void 0 : R.reduce((q, Y) => typeof q != "number" || typeof Y != "number" ? q : Math.abs(Y - K) < Math.abs(q - K) ? Y : q), te = U(l) ? window.innerHeight : window.innerWidth;
    if (D > ua && Math.abs(b) < te * 0.4) {
      const q = V ? 1 : -1;
      if (q > 0 && M && n) {
        P(R[n.length - 1]);
        return;
      }
      if (x && q < 0 && B && j(), C === null) return;
      P(R[C + q]);
      return;
    }
    P(G);
  }
  function z({ draggedDistance: b }) {
    if (k === null) return;
    const j = l === "bottom" || l === "right" ? k - b : k + b;
    (l === "bottom" || l === "right") && j < R[R.length - 1] || (l === "top" || l === "left") && j > R[R.length - 1] || H(r.current, {
      transform: U(l) ? `translate3d(0, ${j}px, 0)` : `translate3d(${j}px, 0, 0)`
    });
  }
  function Z(b, j) {
    if (!n || typeof C != "number" || !R || s === void 0) return null;
    const D = C === s - 1;
    if (C >= s && j)
      return 0;
    if (D && !j) return 1;
    if (!_ && !D) return null;
    const K = D ? C + 1 : C - 1, W = D ? R[K] - R[K - 1] : R[K + 1] - R[K], x = b / Math.abs(W);
    return D ? 1 - x : x;
  }
  return {
    isLastSnapPoint: M,
    activeSnapPoint: f,
    shouldFade: _,
    getPercentageDragged: Z,
    setActiveSnapPoint: c,
    activeSnapPointIndex: C,
    onRelease: g,
    onDrag: z,
    snapPointsOffset: R
  };
}
const Ml = () => () => {
};
function Rl() {
  const { direction: e, isOpen: t, shouldScaleBackground: n, setBackgroundColorOnScale: r, noBodyStyles: o } = Qe(), s = p.useRef(null), i = mt(() => document.body.style.backgroundColor, []);
  function l() {
    return (window.innerWidth - _t) / window.innerWidth;
  }
  p.useEffect(() => {
    if (t && n) {
      s.current && clearTimeout(s.current);
      const u = document.querySelector("[data-vaul-drawer-wrapper]") || document.querySelector("[vaul-drawer-wrapper]");
      if (!u) return;
      xl(r && !o ? Et(document.body, {
        background: "black"
      }) : Ml, Et(u, {
        transformOrigin: U(e) ? "top" : "left",
        transitionProperty: "transform, border-radius",
        transitionDuration: `${$.DURATION}s`,
        transitionTimingFunction: `cubic-bezier(${$.EASE.join(",")})`
      }));
      const m = Et(u, {
        borderRadius: `${da}px`,
        overflow: "hidden",
        ...U(e) ? {
          transform: `scale(${l()}) translate3d(0, calc(env(safe-area-inset-top) + 14px), 0)`
        } : {
          transform: `scale(${l()}) translate3d(calc(env(safe-area-inset-top) + 14px), 0, 0)`
        }
      });
      return () => {
        m(), s.current = window.setTimeout(() => {
          i ? document.body.style.background = i : document.body.style.removeProperty("background");
        }, $.DURATION * 1e3);
      };
    }
  }, [
    t,
    n,
    i
  ]);
}
let Ye = null;
function El({ isOpen: e, modal: t, nested: n, hasBeenOpened: r, preventScrollRestoration: o, noBodyStyles: s }) {
  const [i, l] = p.useState(() => typeof window < "u" ? window.location.href : ""), u = p.useRef(0), m = p.useCallback(() => {
    if (En() && Ye === null && e && !s) {
      Ye = {
        position: document.body.style.position,
        top: document.body.style.top,
        left: document.body.style.left,
        height: document.body.style.height,
        right: "unset"
      };
      const { scrollX: c, innerHeight: d } = window;
      document.body.style.setProperty("position", "fixed", "important"), Object.assign(document.body.style, {
        top: `${-u.current}px`,
        left: `${-c}px`,
        right: "0px",
        height: "auto"
      }), window.setTimeout(() => window.requestAnimationFrame(() => {
        const v = d - window.innerHeight;
        v && u.current >= d && (document.body.style.top = `${-(u.current + v)}px`);
      }), 300);
    }
  }, [
    e
  ]), f = p.useCallback(() => {
    if (En() && Ye !== null && !s) {
      const c = -parseInt(document.body.style.top, 10), d = -parseInt(document.body.style.left, 10);
      Object.assign(document.body.style, Ye), window.requestAnimationFrame(() => {
        if (o && i !== window.location.href) {
          l(window.location.href);
          return;
        }
        window.scrollTo(d, c);
      }), Ye = null;
    }
  }, [
    i
  ]);
  return p.useEffect(() => {
    function c() {
      u.current = window.scrollY;
    }
    return c(), window.addEventListener("scroll", c), () => {
      window.removeEventListener("scroll", c);
    };
  }, []), p.useEffect(() => {
    if (t)
      return () => {
        typeof document > "u" || document.querySelector("[data-vaul-drawer]") || f();
      };
  }, [
    t,
    f
  ]), p.useEffect(() => {
    n || !r || (e ? (!window.matchMedia("(display-mode: standalone)").matches && m(), t || window.setTimeout(() => {
      f();
    }, 500)) : f());
  }, [
    e,
    r,
    i,
    t,
    n,
    m,
    f
  ]), {
    restorePositionSetting: f
  };
}
function pa({ open: e, onOpenChange: t, children: n, onDrag: r, onRelease: o, snapPoints: s, shouldScaleBackground: i = !1, setBackgroundColorOnScale: l = !0, closeThreshold: u = yl, scrollLockTimeout: m = bl, dismissible: f = !0, handleOnly: c = !1, fadeFromIndex: d = s && s.length - 1, activeSnapPoint: v, setActiveSnapPoint: M, fixed: C, modal: _ = !0, onClose: R, nested: k, noBodyStyles: P = !1, direction: g = "bottom", defaultOpen: z = !1, disablePreventScroll: Z = !0, snapToSequentialPoint: b = !1, preventScrollRestoration: j = !1, repositionInputs: D = !0, onAnimationEnd: B, container: K, autoFocus: W = !1 }) {
  var x, V;
  const [G = !1, te] = ha({
    defaultProp: z,
    prop: e,
    onChange: (S) => {
      t == null || t(S), !S && !k && _a(), setTimeout(() => {
        B == null || B(S);
      }, $.DURATION * 1e3), S && !_ && typeof window < "u" && window.requestAnimationFrame(() => {
        document.body.style.pointerEvents = "auto";
      }), S || (document.body.style.pointerEvents = "auto");
    }
  }), [q, Y] = p.useState(!1), [y, Q] = p.useState(!1), [ie, de] = p.useState(!1), pe = p.useRef(null), Me = p.useRef(null), ke = p.useRef(null), ye = p.useRef(null), le = p.useRef(null), Re = p.useRef(!1), re = p.useRef(null), $e = p.useRef(0), Fe = p.useRef(!1), nn = p.useRef(!z), rn = p.useRef(0), E = p.useRef(null), an = p.useRef(((x = E.current) == null ? void 0 : x.getBoundingClientRect().height) || 0), on = p.useRef(((V = E.current) == null ? void 0 : V.getBoundingClientRect().width) || 0), bt = p.useRef(0), Da = p.useCallback((S) => {
    s && S === Ve.length - 1 && (Me.current = /* @__PURE__ */ new Date());
  }, []), { activeSnapPoint: Pa, activeSnapPointIndex: Le, setActiveSnapPoint: sn, onRelease: ja, snapPointsOffset: Ve, onDrag: Na, shouldFade: ln, getPercentageDragged: Ta } = Cl({
    snapPoints: s,
    activeSnapPointProp: v,
    setActiveSnapPointProp: M,
    drawerRef: E,
    fadeFromIndex: d,
    overlayRef: pe,
    onSnapPointChange: Da,
    direction: g,
    container: K,
    snapToSequentialPoint: b
  });
  fl({
    isDisabled: !G || y || !_ || ie || !q || !D || !Z
  });
  const { restorePositionSetting: _a } = El({
    isOpen: G,
    modal: _,
    nested: k ?? !1,
    hasBeenOpened: q,
    preventScrollRestoration: j,
    noBodyStyles: P
  });
  function lt() {
    return (window.innerWidth - _t) / window.innerWidth;
  }
  function Oa(S) {
    var T, I;
    !f && !s || E.current && !E.current.contains(S.target) || (an.current = ((T = E.current) == null ? void 0 : T.getBoundingClientRect().height) || 0, on.current = ((I = E.current) == null ? void 0 : I.getBoundingClientRect().width) || 0, Q(!0), ke.current = /* @__PURE__ */ new Date(), sa() && window.addEventListener("touchend", () => Re.current = !1, {
      once: !0
    }), S.target.setPointerCapture(S.pointerId), $e.current = U(g) ? S.pageY : S.pageX);
  }
  function cn(S, T) {
    var I;
    let N = S;
    const L = (I = window.getSelection()) == null ? void 0 : I.toString(), ne = E.current ? pt(E.current, g) : null, ee = /* @__PURE__ */ new Date();
    if (N.tagName === "SELECT" || N.hasAttribute("data-vaul-no-drag") || N.closest("[data-vaul-no-drag]"))
      return !1;
    if (g === "right" || g === "left")
      return !0;
    if (Me.current && ee.getTime() - Me.current.getTime() < 500)
      return !1;
    if (ne !== null && (g === "bottom" ? ne > 0 : ne < 0))
      return !0;
    if (L && L.length > 0)
      return !1;
    if (le.current && ee.getTime() - le.current.getTime() < m && ne === 0 || T)
      return le.current = ee, !1;
    for (; N; ) {
      if (N.scrollHeight > N.clientHeight) {
        if (N.scrollTop !== 0)
          return le.current = /* @__PURE__ */ new Date(), !1;
        if (N.getAttribute("role") === "dialog")
          return !0;
      }
      N = N.parentNode;
    }
    return !0;
  }
  function Ia(S) {
    if (E.current && y) {
      const T = g === "bottom" || g === "right" ? 1 : -1, I = ($e.current - (U(g) ? S.pageY : S.pageX)) * T, N = I > 0, L = s && !f && !N;
      if (L && Le === 0) return;
      const ne = Math.abs(I), ee = document.querySelector("[data-vaul-drawer-wrapper]"), Ee = g === "bottom" || g === "top" ? an.current : on.current;
      let fe = ne / Ee;
      const _e = Ta(ne, N);
      if (_e !== null && (fe = _e), L && fe >= 1 || !Re.current && !cn(S.target, N)) return;
      if (E.current.classList.add(Dt), Re.current = !0, H(E.current, {
        transition: "none"
      }), H(pe.current, {
        transition: "none"
      }), s && Na({
        draggedDistance: I
      }), N && !s) {
        const me = wl(I), ct = Math.min(me * -1, 0) * T;
        H(E.current, {
          transform: U(g) ? `translate3d(0, ${ct}px, 0)` : `translate3d(${ct}px, 0, 0)`
        });
        return;
      }
      const De = 1 - fe;
      if ((ln || d && Le === d - 1) && (r == null || r(S, fe), H(pe.current, {
        opacity: `${De}`,
        transition: "none"
      }, !0)), ee && pe.current && i) {
        const me = Math.min(lt() + fe * (1 - lt()), 1), ct = 8 - fe * 8, dn = Math.max(0, 14 - fe * 14);
        H(ee, {
          borderRadius: `${ct}px`,
          transform: U(g) ? `scale(${me}) translate3d(0, ${dn}px, 0)` : `scale(${me}) translate3d(${dn}px, 0, 0)`,
          transition: "none"
        }, !0);
      }
      if (!s) {
        const me = ne * T;
        H(E.current, {
          transform: U(g) ? `translate3d(0, ${me}px, 0)` : `translate3d(${me}px, 0, 0)`
        });
      }
    }
  }
  p.useEffect(() => {
    window.requestAnimationFrame(() => {
      nn.current = !0;
    });
  }, []), p.useEffect(() => {
    var S;
    function T() {
      if (!E.current || !D) return;
      const I = document.activeElement;
      if (Tt(I) || Fe.current) {
        var N;
        const L = ((N = window.visualViewport) == null ? void 0 : N.height) || 0, ne = window.innerHeight;
        let ee = ne - L;
        const Ee = E.current.getBoundingClientRect().height || 0, fe = Ee > ne * 0.8;
        bt.current || (bt.current = Ee);
        const _e = E.current.getBoundingClientRect().top;
        if (Math.abs(rn.current - ee) > 60 && (Fe.current = !Fe.current), s && s.length > 0 && Ve && Le) {
          const De = Ve[Le] || 0;
          ee += De;
        }
        if (rn.current = ee, Ee > L || Fe.current) {
          const De = E.current.getBoundingClientRect().height;
          let me = De;
          De > L && (me = L - (fe ? _e : _t)), C ? E.current.style.height = `${De - Math.max(ee, 0)}px` : E.current.style.height = `${Math.max(me, L - _e)}px`;
        } else ol() || (E.current.style.height = `${bt.current}px`);
        s && s.length > 0 && !Fe.current ? E.current.style.bottom = "0px" : E.current.style.bottom = `${Math.max(ee, 0)}px`;
      }
    }
    return (S = window.visualViewport) == null || S.addEventListener("resize", T), () => {
      var I;
      return (I = window.visualViewport) == null ? void 0 : I.removeEventListener("resize", T);
    };
  }, [
    Le,
    s,
    Ve
  ]);
  function Ge(S) {
    Aa(), R == null || R(), S || te(!1), setTimeout(() => {
      s && sn(s[0]);
    }, $.DURATION * 1e3);
  }
  function un() {
    if (!E.current) return;
    const S = document.querySelector("[data-vaul-drawer-wrapper]"), T = pt(E.current, g);
    H(E.current, {
      transform: "translate3d(0, 0, 0)",
      transition: `transform ${$.DURATION}s cubic-bezier(${$.EASE.join(",")})`
    }), H(pe.current, {
      transition: `opacity ${$.DURATION}s cubic-bezier(${$.EASE.join(",")})`,
      opacity: "1"
    }), i && T && T > 0 && G && H(S, {
      borderRadius: `${da}px`,
      overflow: "hidden",
      ...U(g) ? {
        transform: `scale(${lt()}) translate3d(0, calc(env(safe-area-inset-top) + 14px), 0)`,
        transformOrigin: "top"
      } : {
        transform: `scale(${lt()}) translate3d(calc(env(safe-area-inset-top) + 14px), 0, 0)`,
        transformOrigin: "left"
      },
      transitionProperty: "transform, border-radius",
      transitionDuration: `${$.DURATION}s`,
      transitionTimingFunction: `cubic-bezier(${$.EASE.join(",")})`
    }, !0);
  }
  function Aa() {
    !y || !E.current || (E.current.classList.remove(Dt), Re.current = !1, Q(!1), ye.current = /* @__PURE__ */ new Date());
  }
  function ka(S) {
    if (!y || !E.current) return;
    E.current.classList.remove(Dt), Re.current = !1, Q(!1), ye.current = /* @__PURE__ */ new Date();
    const T = pt(E.current, g);
    if (!S || !cn(S.target, !1) || !T || Number.isNaN(T) || ke.current === null) return;
    const I = ye.current.getTime() - ke.current.getTime(), N = $e.current - (U(g) ? S.pageY : S.pageX), L = Math.abs(N) / I;
    if (L > 0.05 && (de(!0), setTimeout(() => {
      de(!1);
    }, 200)), s) {
      ja({
        draggedDistance: N * (g === "bottom" || g === "right" ? 1 : -1),
        closeDrawer: Ge,
        velocity: L,
        dismissible: f
      }), o == null || o(S, !0);
      return;
    }
    if (g === "bottom" || g === "right" ? N > 0 : N < 0) {
      un(), o == null || o(S, !0);
      return;
    }
    if (L > ua) {
      Ge(), o == null || o(S, !1);
      return;
    }
    var ne;
    const ee = Math.min((ne = E.current.getBoundingClientRect().height) != null ? ne : 0, window.innerHeight);
    var Ee;
    const fe = Math.min((Ee = E.current.getBoundingClientRect().width) != null ? Ee : 0, window.innerWidth), _e = g === "left" || g === "right";
    if (Math.abs(T) >= (_e ? fe : ee) * u) {
      Ge(), o == null || o(S, !1);
      return;
    }
    o == null || o(S, !0), un();
  }
  p.useEffect(() => (G && (H(document.documentElement, {
    scrollBehavior: "auto"
  }), Me.current = /* @__PURE__ */ new Date()), () => {
    vl(document.documentElement, "scrollBehavior");
  }), [
    G
  ]);
  function $a(S) {
    const T = S ? (window.innerWidth - Oe) / window.innerWidth : 1, I = S ? -Oe : 0;
    re.current && window.clearTimeout(re.current), H(E.current, {
      transition: `transform ${$.DURATION}s cubic-bezier(${$.EASE.join(",")})`,
      transform: `scale(${T}) translate3d(0, ${I}px, 0)`
    }), !S && E.current && (re.current = setTimeout(() => {
      const N = pt(E.current, g);
      H(E.current, {
        transition: "none",
        transform: U(g) ? `translate3d(0, ${N}px, 0)` : `translate3d(${N}px, 0, 0)`
      });
    }, 500));
  }
  function Fa(S, T) {
    if (T < 0) return;
    const I = (window.innerWidth - Oe) / window.innerWidth, N = I + T * (1 - I), L = -Oe + T * Oe;
    H(E.current, {
      transform: U(g) ? `scale(${N}) translate3d(0, ${L}px, 0)` : `scale(${N}) translate3d(${L}px, 0, 0)`,
      transition: "none"
    });
  }
  function La(S, T) {
    const I = U(g) ? window.innerHeight : window.innerWidth, N = T ? (I - Oe) / I : 1, L = T ? -Oe : 0;
    T && H(E.current, {
      transition: `transform ${$.DURATION}s cubic-bezier(${$.EASE.join(",")})`,
      transform: U(g) ? `scale(${N}) translate3d(0, ${L}px, 0)` : `scale(${N}) translate3d(${L}px, 0, 0)`
    });
  }
  return p.useEffect(() => {
    _ || window.requestAnimationFrame(() => {
      document.body.style.pointerEvents = "auto";
    });
  }, [
    _
  ]), /* @__PURE__ */ p.createElement(as, {
    defaultOpen: z,
    onOpenChange: (S) => {
      !f && !S || (S ? Y(!0) : Ge(!0), te(S));
    },
    open: G
  }, /* @__PURE__ */ p.createElement(oa.Provider, {
    value: {
      activeSnapPoint: Pa,
      snapPoints: s,
      setActiveSnapPoint: sn,
      drawerRef: E,
      overlayRef: pe,
      onOpenChange: t,
      onPress: Oa,
      onRelease: ka,
      onDrag: Ia,
      dismissible: f,
      shouldAnimate: nn,
      handleOnly: c,
      isOpen: G,
      isDragging: y,
      shouldFade: ln,
      closeDrawer: Ge,
      onNestedDrag: Fa,
      onNestedOpenChange: $a,
      onNestedRelease: La,
      keyboardIsOpen: Fe,
      modal: _,
      snapPointsOffset: Ve,
      activeSnapPointIndex: Le,
      direction: g,
      shouldScaleBackground: i,
      setBackgroundColorOnScale: l,
      noBodyStyles: P,
      container: K,
      autoFocus: W
    }
  }, n));
}
const ma = /* @__PURE__ */ p.forwardRef(function({ ...e }, t) {
  const { overlayRef: n, snapPoints: r, onRelease: o, shouldFade: s, isOpen: i, modal: l, shouldAnimate: u } = Qe(), m = la(t, n), f = r && r.length > 0;
  if (!l)
    return null;
  const c = p.useCallback((d) => o(d), [
    o
  ]);
  return /* @__PURE__ */ p.createElement(Jo, {
    onMouseUp: c,
    ref: m,
    "data-vaul-overlay": "",
    "data-vaul-snap-points": i && f ? "true" : "false",
    "data-vaul-snap-points-overlay": i && s ? "true" : "false",
    "data-vaul-animate": u != null && u.current ? "true" : "false",
    ...e
  });
});
ma.displayName = "Drawer.Overlay";
const ga = /* @__PURE__ */ p.forwardRef(function({ onPointerDownOutside: e, style: t, onOpenAutoFocus: n, ...r }, o) {
  const { drawerRef: s, onPress: i, onRelease: l, onDrag: u, keyboardIsOpen: m, snapPointsOffset: f, activeSnapPointIndex: c, modal: d, isOpen: v, direction: M, snapPoints: C, container: _, handleOnly: R, shouldAnimate: k, autoFocus: P } = Qe(), [g, z] = p.useState(!1), Z = la(o, s), b = p.useRef(null), j = p.useRef(null), D = p.useRef(!1), B = C && C.length > 0;
  Rl();
  const K = (x, V, G = 0) => {
    if (D.current) return !0;
    const te = Math.abs(x.y), q = Math.abs(x.x), Y = q > te, y = [
      "bottom",
      "right"
    ].includes(V) ? 1 : -1;
    if (V === "left" || V === "right") {
      if (!(x.x * y < 0) && q >= 0 && q <= G)
        return Y;
    } else if (!(x.y * y < 0) && te >= 0 && te <= G)
      return !Y;
    return D.current = !0, !0;
  };
  p.useEffect(() => {
    B && window.requestAnimationFrame(() => {
      z(!0);
    });
  }, []);
  function W(x) {
    b.current = null, D.current = !1, l(x);
  }
  return /* @__PURE__ */ p.createElement(Zo, {
    "data-vaul-drawer-direction": M,
    "data-vaul-drawer": "",
    "data-vaul-delayed-snap-points": g ? "true" : "false",
    "data-vaul-snap-points": v && B ? "true" : "false",
    "data-vaul-custom-container": _ ? "true" : "false",
    "data-vaul-animate": k != null && k.current ? "true" : "false",
    ...r,
    ref: Z,
    style: f && f.length > 0 ? {
      "--snap-point-height": `${f[c ?? 0]}px`,
      ...t
    } : t,
    onPointerDown: (x) => {
      R || (r.onPointerDown == null || r.onPointerDown.call(r, x), b.current = {
        x: x.pageX,
        y: x.pageY
      }, i(x));
    },
    onOpenAutoFocus: (x) => {
      n == null || n(x), P || x.preventDefault();
    },
    onPointerDownOutside: (x) => {
      if (e == null || e(x), !d || x.defaultPrevented) {
        x.preventDefault();
        return;
      }
      m.current && (m.current = !1);
    },
    onFocusOutside: (x) => {
      if (!d) {
        x.preventDefault();
        return;
      }
    },
    onPointerMove: (x) => {
      if (j.current = x, R || (r.onPointerMove == null || r.onPointerMove.call(r, x), !b.current)) return;
      const V = x.pageY - b.current.y, G = x.pageX - b.current.x, te = x.pointerType === "touch" ? 10 : 2;
      K({
        x: G,
        y: V
      }, M, te) ? u(x) : (Math.abs(G) > te || Math.abs(V) > te) && (b.current = null);
    },
    onPointerUp: (x) => {
      r.onPointerUp == null || r.onPointerUp.call(r, x), b.current = null, D.current = !1, l(x);
    },
    onPointerOut: (x) => {
      r.onPointerOut == null || r.onPointerOut.call(r, x), W(j.current);
    },
    onContextMenu: (x) => {
      r.onContextMenu == null || r.onContextMenu.call(r, x), j.current && W(j.current);
    }
  });
});
ga.displayName = "Drawer.Content";
const Dl = 250, Pl = 120, va = /* @__PURE__ */ p.forwardRef(function({ preventCycle: e = !1, children: t, ...n }, r) {
  const { closeDrawer: o, isDragging: s, snapPoints: i, activeSnapPoint: l, setActiveSnapPoint: u, dismissible: m, handleOnly: f, isOpen: c, onPress: d, onDrag: v } = Qe(), M = p.useRef(null), C = p.useRef(!1);
  function _() {
    if (C.current) {
      P();
      return;
    }
    window.setTimeout(() => {
      R();
    }, Pl);
  }
  function R() {
    if (s || e || C.current) {
      P();
      return;
    }
    if (P(), !i || i.length === 0) {
      m || o();
      return;
    }
    if (l === i[i.length - 1] && m) {
      o();
      return;
    }
    const z = i.findIndex((b) => b === l);
    if (z === -1) return;
    const Z = i[z + 1];
    u(Z);
  }
  function k() {
    M.current = window.setTimeout(() => {
      C.current = !0;
    }, Dl);
  }
  function P() {
    M.current && window.clearTimeout(M.current), C.current = !1;
  }
  return /* @__PURE__ */ p.createElement("div", {
    onClick: _,
    onPointerCancel: P,
    onPointerDown: (g) => {
      f && d(g), k();
    },
    onPointerMove: (g) => {
      f && v(g);
    },
    // onPointerUp is already handled by the content component
    ref: r,
    "data-vaul-drawer-visible": c ? "true" : "false",
    "data-vaul-handle": "",
    "aria-hidden": "true",
    ...n
  }, /* @__PURE__ */ p.createElement("span", {
    "data-vaul-handle-hitarea": "",
    "aria-hidden": "true"
  }, t));
});
va.displayName = "Drawer.Handle";
function jl({ onDrag: e, onOpenChange: t, ...n }) {
  const { onNestedDrag: r, onNestedOpenChange: o, onNestedRelease: s } = Qe();
  if (!r)
    throw new Error("Drawer.NestedRoot must be placed in another drawer");
  return /* @__PURE__ */ p.createElement(pa, {
    nested: !0,
    onClose: () => {
      o(!1);
    },
    onDrag: (i, l) => {
      r(i, l), e == null || e(i, l);
    },
    onOpenChange: (i) => {
      i && o(i);
    },
    onRelease: s,
    ...n
  });
}
function Nl(e) {
  const t = Qe(), { container: n = t.container, ...r } = e;
  return /* @__PURE__ */ p.createElement(os, {
    container: n,
    ...r
  });
}
const he = {
  Root: pa,
  NestedRoot: jl,
  Content: ga,
  Overlay: ma,
  Trigger: es,
  Portal: Nl,
  Handle: va,
  Close: ts,
  Title: ns,
  Description: rs
}, Zt = ({
  shouldScaleBackground: e = !0,
  ...t
}) => /* @__PURE__ */ a.jsx(
  he.Root,
  {
    shouldScaleBackground: e,
    ...t
  }
);
Zt.displayName = "Drawer";
const wa = he.Trigger, Tl = he.Portal, _l = he.Close, xa = h.forwardRef(({ className: e, ...t }, n) => /* @__PURE__ */ a.jsx(
  he.Overlay,
  {
    ref: n,
    className: O("fixed inset-0 z-50 bg-black/80", e),
    ...t
  }
));
xa.displayName = he.Overlay.displayName;
const en = h.forwardRef(
  ({ className: e, children: t, hideBar: n = !0, ...r }, o) => /* @__PURE__ */ a.jsxs(Tl, { children: [
    /* @__PURE__ */ a.jsx(xa, {}),
    /* @__PURE__ */ a.jsxs(
      he.Content,
      {
        ref: o,
        className: O(
          "fixed inset-x-0 bottom-0 z-50 mt-24 flex h-auto flex-col rounded-t-[10px] border bg-background",
          e
        ),
        ...r,
        children: [
          !n && /* @__PURE__ */ a.jsx("div", { className: "mx-auto mt-4 h-2 w-[100px] rounded-full bg-muted" }),
          t
        ]
      }
    )
  ] })
);
en.displayName = "DrawerContent";
const tn = h.forwardRef(({ className: e, ...t }, n) => /* @__PURE__ */ a.jsx(
  he.Title,
  {
    ref: n,
    className: O(
      "text-lg font-semibold leading-none tracking-tight",
      e
    ),
    ...t
  }
));
tn.displayName = he.Title.displayName;
const Ol = h.forwardRef(({ className: e, ...t }, n) => /* @__PURE__ */ a.jsx(
  he.Description,
  {
    ref: n,
    className: O("text-sm text-muted-foreground", e),
    ...t
  }
));
Ol.displayName = he.Description.displayName;
const ya = () => {
  const e = qe(), [t, n] = Ie(!1), r = qn(() => n(!1), []);
  Ce(() => {
    if (t)
      return;
    function s(i) {
      i.key === "k" && (i.metaKey || i.ctrlKey) && (i.preventDefault(), n(!0));
    }
    return window.addEventListener("keydown", s), () => {
      window.removeEventListener("keydown", s);
    };
  }, [t, n]);
  const o = e.plugins.find(As);
  return o ? /* @__PURE__ */ a.jsxs(a.Fragment, { children: [
    /* @__PURE__ */ a.jsxs(
      "button",
      {
        type: "button",
        onClick: () => n(!0),
        className: "flex items-center border border-input hover:bg-accent hover:text-accent-foreground p-4 relative h-8 justify-start rounded-lg bg-background text-sm text-muted-foreground shadow-none w-full sm:w-72",
        children: [
          /* @__PURE__ */ a.jsxs("div", { className: "flex items-center gap-2 flex-grow", children: [
            /* @__PURE__ */ a.jsx(Ko, { size: 14 }),
            "Search"
          ] }),
          /* @__PURE__ */ a.jsx("kbd", { className: "absolute right-[0.3rem] top-[0.3rem] hidden h-5 select-none items-center gap-1 rounded border bg-muted px-1.5 font-mono text-[11px] font-medium opacity-100 sm:flex", children: "K" })
        ]
      }
    ),
    /* @__PURE__ */ a.jsx(Qn, { fallback: null, children: o.renderSearch({
      isOpen: t,
      onClose: r
    }) })
  ] }) : null;
}, Il = () => {
  const { topNavigation: e } = qe();
  return /* @__PURE__ */ a.jsxs(Zt, { direction: "right", children: [
    /* @__PURE__ */ a.jsx("div", { className: "flex lg:hidden justify-self-end", children: /* @__PURE__ */ a.jsx(wa, { className: "lg:hidden", children: /* @__PURE__ */ a.jsx(qo, { size: 22 }) }) }),
    /* @__PURE__ */ a.jsxs(
      en,
      {
        className: "lg:hidden h-screen right-0 left-auto w-[320px] rounded-none",
        "aria-describedby": void 0,
        children: [
          /* @__PURE__ */ a.jsx(Hn, { children: /* @__PURE__ */ a.jsx(tn, { children: "Navigation" }) }),
          /* @__PURE__ */ a.jsx("div", { className: "flex p-4", children: /* @__PURE__ */ a.jsx(ya, {}) }),
          /* @__PURE__ */ a.jsx("ul", { className: "flex flex-col items-center gap-4 p-4", children: e.map((t) => /* @__PURE__ */ a.jsx("li", { children: /* @__PURE__ */ a.jsx(
            Ze,
            {
              className: ({ isActive: n }) => Un(
                "block font-medium border-b-2",
                n ? "border-primary text-foreground" : "border-transparent text-foreground/75 hover:text-foreground hover:border-accent-foreground/25"
              ),
              to: t.id,
              children: /* @__PURE__ */ a.jsx(_l, { children: t.label })
            }
          ) }, t.label)) })
        ]
      }
    )
  ] });
}, Al = () => {
  const { topNavigation: e } = qe();
  return e.length <= 1 ? /* @__PURE__ */ a.jsx("style", { children: ":root { --top-nav-height: 0px; }" }) : /* @__PURE__ */ a.jsx("nav", { className: "hidden lg:block border-b text-sm px-12 h-[--top-nav-height]", children: /* @__PURE__ */ a.jsx("ul", { className: "flex flex-row items-center gap-8", children: e.map((t) => /* @__PURE__ */ a.jsx("li", { children: /* @__PURE__ */ a.jsx(
    Ze,
    {
      className: ({ isActive: n }) => Un(
        "block py-3.5 font-medium -mb-px border-b-2",
        n ? "border-primary text-foreground" : "border-transparent text-foreground/75 hover:text-foreground hover:border-accent-foreground/25"
      ),
      to: t.id,
      children: t.label
    }
  ) }, t.label)) }) });
}, ba = ({ item: e }) => e.children ? /* @__PURE__ */ a.jsxs(el, { children: [
  /* @__PURE__ */ a.jsx(Zr, { children: e.label }),
  /* @__PURE__ */ a.jsx(Zi, { children: /* @__PURE__ */ a.jsx(ea, { children: e.children.map((t, n) => (
    // eslint-disable-next-line react/no-array-index-key
    /* @__PURE__ */ a.jsx(ba, { item: t }, n)
  )) }) })
] }, e.label) : /* @__PURE__ */ a.jsx(At, { to: e.path ?? "", children: /* @__PURE__ */ a.jsx(na, { children: e.label }, e.label) }), Sa = Vn(function() {
  const t = pn(), [n, r] = po(), { isAuthenticated: o, profile: s, isAuthEnabled: i } = pn(), l = qe(), { page: u, plugins: m } = l, f = m.filter((d) => Os(d)).flatMap((d) => d.getProfileMenuItems(l)).map((d) => /* @__PURE__ */ a.jsx(ba, { item: d }, d.label)), c = n ? Qo : Vo;
  return /* @__PURE__ */ a.jsxs("header", { className: "sticky lg:top-0 z-10 bg-background/80 backdrop-blur w-full", children: [
    /* @__PURE__ */ a.jsx(rl, {}),
    /* @__PURE__ */ a.jsxs("div", { className: "max-w-screen-2xl mx-auto", children: [
      /* @__PURE__ */ a.jsxs("div", { className: "grid grid-cols-2 lg:grid-cols-[calc(var(--side-nav-width))_1fr] lg:gap-12 items-center border-b px-10 lg:px-12 h-[--top-header-height]", children: [
        /* @__PURE__ */ a.jsx("div", { className: "flex", children: /* @__PURE__ */ a.jsx(At, { to: "/", children: /* @__PURE__ */ a.jsxs("div", { className: "flex items-center gap-3.5", children: [
          (u == null ? void 0 : u.logo) && /* @__PURE__ */ a.jsxs(a.Fragment, { children: [
            /* @__PURE__ */ a.jsx(
              "img",
              {
                src: u.logo.src.light,
                alt: u.logo.alt ?? u.pageTitle,
                style: { width: u.logo.width },
                className: O("h-10", n && "hidden"),
                loading: "lazy"
              }
            ),
            /* @__PURE__ */ a.jsx(
              "img",
              {
                src: u.logo.src.dark,
                alt: u.logo.alt ?? u.pageTitle,
                style: { width: u.logo.width },
                className: O("h-10", !n && "hidden"),
                loading: "lazy"
              }
            )
          ] }),
          /* @__PURE__ */ a.jsx("span", { className: "font-bold text-2xl text-foreground/85 tracking-wide", children: u == null ? void 0 : u.pageTitle })
        ] }) }) }),
        /* @__PURE__ */ a.jsxs("div", { className: "grid grid-cols-1 lg:grid-cols-[--sidecar-grid-cols] items-center gap-8", children: [
          /* @__PURE__ */ a.jsx("div", { className: "w-full justify-center hidden lg:flex", children: /* @__PURE__ */ a.jsx(ya, {}) }),
          /* @__PURE__ */ a.jsx(Il, {}),
          /* @__PURE__ */ a.jsxs("div", { className: "hidden lg:flex items-center justify-self-end text-sm gap-2", children: [
            /* @__PURE__ */ a.jsx(xe, { name: "head-navigation-start" }),
            i && !o ? /* @__PURE__ */ a.jsx(Pt, { variant: "ghost", onClick: () => t.login(), children: "Login" }) : f.length > 0 && /* @__PURE__ */ a.jsxs(Xi, { modal: !1, children: [
              /* @__PURE__ */ a.jsx(Ji, { asChild: !0, children: /* @__PURE__ */ a.jsx(Pt, { variant: "ghost", children: s != null && s.email ? `${s.email}` : "My Account" }) }),
              /* @__PURE__ */ a.jsxs(ta, { className: "w-56", children: [
                /* @__PURE__ */ a.jsx(ra, { children: "My Account" }),
                /* @__PURE__ */ a.jsx(aa, {}),
                f
              ] })
            ] }),
            /* @__PURE__ */ a.jsx(
              "button",
              {
                type: "button",
                "aria-label": n ? "Switch to light mode" : "Switch to dark mode",
                className: "cursor-pointer hover:bg-secondary p-2.5 -m-2.5 rounded-full",
                onClick: r,
                children: /* @__PURE__ */ a.jsx(c, { size: 18 })
              }
            ),
            /* @__PURE__ */ a.jsx(xe, { name: "head-navigation-end" })
          ] })
        ] })
      ] }),
      /* @__PURE__ */ a.jsx(xe, { name: "top-navigation-before" }),
      /* @__PURE__ */ a.jsx(Al, {}),
      /* @__PURE__ */ a.jsx(xe, { name: "top-navigation-after" })
    ] })
  ] });
}), Ca = {
  Header: Sa
}, kl = Kn(Ca), $l = kl.Provider, Fl = (e) => {
  const [t, n] = Ie(!1);
  Ce(() => {
    const s = localStorage.getItem("theme"), i = window.matchMedia("(prefers-color-scheme: dark)"), l = s === "dark" || !s && i.matches;
    n(l);
  }, [t]);
  const r = qn(() => {
    const s = !t;
    document.documentElement.classList.toggle("dark", s), localStorage.setItem("theme", s ? "dark" : "light"), n(s);
  }, [t]), o = [t, r];
  return /* @__PURE__ */ a.jsx(mo.Provider, { value: o, ...e });
}, Ll = ({
  children: e,
  context: t
}) => ($o({
  queryFn: async () => (await t.initialize(), !0),
  queryKey: ["zudoku-initialize"]
}), /* @__PURE__ */ a.jsx(Fo.Provider, { value: t, children: e })), Bl = ({
  children: e,
  ...t
}) => {
  var c, d;
  const n = mt(
    () => ({ ...Ca, ...t.overrides }),
    [t.overrides]
  ), r = mt(() => {
    var M;
    return {
      ...(t.plugins ?? []).filter(Fs).flatMap(
        (C) => C.getMdxComponents ? [C.getMdxComponents()] : []
      ).reduce((C, _) => ({ ...C, ..._ }), {}),
      ...go,
      ...(M = t.mdx) == null ? void 0 : M.components
    };
  }, [(c = t.mdx) == null ? void 0 : c.components, t.plugins]), { stagger: o } = Co(jt), [s, i] = Ie(!1), l = mt(
    () => s ? { stagger: !0 } : { stagger: o },
    [o, s]
  ), u = Wa();
  Ce(() => {
    s || i(!0);
  }, [s, u.location]);
  const [m] = Ie(() => new Bs(t)), f = (d = t.plugins) == null ? void 0 : d.filter($s).map((v, M) => {
    var C;
    return /* @__PURE__ */ a.jsx(Mo, { children: (C = v.getHead) == null ? void 0 : C.call(v) }, M);
  });
  return /* @__PURE__ */ a.jsxs(Lo, { client: nr, children: [
    /* @__PURE__ */ a.jsx(It, { children: f }),
    /* @__PURE__ */ a.jsx(jt.Provider, { value: l, children: /* @__PURE__ */ a.jsx(Ll, { context: m, children: /* @__PURE__ */ a.jsx(Ua, { components: r, children: /* @__PURE__ */ a.jsx(Fl, { children: /* @__PURE__ */ a.jsx($l, { value: n, children: /* @__PURE__ */ a.jsx(Xa, { slotlets: t.slotlets, children: /* @__PURE__ */ a.jsx(vo, { children: e ?? /* @__PURE__ */ a.jsx(In, {}) }) }) }) }) }) }) })
  ] });
}, Hl = Vn(Bl), Ma = (e) => /* @__PURE__ */ a.jsx(Ts, { FallbackComponent: Hs, children: /* @__PURE__ */ a.jsx(Hl, { ...e }) });
Ma.displayName = "DevPortal";
const Ul = ({
  category: e,
  level: t
}) => {
  var d;
  const n = Yn(), r = za(e), [o, s] = Ie(!1), i = e.collapsible ?? !0, l = e.collapsed ?? !0, u = !!(!i || !l || r), [m, f] = Ie(u);
  Ce(() => {
    r && f(!0);
  }, [r]);
  const c = i && /* @__PURE__ */ a.jsx(
    "button",
    {
      type: "button",
      onClick: (v) => {
        v.preventDefault(), f((M) => !M), s(!0);
      },
      children: /* @__PURE__ */ a.jsx(
        Xn,
        {
          size: 16,
          className: O(
            o && "transition",
            "shrink-0 group-data-[state=open]:rotate-90"
          )
        }
      )
    }
  );
  return /* @__PURE__ */ a.jsxs(
    Do,
    {
      className: O("flex flex-col", t === 0 && "-mx-[--padding-nav-item]"),
      defaultOpen: u,
      open: m,
      onOpenChange: () => f(!0),
      children: [
        /* @__PURE__ */ a.jsx(Po, { className: "group", asChild: !0, disabled: !i, children: /* @__PURE__ */ a.jsxs(
          "div",
          {
            className: O(
              "text-start",
              Je({ isActive: !1, isTopLevel: t === 0 }),
              i ? "cursor-pointer" : "cursor-default hover:bg-transparent"
            ),
            children: [
              e.icon && /* @__PURE__ */ a.jsx(
                e.icon,
                {
                  size: 16,
                  className: "align-[-0.125em] -translate-x-1"
                }
              ),
              ((d = e.link) == null ? void 0 : d.type) === "doc" ? /* @__PURE__ */ a.jsx(
                Ze,
                {
                  to: Ht(n == null ? void 0 : n.id, e.link.id),
                  className: "flex-1",
                  onClick: () => s(!0),
                  children: ({ isActive: v }) => /* @__PURE__ */ a.jsxs(
                    "div",
                    {
                      className: O(
                        "flex items-center gap-2 justify-between w-full",
                        v ? "text-primary font-medium" : "text-foreground/80"
                      ),
                      children: [
                        /* @__PURE__ */ a.jsx("div", { className: "truncate", children: e.label }),
                        c
                      ]
                    }
                  )
                }
              ) : /* @__PURE__ */ a.jsxs("div", { className: "flex items-center justify-between w-full", children: [
                /* @__PURE__ */ a.jsx("div", { className: "flex gap-2 truncate w-full", children: e.label }),
                c
              ] })
            ]
          }
        ) }),
        /* @__PURE__ */ a.jsx(
          jo,
          {
            className: O(
              // CollapsibleContent class is used to animate and it should only be applied when the user has triggered the toggle
              o && "CollapsibleContent",
              "ms-[calc(var(--padding-nav-item)*1.125)]"
            ),
            children: /* @__PURE__ */ a.jsx("ul", { className: "mt-1 border-l ps-2", children: e.items.map((v) => /* @__PURE__ */ a.jsx(
              Ot,
              {
                level: t + 1,
                item: v
              },
              ("id" in v ? v.id : "") + ("href" in v ? v.href : "") + v.label
            )) })
          }
        )
      ]
    }
  );
}, Je = wo(
  "flex items-center gap-2 px-[--padding-nav-item] py-1.5 rounded-lg hover:bg-accent transition-colors duration-300",
  {
    variants: {
      isTopLevel: {
        true: "font-semibold"
      },
      isActive: {
        true: "text-primary font-medium",
        false: "text-foreground/80"
      },
      isMuted: {
        true: "text-foreground/30",
        false: ""
      }
    }
  }
), Ra = "data-anchor", Ot = ({
  item: e,
  level: t = 0
}) => {
  const n = Yn(), { activeAnchor: r } = Ft(), [o] = Ya();
  switch (e.type) {
    case "category":
      return /* @__PURE__ */ a.jsx(Ul, { category: e, level: t });
    case "doc":
      return /* @__PURE__ */ a.jsxs(
        Ze,
        {
          className: ({ isActive: s }) => Je({ isActive: s, isTopLevel: t === 0 }),
          to: Ht(n == null ? void 0 : n.id, e.id),
          children: [
            e.icon && /* @__PURE__ */ a.jsx(e.icon, { size: 16, className: "align-[-0.125em]" }),
            e.badge ? /* @__PURE__ */ a.jsxs(a.Fragment, { children: [
              /* @__PURE__ */ a.jsx("span", { className: "truncate", title: e.label, children: e.label }),
              /* @__PURE__ */ a.jsx(St, { ...e.badge })
            ] }) : e.label
          ]
        }
      );
    case "link":
      return e.href.startsWith("#") ? /* @__PURE__ */ a.jsx(
        Ka,
        {
          to: { hash: e.href, search: o.toString() },
          [Ra]: e.href.slice(1),
          className: O(
            "flex gap-2.5 justify-between",
            t === 0 && "-mx-[--padding-nav-item]",
            Je({
              isActive: e.href.slice(1) === r
            })
          ),
          children: e.badge ? /* @__PURE__ */ a.jsxs(a.Fragment, { children: [
            /* @__PURE__ */ a.jsx("span", { className: "truncate", title: e.label, children: e.label }),
            /* @__PURE__ */ a.jsx(St, { ...e.badge })
          ] }) : /* @__PURE__ */ a.jsx("span", { className: "break-all", children: e.label })
        }
      ) : e.href.startsWith("http") ? /* @__PURE__ */ a.jsxs(
        "a",
        {
          className: O(
            Je({ isTopLevel: t === 0 }),
            "block"
          ),
          href: e.href,
          target: "_blank",
          rel: "noopener noreferrer",
          children: [
            /* @__PURE__ */ a.jsx("span", { className: "whitespace-normal", children: e.label }),
            /* @__PURE__ */ a.jsxs("span", { className: "whitespace-nowrap", children: [
              "",
              /* @__PURE__ */ a.jsx(Go, { className: "inline ml-1", size: 12 })
            ] })
          ]
        }
      ) : /* @__PURE__ */ a.jsx(
        Ze,
        {
          className: O("flex gap-2.5 justify-between", Je()),
          to: e.href,
          children: e.badge ? /* @__PURE__ */ a.jsxs(a.Fragment, { children: [
            /* @__PURE__ */ a.jsx("span", { className: "truncate", title: e.label, children: e.label }),
            /* @__PURE__ */ a.jsx(St, { ...e.badge })
          ] }) : /* @__PURE__ */ a.jsx("span", { className: "break-all", children: e.label })
        }
      );
  }
}, zl = (e, t = { block: "center" }) => {
  if (!e) return;
  const n = e.getBoundingClientRect();
  n.top >= 0 && n.left >= 0 && n.bottom <= (window.innerHeight || document.documentElement.clientHeight) && n.right <= (window.innerWidth || document.documentElement.clientWidth) || e.scrollIntoView(t);
}, Kl = () => {
  const e = kt(), { setActiveAnchor: t } = Ft();
  Ce(() => {
    if (!e.hash) return;
    const n = decodeURIComponent(e.hash.split("/")[0].slice(1)), r = () => {
      const o = document.getElementById(n), s = document.querySelector(`[${Ra}="${n}"]`);
      return o ? (o.scrollIntoView(), zl(s), requestIdleCallback(() => t(n)), !0) : !1;
    };
    if (!r()) {
      const o = new MutationObserver((s, i) => {
        r() && i.disconnect();
      });
      return o.observe(document.body, { childList: !0, subtree: !0 }), () => o.disconnect();
    }
  }, [e.hash, t]);
}, ql = () => {
  const e = kt(), t = Lt(e.pathname);
  Ce(() => {
    t.current !== e.pathname && (window.scrollTo(0, 0), t.current = e.pathname);
  }, [e.pathname]);
}, Ea = Ro(({ children: e, className: t, pushMainContent: n }, r) => /* @__PURE__ */ a.jsx(
  "nav",
  {
    "data-navigation": String(n),
    className: O(
      "scrollbar peer hidden lg:flex flex-col fixed text-sm overflow-y-auto shrink-0",
      "px-[--padding-nav-item] -mx-[--padding-nav-item] pb-20 mt-[--padding-content-top]",
      "w-[--side-nav-width] h-[calc(100%-var(--header-height))] scroll-pt-2 gap-2",
      t
    ),
    ref: r,
    children: e
  }
));
Ea.displayName = "SidebarWrapper";
const Ql = () => {
  const e = Lt(null), t = Bo();
  return /* @__PURE__ */ a.jsxs(a.Fragment, { children: [
    /* @__PURE__ */ a.jsxs(
      Ea,
      {
        ref: e,
        pushMainContent: t.data.items.length > 0,
        children: [
          /* @__PURE__ */ a.jsx(xe, { name: "zudoku-before-navigation" }),
          t.data.items.map((n) => /* @__PURE__ */ a.jsx(Ot, { item: n }, n.label)),
          /* @__PURE__ */ a.jsx(xe, { name: "zudoku-after-navigation" })
        ]
      }
    ),
    /* @__PURE__ */ a.jsxs(
      en,
      {
        className: "lg:hidden h-screen left-0 p-6 w-[320px] rounded-none",
        "aria-describedby": void 0,
        children: [
          /* @__PURE__ */ a.jsx(Hn, { children: /* @__PURE__ */ a.jsx(tn, { children: "Sidebar" }) }),
          t.data.items.map((n) => /* @__PURE__ */ a.jsx(Ot, { item: n }, n.label))
        ]
      }
    )
  ] });
}, Vl = ({ children: e }) => {
  const t = kt(), { setActiveAnchor: n } = Ft(), { meta: r, authentication: o } = qe();
  Kl(), ql();
  const s = Lt(t.pathname);
  return Ce(() => {
    var i;
    (i = o == null ? void 0 : o.pageLoad) == null || i.call(o);
  }, [o]), Ce(() => {
    t.pathname !== s.current && n(""), s.current = t.pathname;
  }, [t.pathname, n]), /* @__PURE__ */ a.jsxs(a.Fragment, { children: [
    !1,
    /* @__PURE__ */ a.jsxs(It, { titleTemplate: r == null ? void 0 : r.title, children: [
      (r == null ? void 0 : r.description) && /* @__PURE__ */ a.jsx("meta", { name: "description", content: r.description }),
      (r == null ? void 0 : r.favicon) && /* @__PURE__ */ a.jsx("link", { rel: "icon", href: r.favicon })
    ] }),
    /* @__PURE__ */ a.jsx(xe, { name: "layout-before-head" }),
    /* @__PURE__ */ a.jsx(Sa, {}),
    /* @__PURE__ */ a.jsx(xe, { name: "layout-after-head" }),
    /* @__PURE__ */ a.jsx("div", { className: "w-full max-w-screen-2xl mx-auto px-10 lg:px-12", children: /* @__PURE__ */ a.jsx(
      Qn,
      {
        fallback: /* @__PURE__ */ a.jsx("main", { className: "grid h-[calc(100vh-var(--header-height))] place-items-center", children: /* @__PURE__ */ a.jsx(ss, {}) }),
        children: /* @__PURE__ */ a.jsxs(Zt, { direction: "left", children: [
          /* @__PURE__ */ a.jsx(Ql, {}),
          /* @__PURE__ */ a.jsx(
            "div",
            {
              className: O(
                "lg:hidden -mx-10 px-10 py-2 sticky bg-background/80 backdrop-blur z-10 top-0 left-0 right-0 border-b",
                "peer-data-[navigation=false]:hidden"
              ),
              children: /* @__PURE__ */ a.jsxs(wa, { className: "flex items-center gap-2", children: [
                /* @__PURE__ */ a.jsx(Wo, { size: 16, strokeWidth: 1.5 }),
                /* @__PURE__ */ a.jsx("span", { className: "text-sm", children: "Menu" })
              ] })
            }
          ),
          /* @__PURE__ */ a.jsxs(
            "main",
            {
              className: O(
                "h-full dark:border-white/10 translate-x-0",
                "lg:overflow-visible",
                // This works in tandem with the `SidebarWrapper` component
                "lg:peer-data-[navigation=true]:w-[calc(100%-var(--side-nav-width))]",
                "lg:peer-data-[navigation=true]:translate-x-[--side-nav-width] lg:peer-data-[navigation=true]:pl-12"
              ),
              children: [
                /* @__PURE__ */ a.jsx(xe, { name: "zudoku-before-content" }),
                e ?? /* @__PURE__ */ a.jsx(In, {}),
                /* @__PURE__ */ a.jsx(xe, { name: "zudoku-after-content" })
              ]
            }
          )
        ] })
      }
    ) })
  ] });
}, dc = qa, fc = xo, hc = Ma, pc = Vl, mc = At, gc = Ja, vc = is, wc = Ms, xc = Rs, yc = Pt, bc = It;
export {
  wc as Bootstrap,
  xc as BootstrapStatic,
  yc as Button,
  fc as Callout,
  hc as DevPortal,
  bc as Head,
  pc as Layout,
  mc as Link,
  gc as RouterError,
  vc as ServerError,
  dc as useMDXComponents
};
