// zudoku/lib/zudoku.plugin-search-inkeep.js
import { j as n } from "./jsx-runtime-B6kdoens.js";
import { useSyncExternalStore as a, lazy as i } from "react";
const o = () => () => {
}, c = (e) => a(
  o,
  () => "client",
  () => "server"
) === "client" ? e.children : null, l = {
  theme: {
    components: {
      AIChatPageWrapper: {
        defaultProps: {
          size: "shrink-vertically",
          variant: "no-shadow"
        }
      },
      SearchBarTrigger: {
        defaultProps: {
          size: "expand",
          variant: "subtle"
          // Choose from 'emphasized' or 'subtle'
        }
      }
    }
  }
}, p = {}, u = i(() => import("./InkeepCustomTrigger-CE5-K5ex.js")), h = ({
  prefilledQuery: e,
  isOpen: r,
  onClose: t,
  settings: s
}) => /* @__PURE__ */ n.jsx(
  u,
  {
    isOpen: r,
    onClose: t,
    baseSettings: { ...l, ...s },
    aiChatSettings: p,
    searchSettings: {
      prefilledQuery: e || void 0
    }
  }
), g = (e) => ({
  renderSearch: ({
    isOpen: r,
    onClose: t
  }) => /* @__PURE__ */ n.jsx(c, { children: /* @__PURE__ */ n.jsx(h, { isOpen: r, onClose: t, settings: e }) })
});
export {
  g as inkeepSearchPlugin
};
